# Pipeline started at 07-31 20:43:16

ln -sf /scratch/jps3dp/DATA/proseq/tutorial_r1.fq.gz /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/raw/tutorial_R1.fastq.gz
ln -sf /scratch/jps3dp/DATA/proseq/tutorial_r2.fq.gz /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/raw/tutorial_R2.fastq.gz
gzip -f -d -c /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/raw/tutorial_R1.fastq.gz > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R1.fastq
gzip -f -d -c /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/raw/tutorial_R2.fastq.gz > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R2.fastq
(fastp --overrepresentation_analysis --thread 1 --in1 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R1.fastq --adapter_sequence TGGAATTCTCGGGTGCCAAGG --length_required 26 --html /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastqc/tutorial_R1_rmAdapter.html --json /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastqc/tutorial_R1_rmAdapter.json --report_title 'tutorial' -o /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R1_noadap.fastq ) 2> /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastqc/tutorial_R1_rmAdapter.txt
seqkit rmdup --threads 1 --by-seq --ignore-case -o /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R1_dedup.fastq /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R1_noadap.fastq
seqtk trimfq -b 8 -L 30 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R1_noadap.fastq | seqtk seq -r - > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R1_trimmed.fastq
grep 'reads with adapter trimmed:' /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastqc/tutorial_R1_rmAdapter.txt | head -n 1 | awk '{print $NF}'
grep 'total bases:' /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastqc/tutorial_R1_rmAdapter.txt | head -n 1 | awk '{print $NF}'
grep 'bases trimmed due to adapters:' /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastqc/tutorial_R1_rmAdapter.txt | awk '{print $NF}'
grep 'reads failed due to too short:' /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastqc/tutorial_R1_rmAdapter.txt | head -n 1 | awk '{print $NF}'
seqtk trimfq -b 8 -L 30 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R1_dedup.fastq | seqtk seq -r - > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R1_processed.fastq
(fastp --overrepresentation_analysis --thread 1 --in1 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R2.fastq --adapter_sequence GATCGTCGGACTGTAGAACTCTGAAC --length_required 26 --html /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastqc/tutorial_R2_rmAdapter.html --json /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastqc/tutorial_R2_rmAdapter.json --report_title 'tutorial' --stdout ) 2> /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastqc/tutorial_R1_rmAdapter.txt | seqtk trimfq -e 8 - | seqtk seq -r - > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R2_trimmed.fastq
cp /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R2_trimmed.fastq /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R2_trimmed_dups.fastq
fastq_pair -t 1800000 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R1_processed.fastq /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R2_trimmed.fastq
mv /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R1_processed.fastq.paired.fq /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R1_processed.fastq
mv /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R2_trimmed.fastq.paired.fq /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R2_trimmed.fastq
touch repaired.flag
fastq_pair -t 1800000 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R1_trimmed.fastq /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R2_trimmed_dups.fastq
mv /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R1_trimmed.fastq.paired.fq /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R1_trimmed.fastq
mv /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R2_trimmed_dups.fastq.paired.fq /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R2_trimmed_dups.fastq
touch dups_repaired.flag
mkfifo /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/human_rDNA_bt2
perl /scratch/jps3dp/tools/databio/peppro/tools/filter_paired_fq.pl /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/human_rDNA_bt2 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R1_processed.fastq /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R2_trimmed.fastq /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_unmap_R1.fq /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_unmap_R2.fq
(bowtie2 -p 1 -k 1 -D 20 -R 3 -N 1 -L 20 -i S,1,0.50 -x /nv/t1/genomes/human_rDNA/bowtie2_index/human_rDNA --rg-id tutorial -U /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R1_processed.fastq --un /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/human_rDNA_bt2 > /dev/null) 2>/sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_bt_aln_summary.log
grep 'aligned exactly 1 time' /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_bt_aln_summary.log | awk '{print $1}'
mkfifo /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/human_rDNA_dups_bt2
perl /scratch/jps3dp/tools/databio/peppro/tools/filter_paired_fq.pl /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/human_rDNA_dups_bt2 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R1_trimmed.fastq /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R2_trimmed_dups.fastq /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_unmap_dups_R1.fq /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_unmap_dups_R2.fq
(bowtie2 -p 1 -k 1 -D 20 -R 3 -N 1 -L 20 -i S,1,0.50 -x /nv/t1/genomes/human_rDNA/bowtie2_index/human_rDNA --rg-id tutorial -U /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/fastq/tutorial_R1_trimmed.fastq --un /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/human_rDNA_dups_bt2 > /dev/null) 2>/sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_bt_aln_dups_summary.log
mkfifo /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/rCRSd_bt2
perl /scratch/jps3dp/tools/databio/peppro/tools/filter_paired_fq.pl /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/rCRSd_bt2 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_unmap_R1.fq /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_unmap_R2.fq /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_rCRSd_unmap_R1.fq /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_rCRSd_unmap_R2.fq
(bowtie2 -p 1 -k 1 -D 20 -R 3 -N 1 -L 20 -i S,1,0.50 -x /nv/t1/genomes/rCRSd/bowtie2_index/rCRSd --rg-id tutorial -U /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_unmap_R1.fq --un /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/rCRSd_bt2 > /dev/null) 2>/sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_rCRSd_bt_aln_summary.log
grep 'aligned exactly 1 time' /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_rCRSd_bt_aln_summary.log | awk '{print $1}'
mkfifo /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/rCRSd_dups_bt2
perl /scratch/jps3dp/tools/databio/peppro/tools/filter_paired_fq.pl /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/rCRSd_dups_bt2 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_unmap_dups_R1.fq /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_unmap_dups_R2.fq /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_rCRSd_unmap_dups_R1.fq /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_rCRSd_unmap_dups_R2.fq
(bowtie2 -p 1 -k 1 -D 20 -R 3 -N 1 -L 20 -i S,1,0.50 -x /nv/t1/genomes/rCRSd/bowtie2_index/rCRSd --rg-id tutorial -U /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_unmap_dups_R1.fq --un /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/rCRSd_dups_bt2 > /dev/null) 2>/sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_rCRSd_bt_aln_dups_summary.log
bowtie2 -p 1 --very-sensitive -X 2000 --rg-id tutorial -x /nv/t1/genomes/hg38/bowtie2_index/hg38 --rf -1 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_rCRSd_unmap_R1.fq -2 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_rCRSd_unmap_R2.fq | samtools view -bS - -@ 1  | samtools sort - -@ 1 -T /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tmplh2_a83a -o /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_temp.bam
samtools view -q 10 -b -@ 1 -U /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_fail_qc.bam /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_temp.bam > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort.bam
samtools depth -b /nv/t1/genomes/hg38/refgene_anno/hg38_pre-mRNA.bed /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort.bam | awk '{counter++;sum+=$3}END{print sum/counter}'
bowtie2 -p 1 --very-sensitive -X 2000 --rg-id tutorial -x /nv/t1/genomes/hg38/bowtie2_index/hg38 --rf -1 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_rCRSd_unmap_dups_R1.fq -2 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_rCRSd_unmap_dups_R2.fq | samtools view -bS - -@ 1  | samtools sort - -@ 1 -T /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tmplh2_a83a -o /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_temp_dups.bam
samtools view -q 10 -b -@ 1 -U /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_fail_qc_dups.bam /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_temp_dups.bam > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort_dups.bam
gzip -f /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_unmap_R1.fq
gzip -f /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_unmap_R2.fq
gzip -f /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_rCRSd_unmap_R1.fq
gzip -f /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/prealignments/tutorial_rCRSd_unmap_R2.fq
samtools index /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_temp.bam
samtools idxstats /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_temp.bam | grep -we 'chrM' -we 'chrMT' -we 'M' -we 'MT' -we 'rCRSd' -we 'rCRSd_3k'| cut -f 3
samtools index /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort.bam
samtools idxstats /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort.bam | cut -f 1 | grep -vwe 'chrM' -vwe 'chrMT' -vwe 'M' -vwe 'MT' -vwe 'rCRSd' -vwe 'rCRSd_3k'| xargs samtools view -b -@ 1 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort.bam > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_noMT.bam
mv /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_noMT.bam /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort.bam
samtools index /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort.bam
samtools stats /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort.bam | grep '^SN' | cut -f 2- | grep 'maximum length:' | cut -f 2-
samtools view -b -f 64 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort.bam | samtools sort - -@ 1 > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam
samtools view -b -f 128 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort.bam | samtools sort - -@ 1 > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE2.bam
samtools index /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_temp_dups.bam
samtools idxstats /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_temp_dups.bam | grep -we 'chrM' -we 'chrMT' -we 'M' -we 'MT' -we 'rCRSd' -we 'rCRSd_3k'| cut -f 3
samtools index /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort_dups.bam
samtools idxstats /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort_dups.bam | cut -f 1 | grep -vwe 'chrM' -vwe 'chrMT' -vwe 'M' -vwe 'MT' -vwe 'rCRSd' -vwe 'rCRSd_3k'| xargs samtools view -b -@ 1 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort_dups.bam > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_noMT_dups.bam
mv /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_noMT_dups.bam /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort_dups.bam
samtools index /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort_dups.bam
samtools view -b -f 64 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort_dups.bam | samtools sort - -@ 1 > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_dups_PE1.bam
samtools view -b -f 128 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort_dups.bam | samtools sort - -@ 1 > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_dups_PE2.bam
preseq c_curve -v -o /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_preseq_out.txt -B /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_dups_PE1.bam
preseq lc_extrap -v -o /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_preseq_yield.txt -B /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_dups_PE1.bam
echo '/sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_preseq_yield.txt '$(samtools view -c -F 4 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_dups_PE1.bam)' '$(samtools view -c -F 4 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam) > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_preseq_counts.txt
awk '{sum+=$2} END {printf "%.0f", sum}' /nv/t1/genomes/hg38/hg38.chrom.sizes
Rscript /scratch/jps3dp/tools/databio/peppro/tools/PEPPRO.R preseq -i /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_preseq_yield.txt -r /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_preseq_counts.txt -o /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_preseq_plot
samtools index /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam
/scratch/jps3dp/tools/databio/peppro/tools/bamQC.py -i /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam -c 1 -o /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_bamQC.tsv
awk '{ for (i=1; i<=NF; ++i) { if ($i ~ "NRF") c=i } getline; print $c }' /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_bamQC.tsv
awk '{ for (i=1; i<=NF; ++i) { if ($i ~ "PBC1") c=i } getline; print $c }' /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_bamQC.tsv
awk '{ for (i=1; i<=NF; ++i) { if ($i ~ "PBC2") c=i } getline; print $c }' /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_bamQC.tsv
samtools view -b -@ 1 -f 12  /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_temp.bam > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_unmap.bam
samtools view -c -f 4 -@ 1 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_temp.bam
samtools view -bh -F 20 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam
samtools view -bh -f 16 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam
sed -n -e '/[[:space:]]+/w /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/plus_TSS.tsv' -e '/[[:space:]]-/w /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/minus_TSS.tsv' /nv/t1/genomes/hg38/refgene_anno/hg38_TSS.bed
/scratch/jps3dp/tools/databio/peppro/tools/pyTssEnrichment.py -a /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam -b /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/plus_TSS.tsv -p ends -c 1 -z -v -s 6 -o /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_plus_TssEnrichment.txt
/scratch/jps3dp/tools/databio/peppro/tools/pyTssEnrichment.py -a /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam -b /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/minus_TSS.tsv -p ends -c 1 -z -v -s 6 -o /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_minus_TssEnrichment.txt
Rscript /scratch/jps3dp/tools/databio/peppro/tools/PEPPRO.R tss -i /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_plus_TssEnrichment.txt /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_minus_TssEnrichment.txt
samtools view -H /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam | grep 'SN:' | awk -F':' '{print $2,$3}' | awk -F' ' -v OFS='	' '{print $1,$3}' > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt
cut -f 1 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_keep.txt
grep -wf /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_keep.txt /nv/t1/genomes/hg38/ensembl_gtf/hg38_ensembl_TSS.bed | bedtools sort -i stdin -faidx /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/hg38_ensembl_tss.bed
grep -wf /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_keep.txt /nv/t1/genomes/hg38/ensembl_gtf/hg38_ensembl_gene_body.bed | bedtools sort -i stdin -faidx /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/hg38_ensembl_gene_body.bed
bedtools coverage -sorted -counts -s -a /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/hg38_ensembl_tss.bed -b /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam -g /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt | awk '$7>0' | sort -k4,4 -k7,7nr | sort -k4,4 -u > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_TSS_density.bed
bedtools coverage -sorted -counts -s -a /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/hg38_ensembl_gene_body.bed -b /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam -g /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt | awk '$7>0' | sort -k4 > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_gene_body_density.bed
join -j4 -o 1.1 1.2 1.3 1.4 1.6 1.7 2.2 2.3 2.7 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_TSS_density.bed /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_gene_body_density.bed | awk '{print ($6/($3-$2))/($9/($8-$7))}' > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_pause_index.txt
sort -n /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_pause_index.txt | awk ' { a[i++]=$1; } END { x=int((i+1)/2); if (x < (i+1)/2) print (a[x-1]+a[x])/2; else print a[x-1]; }'
Rscript /scratch/jps3dp/tools/databio/peppro/tools/PEPPRO.R pi -i /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_pause_index.txt
samtools view -@ 4 -c -L /nv/t1/genomes/hg38/refgene_anno/hg38_pre-mRNA.bed /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam
samtools view -@ 4 -c -L /nv/t1/genomes/hg38/refgene_anno/hg38_pre-mRNA.bed /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam
perl /scratch/jps3dp/tools/databio/peppro/tools/fragment_length_dist.pl /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_fragLen.txt
sort -n  /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_fragLen.txt | uniq -c  > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_fragCount.txt
Rscript /scratch/jps3dp/tools/databio/peppro/tools/PEPPRO.R frag -l /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_fragLen.txt -c /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_fragCount.txt -p /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_fragLenDistribution.pdf -t /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_fragLenDistribution.txt
ln -sf /nv/t1/genomes/hg38/feat_annotation/hg38_annotations.bed.gz /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/raw/hg38_annotations.bed.gz
gzip -f -d -c /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/raw/hg38_annotations.bed.gz > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/raw/hg38_annotations.bed
cut -f 4 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/raw/hg38_annotations.bed | sort -u
awk -F'	' '{print>"/sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/"$4}' /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/raw/hg38_annotations.bed
mv "/sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/3' UTR" "/sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/3_UTR"
cut -f 1 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt | grep -wf - /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/3_UTR | cut -f 1-3 | bedtools sort -i stdin -faidx /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/3_UTR_sort.bed
bedtools coverage -sorted -counts -a /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/3_UTR_sort.bed -b /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam -g /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_3_UTR_plus_coverage.bed
bedtools coverage -sorted -counts -a /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/3_UTR_sort.bed -b /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam -g /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_3_UTR_minus_coverage.bed
mv "/sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/5' UTR" "/sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/5_UTR"
cut -f 1 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt | grep -wf - /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/5_UTR | cut -f 1-3 | bedtools sort -i stdin -faidx /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/5_UTR_sort.bed
bedtools coverage -sorted -counts -a /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/5_UTR_sort.bed -b /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam -g /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_5_UTR_plus_coverage.bed
bedtools coverage -sorted -counts -a /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/5_UTR_sort.bed -b /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam -g /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_5_UTR_minus_coverage.bed
cut -f 1 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt | grep -wf - /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/Enhancer | cut -f 1-3 | bedtools sort -i stdin -faidx /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/Enhancer_sort.bed
bedtools coverage -sorted -counts -a /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/Enhancer_sort.bed -b /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam -g /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_Enhancer_plus_coverage.bed
bedtools coverage -sorted -counts -a /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/Enhancer_sort.bed -b /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam -g /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_Enhancer_minus_coverage.bed
cut -f 1 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt | grep -wf - /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/Exon | cut -f 1-3 | bedtools sort -i stdin -faidx /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/Exon_sort.bed
bedtools coverage -sorted -counts -a /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/Exon_sort.bed -b /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam -g /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_Exon_plus_coverage.bed
bedtools coverage -sorted -counts -a /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/Exon_sort.bed -b /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam -g /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_Exon_minus_coverage.bed
cut -f 1 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt | grep -wf - /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/Intron | cut -f 1-3 | bedtools sort -i stdin -faidx /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/Intron_sort.bed
bedtools coverage -sorted -counts -a /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/Intron_sort.bed -b /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam -g /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_Intron_plus_coverage.bed
bedtools coverage -sorted -counts -a /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/Intron_sort.bed -b /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam -g /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_Intron_minus_coverage.bed
cut -f 1 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt | grep -wf - /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/Promoter | cut -f 1-3 | bedtools sort -i stdin -faidx /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/Promoter_sort.bed
bedtools coverage -sorted -counts -a /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/Promoter_sort.bed -b /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam -g /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_Promoter_plus_coverage.bed
bedtools coverage -sorted -counts -a /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/Promoter_sort.bed -b /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam -g /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_Promoter_minus_coverage.bed
mv "/sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/Promoter Flanking Region" "/sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/Promoter_Flanking_Region"
cut -f 1 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt | grep -wf - /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/Promoter_Flanking_Region | cut -f 1-3 | bedtools sort -i stdin -faidx /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/Promoter_Flanking_Region_sort.bed
bedtools coverage -sorted -counts -a /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/Promoter_Flanking_Region_sort.bed -b /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam -g /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_Promoter_Flanking_Region_plus_coverage.bed
bedtools coverage -sorted -counts -a /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/Promoter_Flanking_Region_sort.bed -b /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam -g /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_Promoter_Flanking_Region_minus_coverage.bed
samtools view -@ 1 -q 10 -c -F4 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam
Rscript /scratch/jps3dp/tools/databio/peppro/tools/PEPPRO.R frif -n tutorial -r 166413 -o /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_plus_frif.pdf --bed /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_3_UTR_plus_coverage.bed /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_5_UTR_plus_coverage.bed /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_Enhancer_plus_coverage.bed /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_Exon_plus_coverage.bed /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_Intron_plus_coverage.bed /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_Promoter_plus_coverage.bed /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_Promoter_Flanking_Region_plus_coverage.bed
samtools view -@ 1 -q 10 -c -F4 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam
Rscript /scratch/jps3dp/tools/databio/peppro/tools/PEPPRO.R frif -n tutorial -r 163584 -o /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_minus_frif.pdf --bed /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_3_UTR_minus_coverage.bed /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_5_UTR_minus_coverage.bed /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_Enhancer_minus_coverage.bed /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_Exon_minus_coverage.bed /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_Intron_minus_coverage.bed /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_Promoter_minus_coverage.bed /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_Promoter_Flanking_Region_minus_coverage.bed
grep -wf /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_keep.txt /nv/t1/genomes/hg38/refgene_anno/hg38_exons.bed | bedtools sort -i stdin -faidx /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/hg38_exons_sort.bed
grep -wf /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_keep.txt /nv/t1/genomes/hg38/refgene_anno/hg38_introns.bed | bedtools sort -i stdin -faidx /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt | bedtools sort -i stdin -faidx /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/hg38_introns_sort.bed
bedtools coverage -sorted -counts -s -a /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/hg38_exons_sort.bed -b /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam -g /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_exons_coverage.bed
bedtools coverage -sorted -counts -s -a /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/hg38_introns_sort.bed -b /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam -g /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_introns_coverage.bed
awk -v OFS='	' '{readCount[$4] += $7; exonCount[$4] += 1; geneSizeKB[$4] += (sqrt(($3-$2+0.00000001)^2)/1000); gene[$4] = $4} END { for (a in readCount) { print gene[a], (readCount[a]/0.326735)/geneSizeKB[a]}}' /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_exons_coverage.bed | awk '$2>0' | sort -k1 > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_exons_rpkm.tsv
awk -v OFS='	' '{readCount[$4] += $7; exonCount[$4] += 1; geneSizeKB[$4] += (sqrt(($3-$2+0.00000001)^2)/1000); gene[$4] = $4} END { for (a in readCount) { print gene[a], (readCount[a]/0.326735)/geneSizeKB[a]}}' /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_introns_coverage.bed | awk '$2>0' | sort -k1 > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_introns_rpkm.tsv
join -a1 -a2 -j1 -e0 -o 0 1.2 2.2 /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_introns_rpkm.tsv /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_exons_rpkm.tsv > /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_intron_exon.tsv
awk '$2>0' /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_intron_exon.tsv | awk '{print $3/$2}' | sort -n | awk ' { a[i++]=$1; } END { x=int((i+1)/2); if (x < (i+1)/2) print (a[x-1]+a[x])/2; else print a[x-1]; }'
Rscript /scratch/jps3dp/tools/databio/peppro/tools/PEPPRO.R mrna -i /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/QC_hg38/tutorial_intron_exon.tsv --raw
samtools index /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam
/scratch/jps3dp/tools/databio/peppro/tools/bamSitesToWig.py -i /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam -c /nv/t1/genomes/hg38/hg38.chrom.sizes -o /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/signal_hg38/tutorial_plus_body_0-mer.bw -p 1 --variable-step --tail-edge
samtools index /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam
/scratch/jps3dp/tools/databio/peppro/tools/bamSitesToWig.py -i /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam -c /nv/t1/genomes/hg38/hg38.chrom.sizes -o /sfs/lustre/allocations/shefflab/processed/peppro_tutorial/pe/07-31-19/results_pipeline/tutorial/signal_hg38/tutorial_minus_body_0-mer.bw -p 1 --variable-step --tail-edge
