# Pipeline started at 11-27 13:54:59

ln -sf /scratch/jps3dp/DATA/proseq/data/fastq/tutorial_r1.fq.gz /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/raw/tutorial_R1.fastq.gz
ln -sf /scratch/jps3dp/DATA/proseq/data/fastq/tutorial_r2.fq.gz /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/raw/tutorial_R2.fastq.gz
gzip -f -d -c /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/raw/tutorial_R1.fastq.gz > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastq/tutorial_R1.fastq
gzip -f -d -c /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/raw/tutorial_R2.fastq.gz > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastq/tutorial_R2.fastq
( (fastp --overrepresentation_analysis --thread 1 --in1 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastq/tutorial_R1.fastq --adapter_sequence TGGAATTCTCGGGTGCCAAGG --length_required 18 --html /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastp/tutorial_R1_fastp_adapter.html --json /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastp/tutorial_R1_fastp_adapter.json --report_title 'tutorial' --stdout ) 2> /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastp/tutorial_R1_fastp_adapter.txt | seqtk trimfq -L 30 - | seqtk seq -r - > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastq/tutorial_R1_processed.fastq ) 2> /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastqc/tutorial_R1_rmAdapter.txt
fastqc --noextract --outdir /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastqc /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastq/tutorial_R1_processed.fastq
( (fastp --overrepresentation_analysis --thread 1 --in1 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastq/tutorial_R2.fastq --adapter_sequence GATCGTCGGACTGTAGAACTCTGAAC --length_required 18 --html /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastp/tutorial_R2_fastp_adapter.html --json /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastp/tutorial_R2_fastp_adapter.json --report_title 'tutorial' --stdout ) 2> /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastp/tutorial_R1_fastp_adapter.txt | seqtk trimfq -L 30 - | seqtk seq -r - > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastq/tutorial_R2_trimmed.fastq ) 2> /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastqc/tutorial_R1_rmAdapter.txt
cp /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastq/tutorial_R2_trimmed.fastq /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastq/tutorial_R2_trimmed_dups.fastq
fastq_pair -t 1800000 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastq/tutorial_R1_processed.fastq /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastq/tutorial_R2_trimmed.fastq
mv /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastq/tutorial_R1_processed.fastq.paired.fq /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastq/tutorial_R1_processed.fastq
mv /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastq/tutorial_R2_trimmed.fastq.paired.fq /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastq/tutorial_R2_trimmed.fastq
touch repaired.flag
mkfifo /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/prealignments/human_rDNA_bt2
perl /scratch/jps3dp/tools/databio/peppro/tools/filter_paired_fq.pl /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/prealignments/human_rDNA_bt2 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastq/tutorial_R1_processed.fastq /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastq/tutorial_R2_trimmed.fastq /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_unmap_R1.fq /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_unmap_R2.fq
(bowtie2 -p 1 -k 1 -D 20 -R 3 -N 1 -L 20 -i S,1,0.50 -x /scratch/jps3dp/DATA/genomes/human_rDNA/bowtie2_index/default/human_rDNA --rg-id tutorial -U /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/fastq/tutorial_R1_processed.fastq --un /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/prealignments/human_rDNA_bt2 > /dev/null) 2>/project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_bt_aln_summary.log
grep 'aligned exactly 1 time' /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_bt_aln_summary.log | awk '{print $1}'
bowtie2 -p 1 --very-sensitive -X 2000 --rg-id tutorial -x /scratch/jps3dp/DATA/genomes/hg38/bowtie2_index/default/hg38 --rf -1 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_unmap_R1.fq -2 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_unmap_R2.fq | samtools view -bS - -@ 1  | samtools sort - -@ 1 -T /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tmp7b3a497a -o /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_temp.bam
samtools view -q 10 -b -@ 1 -U /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_fail_qc.bam /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_temp.bam > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort.bam
samtools depth -b /scratch/jps3dp/DATA/genomes/hg38/refgene_anno/default/hg38_pre-mRNA.bed /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort.bam | awk '{counter++;sum+=$3}END{print sum/counter}'
gzip -f /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_unmap_R1.fq
gzip -f /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/prealignments/tutorial_human_rDNA_unmap_R2.fq
samtools index /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_temp.bam
samtools idxstats /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_temp.bam | grep -we 'chrM' -we 'chrMT' -we 'M' -we 'MT' -we 'rCRSd' -we 'rCRSd_3k'| cut -f 3
samtools index /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort.bam
samtools idxstats /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort.bam | cut -f 1 | grep -vwe 'chrM' -vwe 'chrMT' -vwe 'M' -vwe 'MT' -vwe 'rCRSd' -vwe 'rCRSd_3k'| xargs samtools view -b -@ 1 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort.bam > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_noMT.bam
mv /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_noMT.bam /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort.bam
samtools index /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort.bam
samtools view -b -f 64 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort.bam | samtools sort - -@ 1 > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam
samtools view -b -f 128 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_sort.bam | samtools sort - -@ 1 > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE2.bam
samtools index /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam
/scratch/jps3dp/tools/databio/peppro/tools/bamQC.py --silent -i /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam -c 1 -o /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_bamQC.tsv
awk '{ for (i=1; i<=NF; ++i) { if ($i ~ "NRF") c=i } getline; print $c }' /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_bamQC.tsv
awk '{ for (i=1; i<=NF; ++i) { if ($i ~ "PBC1") c=i } getline; print $c }' /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_bamQC.tsv
awk '{ for (i=1; i<=NF; ++i) { if ($i ~ "PBC2") c=i } getline; print $c }' /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_bamQC.tsv
samtools view -b -@ 1 -f 12  /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_temp.bam > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_unmap.bam
samtools view -c -f 4 -@ 1 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_temp.bam
samtools view -bh -F 20 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam
samtools view -bh -f 16 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam
sed -n -e '/[[:space:]]+/w /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/plus_TSS.tsv' -e '/[[:space:]]-/w /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/minus_TSS.tsv' /scratch/jps3dp/DATA/genomes/hg38/refgene_anno/default/hg38_TSS.bed
/scratch/jps3dp/tools/databio/peppro/tools/pyTssEnrichment.py -a /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam -b /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/plus_TSS.tsv -p ends -c 1 -z -v -s 6 -o /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_plus_TssEnrichment.txt
/scratch/jps3dp/tools/databio/peppro/tools/pyTssEnrichment.py -a /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam -b /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/minus_TSS.tsv -p ends -c 1 -z -v -s 6 -o /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_minus_TssEnrichment.txt
Rscript /scratch/jps3dp/tools/databio/peppro/tools/PEPPRO.R tss -i /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_plus_TssEnrichment.txt /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_minus_TssEnrichment.txt
samtools view -H /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam | grep 'SN:' | awk -F':' '{print $2,$3}' | awk -F' ' -v OFS='	' '{print $1,$3}' > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt
cut -f 1 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_keep.txt
grep -wf /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_keep.txt /scratch/jps3dp/DATA/genomes/hg38/ensembl_gtf/default/hg38_ensembl_TSS.bed | bedtools sort -i stdin -faidx /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/hg38_ensembl_tss.bed
grep -wf /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_keep.txt /scratch/jps3dp/DATA/genomes/hg38/ensembl_gtf/default/hg38_ensembl_gene_body.bed | bedtools sort -i stdin -faidx /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/hg38_ensembl_gene_body.bed
bedtools coverage -sorted -counts -s -a /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/hg38_ensembl_tss.bed -b /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam -g /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt | awk '$7>0' | sort -k4,4 -k7,7nr | sort -k4,4 -u > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_TSS_density.bed
bedtools coverage -sorted -counts -s -a /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/hg38_ensembl_gene_body.bed -b /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam -g /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt | awk '$7>0' | sort -k4 > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_gene_body_density.bed
join --nocheck-order -j4 -o 1.1 1.2 1.3 1.4 1.6 1.7 2.2 2.3 2.7 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_TSS_density.bed /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_gene_body_density.bed | awk -v OFS='	' '{print $1, $2, $3, $4, ($6/($3-$2))/($9/($8-$7)), $5}' | env LC_COLLATE=C sort -k1,1 -k2,2n > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_pause_index.bed
sort -k5,5n /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_pause_index.bed | awk ' { a[i++]=$5; } END { x=int((i+1)/2); if (x < (i+1)/2) print (a[x-1]+a[x])/2; else print a[x-1]; }'
Rscript /scratch/jps3dp/tools/databio/peppro/tools/PEPPRO.R pi -i /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_pause_index.bed
gzip -f -f /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_pause_index.bed
samtools view -@ 4 -c -L /scratch/jps3dp/DATA/genomes/hg38/refgene_anno/default/hg38_pre-mRNA.bed /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam
samtools view -@ 4 -c -L /scratch/jps3dp/DATA/genomes/hg38/refgene_anno/default/hg38_pre-mRNA.bed /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam
perl /scratch/jps3dp/tools/databio/peppro/tools/fragment_length_dist.pl /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_fragLen.txt
sort -n  /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_fragLen.txt | uniq -c  > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_fragCount.txt
Rscript /scratch/jps3dp/tools/databio/peppro/tools/PEPPRO.R frag -l /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_fragLen.txt -c /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_fragCount.txt -p /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_fragLenDistribution.pdf -t /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_fragLenDistribution.txt
ln -sf /scratch/jps3dp/DATA/genomes/hg38/feat_annotation/default/hg38_annotations.bed.gz /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/raw/hg38_annotations.bed.gz
gzip -f -d -c /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/raw/hg38_annotations.bed.gz > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/raw/hg38_annotations.bed
cut -f 4 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/raw/hg38_annotations.bed | sort -u
awk -F'	' '{print>"/project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/"$4}' /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/raw/hg38_annotations.bed
mv "/project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/3' UTR" "/project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/3_UTR"
cut -f 1 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt | grep -wf - /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/3_UTR | cut -f 1-3 | bedtools sort -i stdin -faidx /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/3_UTR_sort.bed
bedtools coverage -sorted -counts -a /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/3_UTR_sort.bed -b /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam -g /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_3_UTR_plus_coverage.bed
bedtools coverage -sorted -counts -a /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/3_UTR_sort.bed -b /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam -g /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_3_UTR_minus_coverage.bed
mv "/project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/5' UTR" "/project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/5_UTR"
cut -f 1 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt | grep -wf - /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/5_UTR | cut -f 1-3 | bedtools sort -i stdin -faidx /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/5_UTR_sort.bed
bedtools coverage -sorted -counts -a /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/5_UTR_sort.bed -b /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam -g /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_5_UTR_plus_coverage.bed
bedtools coverage -sorted -counts -a /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/5_UTR_sort.bed -b /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam -g /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_5_UTR_minus_coverage.bed
cut -f 1 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt | grep -wf - /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/Enhancer | cut -f 1-3 | bedtools sort -i stdin -faidx /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/Enhancer_sort.bed
bedtools coverage -sorted -counts -a /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/Enhancer_sort.bed -b /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam -g /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_Enhancer_plus_coverage.bed
bedtools coverage -sorted -counts -a /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/Enhancer_sort.bed -b /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam -g /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_Enhancer_minus_coverage.bed
cut -f 1 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt | grep -wf - /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/Exon | cut -f 1-3 | bedtools sort -i stdin -faidx /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/Exon_sort.bed
bedtools coverage -sorted -counts -a /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/Exon_sort.bed -b /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam -g /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_Exon_plus_coverage.bed
bedtools coverage -sorted -counts -a /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/Exon_sort.bed -b /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam -g /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_Exon_minus_coverage.bed
cut -f 1 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt | grep -wf - /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/Intron | cut -f 1-3 | bedtools sort -i stdin -faidx /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/Intron_sort.bed
bedtools coverage -sorted -counts -a /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/Intron_sort.bed -b /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam -g /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_Intron_plus_coverage.bed
bedtools coverage -sorted -counts -a /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/Intron_sort.bed -b /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam -g /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_Intron_minus_coverage.bed
cut -f 1 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt | grep -wf - /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/Promoter | cut -f 1-3 | bedtools sort -i stdin -faidx /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/Promoter_sort.bed
bedtools coverage -sorted -counts -a /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/Promoter_sort.bed -b /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam -g /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_Promoter_plus_coverage.bed
bedtools coverage -sorted -counts -a /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/Promoter_sort.bed -b /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam -g /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_Promoter_minus_coverage.bed
mv "/project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/Promoter Flanking Region" "/project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/Promoter_Flanking_Region"
cut -f 1 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt | grep -wf - /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/Promoter_Flanking_Region | cut -f 1-3 | bedtools sort -i stdin -faidx /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/Promoter_Flanking_Region_sort.bed
bedtools coverage -sorted -counts -a /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/Promoter_Flanking_Region_sort.bed -b /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam -g /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_Promoter_Flanking_Region_plus_coverage.bed
bedtools coverage -sorted -counts -a /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/Promoter_Flanking_Region_sort.bed -b /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam -g /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_Promoter_Flanking_Region_minus_coverage.bed
samtools view -@ 1 -q 10 -c -F4 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam
Rscript /scratch/jps3dp/tools/databio/peppro/tools/PEPPRO.R frif -n tutorial -r 16265 -o /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_plus_frif.pdf --bed /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_3_UTR_plus_coverage.bed /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_5_UTR_plus_coverage.bed /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_Enhancer_plus_coverage.bed /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_Exon_plus_coverage.bed /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_Intron_plus_coverage.bed /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_Promoter_plus_coverage.bed /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_Promoter_Flanking_Region_plus_coverage.bed
samtools view -@ 1 -q 10 -c -F4 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam
Rscript /scratch/jps3dp/tools/databio/peppro/tools/PEPPRO.R frif -n tutorial -r 15685 -o /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_minus_frif.pdf --bed /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_3_UTR_minus_coverage.bed /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_5_UTR_minus_coverage.bed /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_Enhancer_minus_coverage.bed /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_Exon_minus_coverage.bed /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_Intron_minus_coverage.bed /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_Promoter_minus_coverage.bed /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_Promoter_Flanking_Region_minus_coverage.bed
grep -wf /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_keep.txt /scratch/jps3dp/DATA/genomes/hg38/refgene_anno/default/hg38_exons.bed | bedtools sort -i stdin -faidx /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/hg38_exons_sort.bed
grep -wf /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_keep.txt /scratch/jps3dp/DATA/genomes/hg38/refgene_anno/default/hg38_introns.bed | bedtools sort -i stdin -faidx /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt | bedtools sort -i stdin -faidx /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/hg38_introns_sort.bed
bedtools coverage -sorted -counts -s -a /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/hg38_exons_sort.bed -b /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam -g /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_exons_coverage.bed
bedtools coverage -sorted -counts -s -a /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/hg38_introns_sort.bed -b /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_PE1.bam -g /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/chr_order.txt > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_introns_coverage.bed
awk -v OFS='	' '{chrom[$4] = $1; if($4!=prev4) {chromStart[$4] = $2} strand[$4] = $6; readCount[$4] += $7; exonCount[$4] += 1; geneSizeKB[$4] += (sqrt(($3-$2+0.00000001)^2)/1000); gene[$4] = $4; chromEnd[$4]=$3; prev4=$4} END { for (a in readCount) { print chrom[a], chromStart[a], chromEnd[a], gene[a], (readCount[a]/0.156906)/geneSizeKB[a], strand[a]}}' /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_exons_coverage.bed | awk '$5>0' | sort -k4 > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_exons_rpkm.bed
awk -v OFS='	' '{chrom[$4] = $1; if($4!=prev4) {chromStart[$4] = $2} strand[$4] = $6; readCount[$4] += $7; exonCount[$4] += 1; geneSizeKB[$4] += (sqrt(($3-$2+0.00000001)^2)/1000); gene[$4] = $4; chromEnd[$4]=$3; prev4=$4} END { for (a in readCount) { print chrom[a], chromStart[a], chromEnd[a], gene[a], (readCount[a]/0.156906)/geneSizeKB[a], strand[a]}}' /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_introns_coverage.bed | awk '$5>0' | sort -k4 > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_introns_rpkm.bed
join --nocheck-order -a1 -a2 -j4 /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_introns_rpkm.bed /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_exons_rpkm.bed | awk -v OFS='	' 'NF==11 {print $7, $8, $9, $1, ($10/$5), $11}' | sort -k1,1 -k2,2n > /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_exon_intron_ratios.bed
awk '{print $5}' /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_exon_intron_ratios.bed | sort -n | awk ' { a[i++]=$1; } END { x=int((i+1)/2); if (x < (i+1)/2) print (a[x-1]+a[x])/2; else print a[x-1]; }'
Rscript /scratch/jps3dp/tools/databio/peppro/tools/PEPPRO.R mrna -i /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_exon_intron_ratios.bed --raw
gzip -f -f /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/QC_hg38/tutorial_exon_intron_ratios.bed
samtools index /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam
/scratch/jps3dp/tools/databio/peppro/tools/bamSitesToWig.py -i /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_plus.bam -c /scratch/jps3dp/DATA/genomes/hg38/fasta/default/hg38.chrom.sizes -o /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/signal_hg38/tutorial_plus_body_0-mer.bw -p 1 --variable-step --tail-edge
samtools index /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam
/scratch/jps3dp/tools/databio/peppro/tools/bamSitesToWig.py -i /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/aligned_hg38/tutorial_minus.bam -c /scratch/jps3dp/DATA/genomes/hg38/fasta/default/hg38.chrom.sizes -o /project/shefflab/processed/proseq/peppro_tutorial/pe/11-27-19/results_pipeline/tutorial/signal_hg38/tutorial_minus_body_0-mer.bw -p 1 --variable-step --tail-edge
