#! /usr/bin/env Rscript
###############################################################################
# PEPPRO R parser

###############################################################################
version <- 0.5
##### Load dependencies #####

required_libraries <- c("PEPPROr")
for (i in required_libraries) {
    loadLibrary <- tryCatch (
        {
            suppressPackageStartupMessages(
                suppressWarnings(library(i, character.only=TRUE)))
        },
        error=function(e) {
            message("Error: Install the \"", i,
                    "\" R package before proceeding.")
            message('i.e. devtools::install_github("databio/peppro",',
                    ' subdir="PEPPROr")')
            return(NULL)
        },
        warning=function(e) {
            message(e)
            return(1)
        }
    )
    if (length(loadLibrary)!=0) {
        suppressWarnings(library(i, character.only=TRUE))
    } else {
        quit()
    }
}

###############################################################################

#### Identify command ####

subcmd <- opt_get_verb()

if (is.na(subcmd) || grepl("/R", subcmd)) {
    usage <- paste0(
        "\n",
        "Usage:   PEPPRO.R [command] {args}\n",
        "Version: ", version, "\n\n",
        "Command: preseq\t\t plot preseq complexity curves\n",
        "\t frif\t\t plot fraction of reads in features\n",
        "\t tss\t\t plot TSS enrichment\n",
        "\t frag\t\t plot fragment length distribution\n",
        "\t mrna\t\t plot mRNA contamination distribution\n",
        "\t pi\t\t plot pause indicies distribution\n",
        "\t cutadapt\t plot adapter insertion distribution\n"
    )
    message(usage)
} else if (!is.na(subcmd) && tolower(subcmd) == "preseq") {
    usage <- paste0(
        "\n",
        "Usage:   PEPPRO.R [command] {args}\n",
        "Version: ", version, "\n\n",
        "Command: preseq \t plot preseq complexity curves\n\n",
        " -i, --input\t\t Input files generated by preseq.\n",
        " -c, --coverage\t\t Use coverage on axes instead of read counts. ",
                               "Enter number of base pairs of reference.\n",
        " -l, --read_length\t Sequence read length, for use in coverage ",
                               "calculations.\n",
        " -r, --real_counts\t File name for file with three columns - ",
                               "preseq filename, total number reads, number",
                               " of unique reads (unique optional, ",
                               "whitespace delimited)\n",
        " -u, --ignore_unique\t Ignore any information about unique read ",
                               "counts found in --real_counts file.\n",
        " -o, --output_name\t Output name (.png/.pdf will be automatically ",
                               "added). Default: 'complexity_curves'.\n",
        " -m, --x_min\t\t Lower x-limit (default 0).\n",
        " -x, --x_max\t\t Upper x-limit (default 500 million).\n"
    )

    help <- opt_get(name = c("help", "?", "h"), required=FALSE,
                    default=FALSE, n=0)
    if (!help) {
        help <- suppressWarnings(
            if(length(opt_get_args()) == 1) {TRUE} else {FALSE}
        )
    }
    if (help) {
        message(usage)
        quit()
    } else {
        # get arguments
        a       <- opt_get_args()

        # Determine the number of input files
        inArgs  <- 0
        argList <- c("-c", "-l", "-r", "-u", "-o", "-m", "-x",
                     "--coverage", "--read_length", "--real_counts",
                     "--ignore_unique", "--output_name", "--x_min", "--x_max")
        p       <- 1
        val     <- a[p]
        while (!(val %in% c("-i", "--input"))) {
            p   <- p + 1
            val <- a[p]
        }
        while (!(val %in% argList) && p < length(opt_get_args())) {
            p   <- p + 1
            val <- a[p]
            if (!(val %in% argList)) {
                inArgs <- inArgs + 1
            }
        }

        # Determine the number of real counts files
        cArgs   <- 0
        argList <- c("-i", "-c", "-l", "-u", "-o", "-m", "-x",
                     "--input", "--coverage", "--read_length",
                     "--ignore_unique", "--output_name", "--x_min", "--x_max")
        p       <- 1
        val     <- a[p]
        while (!(val %in% c("-r", "--real_counts"))) {
            p   <- p + 1
            val <- a[p]
        }
        while (!(val %in% argList) && p < length(opt_get_args())) {
            p   <- p + 1
            val <- a[p]
            if (!(val %in% argList)) {
                cArgs <- cArgs + 1
            }
        }
        input       <- opt_get(name = c("input", "i"), required=TRUE,
                               n=inArgs,
                               description="Input files generated by preseq.")
        cov_help    <- paste0("Use coverage on axes instead of read ",
                              "counts. Enter number of base pairs of reference.")
        coverage    <- opt_get(name = c("coverage", "c"), required=FALSE,
                               default=0, description=cov_help)
        rl_help     <- paste0("Sequence read length, for use in coverage",
                              " calculations")
        read_length <- opt_get(name = c("read_length", "l"), required=FALSE,
                               default=0, description=rl_help)
        rc_help     <- paste0("File name for file with three columns - ",
                              "preseq filename, total number reads, number of ",
                              "unique reads (unique optional, whitespace ",
                              "delimited)")
        real_counts <- opt_get(name = c("real_counts", "r"), required=FALSE,
                               n=cArgs, description=rc_help)
        uni_help    <- paste0("Ignore any information about unique read ",
                              "counts found in --real_counts file.")
        ign_unique  <- opt_get(name = c("ignore_unique", "u"), required=FALSE,
                               default=FALSE, description=uni_help)
        output_help <- "Output name (.png/.pdf will be automatically added)"
        output_name <- opt_get(name=c("o", "output_name"), required=FALSE,
                               default='complexity_curves',
                               description=output_help)
        x_min       <- opt_get(name=c("m", "x_min"),
                               required=FALSE, default=0,
                               description="Lower x-limit (default 0)")
        x_max       <- opt_get(name=c("x", "x_max"),
                               required=FALSE, default=500000000,
                               description="Upper x-limit (default 500 million)")

        p <- plotComplexityCurves(ccurves = input,
                                  coverage = coverage,
                                  read_length = read_length,
                                  real_counts_path = real_counts,
                                  ignore_unique = ign_unique,
                                  x_min = x_min,
                                  x_max = x_max)
        # now save the plot
        pdf(file = paste0(tools::file_path_sans_ext(output_name), ".pdf"),
            width= 10, height = 7, useDingbats=F)
        print(fig)
        invisible(dev.off())
        png(filename = paste0(tools::file_path_sans_ext(output_name), ".png"),
            width = 686, height = 480)
        print(fig)
        invisible(dev.off())

        if (exists("p")) {
            write("Library complexity plot completed!\n", stdout())
        } else {
            write("Unable to produce library complexity plot!\n", stdout())
        }
    }
} else if (!is.na(subcmd) && tolower(subcmd) == "frif") {
    usage <- paste0(
        "\n",
        "Usage:   PEPPRO.R [command] {args}\n",
        "Version: ", version, "\n\n",
        "Command: frif \t plot fraction of reads in features\n\n",
        " -n, --sample_name\t   Sample name.\n",
        " -r, --reads\t\t   Number of mapped reads.\n",
        " -o, --output_name\t   Output file name.\n",
        " -b, --bed\t\t   Coverage file(s).\n"
    )

    help <- opt_get(name = c("help", "?", "h"), required=FALSE,
                    default=FALSE, n=0)
    if (!help) {
        help <- suppressWarnings(
            if(length(opt_get_args()) == 1) {TRUE} else {FALSE}
        )
    }
    if (help) {
        message(usage)
        quit()
    } else {
        sample_name <- opt_get(name = c("sample_name", "n"), required=TRUE,
                               description="Sample name.")
        reads       <- opt_get(name = c("reads", "r"), required=TRUE,
                               description="Number of mapped reads.")
        output_name <- opt_get(name = c("output_name", "o"), required=TRUE,
                               description="Output file name.")
        numArgs     <- length(opt_get_args())
        bed         <- opt_get(name = c("bed", "b"), required=TRUE,
                               n=(numArgs - 8),
                               description="Coverage file(s).")

        p <- plotFRiF(sample_name = sample_name,
                      num_reads = reads,
                      output_name = output_name,
                      bedFile = bed)

        pdf(file = paste0(tools::file_path_sans_ext(output_name), ".pdf"),
        width= 7, height = 7, useDingbats=F)
        print(p)
        invisible(dev.off())
        png(filename = paste0(tools::file_path_sans_ext(output_name), ".png"),
            width = 480, height = 480)
        print(p)
        invisible(dev.off())

        if (exists("p")) {
            write("Cumulative FRiF plot completed!\n", stdout())
        } else {
            write("Unable to produce FRiF plot!\n", stdout())
        }
    }
} else if (!is.na(subcmd) && tolower(subcmd) == "tss") {
    usage <- paste0(
        "\n",
        "Usage:   PEPPRO.R [command] {args}\n",
        "Version: ", version, "\n\n",
        "Command: tss \t plot TSS enrichment\n\n",
        " -i, --input\t TSS enrichment file.\n"
    )

    help <- opt_get(name = c("help", "?", "h"), required=FALSE,
                    default=FALSE, n=0)
    if (!help) {
        help <- suppressWarnings(
            if(length(opt_get_args()) == 1) {TRUE} else {FALSE}
        )
    }
    if (help) {
        message(usage)
        quit()
    } else {
        TSS_CUTOFF <- 6
        # get arguments
        a   <- opt_get_args()
        # Determine the number of input files
        inArgs <- 0
        p      <- 1
        val    <- a[p]
        while (!(val %in% c("-i", "--input"))) {
            p   <- p + 1
            val <- a[p]
        }
        while (p < length(opt_get_args())) {
            p   <- p + 1
            val <- a[p]
            if (!(val %in% c("-i", "--input"))) {
                inArgs <- inArgs + 1
            }
        }

        TSSfile     <- opt_get(name = c("input", "i"), required=TRUE, n=inArgs,
                               description="TSS enrichment file.")

        p           <- plotTSS(TSSfile = TSSfile)

        sample_name <- sampleName(TSS_file[1])

        png(filename = paste0(sample_name, "_TSSenrichment.png"),
        width = 480, height = 480)
        print(p)
        invisible(dev.off())

        pdf(file = paste0(sample_name, "_TSSenrichment.pdf"),
            width= 7, height = 7, useDingbats=F)
        print(p)
        invisible(dev.off())

        if (exists("p")) {
            write("TSS enrichment plot completed!\n", stdout())
        } else {
            write("Unable to produce TSS enrichment plot!\n", stdout())
        }
    }
} else if (!is.na(subcmd) && tolower(subcmd) == "frag") {
    usage <- paste0(
        "\n",
        "Usage:   PEPPRO.R [command] {args}\n",
        "Version: ", version, "\n\n",
        "Command: frag \t plot fragment length distribution\n\n",
        " -l, --frag_len\t\t Column of fragment lengths.\n",
        " -c, --frag_count\t Counts of each fragment length.\n",
        " -p, --pdf\t\t PDF output file name.\n",
        " -t, --text\t\t Fragment length distribution stats file.\n"
    )

    help <- opt_get(name = c("help", "?", "h"), required=FALSE,
                    default=FALSE, n=0)
    if (!help) {
        help <- suppressWarnings(
            if(length(opt_get_args()) == 1) {TRUE} else {FALSE}
        )
    }
    if (help) {
        message(usage)
        quit()
    } else {
        fragL       <- opt_get(name = c("frag_len", "l"), required=TRUE,
                               description="Column of fragment lengths.")
        fragL_count <- opt_get(name = c("frag_count", "c"), required=TRUE,
                               description="Counts of each fragment length.")
        fragL_name  <- opt_get(name = c("pdf", "p"), required=TRUE,
                               description="PDF output file name.")
        fragL_txt   <- opt_get(name = c("text", "t"), required=TRUE,
                               description="Fragment length distribution stats file.")

        p <- plotFLD(fragL = fragL,
                     fragL_count = fragL_count,
                     fragL_dis2 = fragL_txt)

        # Save plot to pdf file
        pdf(file=fragL_name, width= 7, height = 7, useDingbats=F)
        print(p)
        invisible(dev.off())

        # Save plot to png file
        outfile_png <- gsub('pdf', 'png', fragL_name)
        png(filename=outfile_png, width = 480, height = 480)
        print(p)
        invisible(dev.off())

        if (exists("p")) {
            write("Fragment distribution plot completed!\n", stdout())
        } else {
            write("Unable to produce fragment distribution plot!\n", stdout())
        }
    }
} else if (!is.na(subcmd) && tolower(subcmd) == "mrna") {
    usage <- paste0(
        "\n",
        "Usage:   PEPPRO.R [command] {args}\n",
        "Version: ", version, "\n\n",
        "Command: mrna \t plot mRNA contamination distribution\n\n",
        " -i, --rpkm\t Three column TSV of intron and exon RPKM by gene.\n",
        " -w, --raw\t Plot raw exon/intron ratios instead of log10.\n"
    )

    help <- opt_get(name = c("help", "?", "h"), required=FALSE,
                    default=FALSE, n=0)
    if (!help) {
        help <- suppressWarnings(
            if(length(opt_get_args()) == 1) {TRUE} else {FALSE}
        )
    }
    if (help) {
        message(usage)
        quit()
    } else {
        rpkm <- opt_get(name = c("rpkm", "i"), required=TRUE,
                        description="Three column TSV containing gene exon and intron RPKMs.")
        raw  <- opt_get(name = c("raw", "w"), required=FALSE, default=FALSE,
                        description="Plot raw ratios (Default = FALSE).")

        suppressWarnings(p <- mRNAcontamination(rpkm=rpkm, raw=raw))

        sample_name <- sampleName(rpkm)

        # Save plot to pdf file
        pdf(file=paste0(sample_name, "_mRNA_contamination.pdf"),
            width= 7, height = 7, useDingbats=F)
        print(q)
        invisible(dev.off())
             
        # Save plot to png file
        png(filename = paste0(sample_name, "_mRNA_contamination.png"),
            width = 480, height = 480)
        print(q)
        invisible(dev.off())

        if (exists("p")) {
            write("mRNA contamination plot completed!\n", stdout())
        } else {
            write("Unable to produce mRNA contamination plot!\n", stdout())
        }
    }
} else if (!is.na(subcmd) && tolower(subcmd) == "pi") {
    usage <- paste0(
        "\n",
        "Usage:   PEPPRO.R [command] {args}\n",
        "Version: ", version, "\n\n",
        "Command: pi \t plot pause indicies distribution\n\n",
        " -i, --input\t Pause density/gene body density ratios.\n"
    )

    help <- opt_get(name = c("help", "?", "h"), required=FALSE,
                    default=FALSE, n=0)
    if (!help) {
        help <- suppressWarnings(
            if(length(opt_get_args()) == 1) {TRUE} else {FALSE}
        )
    }
    if (help) {
        message(usage)
        quit()
    } else {
        input <- opt_get(name = c("input", "i"), required=TRUE,
                         description="Pause density/gene body density ratios.")

        suppressWarnings(p <- plotPI(pi=input))

        sample_name <- sampleName(input)

        # Save plot to pdf file
        pdf(file=paste0(sample_name, "_pause_index.pdf"),
            width= 7, height = 7, useDingbats=F)
        print(p)
        invisible(dev.off())
             
        # Save plot to png file
        png(filename = paste0(sample_name, "_pause_index.png"),
            width = 480, height = 480)
        print(p)
        invisible(dev.off())

        if (exists("p")) {
            write("Pause index plot completed!\n", stdout())
        } else {
            write("Unable to produce pause index plot!\n", stdout())
        }
    }
} else if (!is.na(subcmd) && tolower(subcmd) == "cutadapt") {
    usage <- paste0(
        "\n",
        "Usage:   PEPPRO.R [command] {args}\n",
        "Version: ", version, "\n\n",
        "Command: cutadapt \t plot adapter insertion distribution\n\n",
        " -i, --input\t cutadapt report.\n",
        " -o, --output\t output directory.\n"
    )

    help <- opt_get(name = c("help", "?", "h"), required=FALSE,
                    default=FALSE, n=0)
    if (!help) {
        help <- suppressWarnings(
            if(length(opt_get_args()) == 1) {TRUE} else {FALSE}
        )
    }
    if (help) {
        message(usage)
        quit()
    } else {
        input  <- opt_get(name = c("input", "i"), required=TRUE,
                          description="cutadapt report.")
        output <- opt_get(name = c("output", "o"), required=TRUE,
                          description="output destination directory.")

        suppressWarnings(p <- plotCutadapt(input=input))

        name        <- basename(tools::file_path_sans_ext(input))
        sample_name <- paste(output, name, sep="/")

        # Save plot to pdf file
        pdf(file=paste0(sample_name, "_adapter_insertion_distribution.pdf"),
            width= 7, height = 7, useDingbats=F)
        print(p)
        invisible(dev.off())
             
        # Save plot to png file
        png(filename = paste0(sample_name,
                              "_adapter_insertion_distribution.png"),
            width = 480, height = 480)
        print(p)
        invisible(dev.off())

        if (exists("p")) {
            write("Adapter insertion distribution plot completed!\n", stdout())
        } else {
            write("Unable to produce adapter insertion distribution plot!\n",
                  stdout())
        }
    }
} else {
    usage <- paste0(
        "\n",
        "Usage:   PEPPRO.R [command] {args}\n",
        "Version: ", version, "\n\n",
        "Command: preseq\t\t plot preseq complexity curves\n",
        "\t frif\t\t plot fraction of reads in features\n",
        "\t tss\t\t plot TSS enrichment\n",
        "\t frag\t\t plot fragment length distribution\n",
        "\t mrna\t\t plot mRNA contamination distribution\n",
        "\t pi\t\t plot pause indicies distribution\n",
        "\t cutadapt\t plot adapter insertion distribution\n"
    )
    message(usage)
    quit()
}
