# Browse the output directory

This is an interactive display of exactly what results you'll get as output from PEPPRO:

* ??? danger ":fa-folder-open-o: PEPPRO_reports/"
    * :fa-file-code-o: [adapter_insertion_distribution.html](files/examples/paper/PEPPRO_reports/adapter_insertion_distribution.html)
    * :fa-file-code-o: [cfrif.html](files/examples/paper/PEPPRO_reports/cfrif.html)
    * :fa-file-code-o: [fastqc_report_r1.html](files/examples/paper/PEPPRO_reports/fastqc_report_r1.html)
    * :fa-file-code-o: [fastqc_report_r2.html](files/examples/paper/PEPPRO_reports/fastqc_report_r2.html)
    * :fa-file-code-o: [frif.html](files/examples/paper/PEPPRO_reports/frif.html)
    * :fa-file-code-o: [h9_pro-seq_1.html](files/examples/paper/PEPPRO_reports/h9_pro-seq_1.html)
    * :fa-file-code-o: [h9_pro-seq_2.html](files/examples/paper/PEPPRO_reports/h9_pro-seq_2.html)
    * :fa-file-code-o: [h9_pro-seq_3.html](files/examples/paper/PEPPRO_reports/h9_pro-seq_3.html)
    * :fa-file-code-o: [h9_treated_pro-seq_1.html](files/examples/paper/PEPPRO_reports/h9_treated_pro-seq_1.html)
    * :fa-file-code-o: [h9_treated_pro-seq_2.html](files/examples/paper/PEPPRO_reports/h9_treated_pro-seq_2.html)
    * :fa-file-code-o: [h9_treated_pro-seq_3.html](files/examples/paper/PEPPRO_reports/h9_treated_pro-seq_3.html)
    * :fa-file-code-o: [h9_pro-seq_10.html](files/examples/paper/PEPPRO_reports/h9_pro-seq_10.html)
    * :fa-file-code-o: [h9_pro-seq_20.html](files/examples/paper/PEPPRO_reports/h9_pro-seq_20.html)
    * :fa-file-code-o: [h9_pro-seq_30.html](files/examples/paper/PEPPRO_reports/h9_pro-seq_30.html)
    * :fa-file-code-o: [h9_pro-seq_40.html](files/examples/paper/PEPPRO_reports/h9_pro-seq_40.html)
    * :fa-file-code-o: [h9_pro-seq_50.html](files/examples/paper/PEPPRO_reports/h9_pro-seq_50.html)
    * :fa-file-code-o: [h9_pro-seq_60.html](files/examples/paper/PEPPRO_reports/h9_pro-seq_60.html)
    * :fa-file-code-o: [h9_pro-seq_70.html](files/examples/paper/PEPPRO_reports/h9_pro-seq_70.html)
    * :fa-file-code-o: [h9_pro-seq_80.html](files/examples/paper/PEPPRO_reports/h9_pro-seq_80.html)
    * :fa-file-code-o: [h9_pro-seq_90.html](files/examples/paper/PEPPRO_reports/h9_pro-seq_90.html)
    * :fa-file-code-o: [h9_pro-seq_100.html](files/examples/paper/PEPPRO_reports/h9_pro-seq_100.html)
    * :fa-file-code-o: [hek_arf_pro-seq.html](files/examples/paper/PEPPRO_reports/hek_arf_pro-seq.html)
    * :fa-file-code-o: [hek_pro-seq.html](files/examples/paper/PEPPRO_reports/hek_pro-seq.html)
    * :fa-file-code-o: [helas3_gro-seq.html](files/examples/paper/PEPPRO_reports/helas3_gro-seq.html)
    * :fa-file-code-o: [jurkat_chro-seq_1.html](files/examples/paper/PEPPRO_reports/jurkat_chro-seq_1.html)
    * :fa-file-code-o: [jurkat_chro-seq_2.html](files/examples/paper/PEPPRO_reports/jurkat_chro-seq_2.html)
    * :fa-file-code-o: [k562_gro-seq.html](files/examples/paper/PEPPRO_reports/k562_gro-seq.html)
    * :fa-file-code-o: [k562_pro-seq_02.html](files/examples/paper/PEPPRO_reports/k562_pro-seq_02.html)
    * :fa-file-code-o: [k562_pro-seq_04.html](files/examples/paper/PEPPRO_reports/k562_pro-seq_04.html)
    * :fa-file-code-o: [k562_pro-seq_06.html](files/examples/paper/PEPPRO_reports/k562_pro-seq_06.html)
    * :fa-file-code-o: [k562_pro-seq_08.html](files/examples/paper/PEPPRO_reports/k562_pro-seq_08.html)
    * :fa-file-code-o: [k562_pro-seq_10.html](files/examples/paper/PEPPRO_reports/k562_pro-seq_10.html)
    * :fa-file-code-o: [k562_pro-seq_20.html](files/examples/paper/PEPPRO_reports/k562_pro-seq_20.html)
    * :fa-file-code-o: [k562_pro-seq_30.html](files/examples/paper/PEPPRO_reports/k562_pro-seq_30.html)
    * :fa-file-code-o: [k562_pro-seq_40.html](files/examples/paper/PEPPRO_reports/k562_pro-seq_40.html)
    * :fa-file-code-o: [k562_pro-seq_50.html](files/examples/paper/PEPPRO_reports/k562_pro-seq_50.html)
    * :fa-file-code-o: [k562_pro-seq_60.html](files/examples/paper/PEPPRO_reports/k562_pro-seq_60.html)
    * :fa-file-code-o: [k562_pro-seq_70.html](files/examples/paper/PEPPRO_reports/k562_pro-seq_70.html)
    * :fa-file-code-o: [k562_pro-seq_80.html](files/examples/paper/PEPPRO_reports/k562_pro-seq_80.html)
    * :fa-file-code-o: [k562_pro-seq_90.html](files/examples/paper/PEPPRO_reports/k562_pro-seq_90.html)
    * :fa-file-code-o: [k562_pro-seq_100.html](files/examples/paper/PEPPRO_reports/k562_pro-seq_100.html)
    * :fa-file-code-o: [k562_rna-seq_0.html](files/examples/paper/PEPPRO_reports/k562_rna-seq_0.html)
    * :fa-file-code-o: [k562_rna-seq_10.html](files/examples/paper/PEPPRO_reports/k562_rna-seq_10.html)
    * :fa-file-code-o: [k562_rna-seq_20.html](files/examples/paper/PEPPRO_reports/k562_rna-seq_20.html)
    * :fa-file-code-o: [k562_rna-seq_30.html](files/examples/paper/PEPPRO_reports/k562_rna-seq_30.html)
    * :fa-file-code-o: [k562_rna-seq_40.html](files/examples/paper/PEPPRO_reports/k562_rna-seq_40.html)
    * :fa-file-code-o: [k562_rna-seq_50.html](files/examples/paper/PEPPRO_reports/k562_rna-seq_50.html)
    * :fa-file-code-o: [k562_rna-seq_60.html](files/examples/paper/PEPPRO_reports/k562_rna-seq_60.html)
    * :fa-file-code-o: [k562_rna-seq_70.html](files/examples/paper/PEPPRO_reports/k562_rna-seq_70.html)
    * :fa-file-code-o: [k562_rna-seq_80.html](files/examples/paper/PEPPRO_reports/k562_rna-seq_80.html)
    * :fa-file-code-o: [k562_rna-seq_90.html](files/examples/paper/PEPPRO_reports/k562_rna-seq_90.html)
    * :fa-file-code-o: [k562_rna-seq_100.html](files/examples/paper/PEPPRO_reports/k562_rna-seq_100.html)
    * :fa-file-code-o: [library_complexity.html](files/examples/paper/PEPPRO_reports/library_complexity.html)
    * :fa-file-code-o: [mrna_contamination.html](files/examples/paper/PEPPRO_reports/mrna_contamination.html)
    * :fa-file-code-o: [objects.html](files/examples/paper/PEPPRO_reports/objects.html)
    * :fa-file-code-o: [pause_index.html](files/examples/paper/PEPPRO_reports/pause_index.html)
    * :fa-file-code-o: [samples.html](files/examples/paper/PEPPRO_reports/samples.html)
    * :fa-file-code-o: [status.html](files/examples/paper/PEPPRO_reports/status.html)
    * :fa-file-code-o: [tss_enrichment.html](files/examples/paper/PEPPRO_reports/tss_enrichment.html)
* ??? danger ":fa-folder-open-o: results_pipeline/"
    * ??? danger ":fa-folder-open-o: H9_PRO-seq_1/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: H9_PRO-seq_1_fail_qc.bam
            * :fa-file-o: H9_PRO-seq_1_sort_dups.bam
            * :fa-file-o: H9_PRO-seq_1_sort.bam
            * :fa-file-o: H9_PRO-seq_1_PE1.bam
            * :fa-file-o: H9_PRO-seq_1_PE1.bam.bai
            * :fa-file-o: H9_PRO-seq_1_PE2.bam
            * :fa-file-o: H9_PRO-seq_1_dups_PE2.bam
            * :fa-file-o: H9_PRO-seq_1_unmap.bam
            * :fa-file-o: H9_PRO-seq_1_plus.bam 
            * :fa-file-o: H9_PRO-seq_1_plus.bam.bai
            * :fa-file-o: H9_PRO-seq_1_minus.bam 
            * :fa-file-o: H9_PRO-seq_1_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: H9_PRO-seq_1_adapter_insertion_distribution.pdf](files/examples/paper/results_pipeline/H9_PRO-seq_1/cutadapt/H9_PRO-seq_1_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: H9_PRO-seq_1_adapter_insertion_distribution.png](files/examples/paper/results_pipeline/H9_PRO-seq_1/cutadapt/H9_PRO-seq_1_adapter_insertion_distribution.png)
            * :fa-file-text-o: H9_PRO-seq_1_R1_cutadapt.txt
            * :fa-file-text-o: H9_PRO-seq_1_R2_cutadapt.txt
            * :fa-file-text-o: H9_PRO-seq_1.hist
            * :fa-file-text-o: H9_PRO-seq_1.histogram
            * :fa-file-text-o: H9_PRO-seq_1.extendedFrags.fastq.gz
            * :fa-file-text-o: H9_PRO-seq_1.notCombined_1.fastq.gz
            * :fa-file-text-o: H9_PRO-seq_1.notCombined_2.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * :fa-file-code-o: H9_PRO-seq_1_R1_processed_fastqc.zip
            * [:fa-file-code-o: H9_PRO-seq_1_R1_processed_fastqc.html](files/examples/paper/results_pipeline/H9_PRO-seq_1/fastqc/H9_PRO-seq_1_R1_processed_fastqc.html)
            * :fa-file-text-o: H9_PRO-seq_1_R2_trimmed_fastqc.zip
            * [:fa-file-code-o: H9_PRO-seq_1_R2_trimmed_fastqc.html](files/examples/paper/results_pipeline/H9_PRO-seq_1/fastqc/H9_PRO-seq_1_R2_trimmed_fastqc.html)
        * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-archive-o: H9_PRO-seq_1_human_rDNA_unmap_R1.fq.gz
            * :fa-file-archive-o: H9_PRO-seq_1_human_rDNA_unmap_R2.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: H9_PRO-seq_1_bamQC.tsv
            * :fa-file-o: H9_PRO-seq_1_exon_intron_ratios.bed.gz
            * [:fa-file-pdf-o: H9_PRO-seq_1_cFRiF.pdf](files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_cFRiF.pdf)
            * [:fa-file-image-o: H9_PRO-seq_1_cFRiF.png](files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_cFRiF.png)
            * [:fa-file-pdf-o: H9_PRO-seq_1_mRNA_contamination.pdf](files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_mRNA_contamination.pdf)
            * [:fa-file-image-o: H9_PRO-seq_1_mRNA_contamination.png](files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_mRNA_contamination.png)
            * :fa-file-o: H9_PRO-seq_1_pause_index.bed.gz
            * [:fa-file-pdf-o: H9_PRO-seq_1_pause_index.pdf](files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_pause_index.pdf)
            * [:fa-file-image-o: H9_PRO-seq_1_pause_index.png](files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_pause_index.png)          
            * :fa-file-text-o: H9_PRO-seq_1_preseq_counts.txt
            * :fa-file-text-o: H9_PRO-seq_1_preseq_out.txt
            * [:fa-file-pdf-o: H9_PRO-seq_1_preseq_plot.pdf](files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_preseq_plot.pdf)
            * [:fa-file-image-o: H9_PRO-seq_1_preseq_plot.png](files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_preseq_plot.png)
            * :fa-file-text-o: H9_PRO-seq_1_preseq_yield.txt
            * [:fa-file-pdf-o: H9_PRO-seq_1_FRiF.pdf](files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_FRiF.pdf)
            * [:fa-file-image-o: H9_PRO-seq_1_FRiF.png](files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_FRiF.png)
            * [:fa-file-pdf-o: H9_PRO-seq_1_TSSenrichment.pdf](files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_TSSenrichment.pdf)
            * [:fa-file-image-o: H9_PRO-seq_1_TSSenrichment.png](files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: H9_PRO-seq_1_r1.fastq.gz
            * :fa-file-archive-o: H9_PRO-seq_1_r2.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: H9_PRO-seq_1_gene_coverage.bed
            * :fa-file-o: H9_PRO-seq_1_minus_exact_body_0-mer.bw
            * :fa-file-o: H9_PRO-seq_1_plus_exact_body_0.bw
            * :fa-file-o: H9_PRO-seq_1_minus_smooth_body_0-mer.bw
            * :fa-file-o: H9_PRO-seq_1_plus_smooth_body_0.bw 
        * [:fa-file-text-o: objects.tsv](files/examples/paper/results_pipeline/H9_PRO-seq_1/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](files/examples/paper/results_pipeline/H9_PRO-seq_1/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](files/examples/paper/results_pipeline/H9_PRO-seq_1/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](files/examples/paper/results_pipeline/H9_PRO-seq_1/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](files/examples/paper/results_pipeline/H9_PRO-seq_1/stats.tsv)
    * ??? danger ":fa-folder-open-o: H9_PRO-seq_2/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: H9_PRO-seq_2_fail_qc.bam
            * :fa-file-o: H9_PRO-seq_2_sort_dups.bam
            * :fa-file-o: H9_PRO-seq_2_sort.bam
            * :fa-file-o: H9_PRO-seq_2_PE1.bam
            * :fa-file-o: H9_PRO-seq_2_PE1.bam.bai
            * :fa-file-o: H9_PRO-seq_2_PE2.bam
            * :fa-file-o: H9_PRO-seq_2_dups_PE2.bam
            * :fa-file-o: H9_PRO-seq_2_unmap.bam
            * :fa-file-o: H9_PRO-seq_2_plus.bam 
            * :fa-file-o: H9_PRO-seq_2_plus.bam.bai
            * :fa-file-o: H9_PRO-seq_2_minus.bam 
            * :fa-file-o: H9_PRO-seq_2_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: H9_PRO-seq_2_adapter_insertion_distribution.pdf](files/examples/paper/results_pipeline/H9_PRO-seq_2/cutadapt/H9_PRO-seq_2_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: H9_PRO-seq_2_adapter_insertion_distribution.png](files/examples/paper/results_pipeline/H9_PRO-seq_2/cutadapt/H9_PRO-seq_2_adapter_insertion_distribution.png)
            * :fa-file-text-o: H9_PRO-seq_2_R1_cutadapt.txt
            * :fa-file-text-o: H9_PRO-seq_2_R2_cutadapt.txt
            * :fa-file-text-o: H9_PRO-seq_2.hist
            * :fa-file-text-o: H9_PRO-seq_2.histogram
            * :fa-file-text-o: H9_PRO-seq_2.extendedFrags.fastq.gz
            * :fa-file-text-o: H9_PRO-seq_2.notCombined_1.fastq.gz
            * :fa-file-text-o: H9_PRO-seq_2.notCombined_2.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * :fa-file-code-o: H9_PRO-seq_2_R1_processed_fastqc.zip
            * [:fa-file-code-o: H9_PRO-seq_2_R1_processed_fastqc.html](files/examples/paper/results_pipeline/H9_PRO-seq_2/fastqc/H9_PRO-seq_2_R1_processed_fastqc.html)
            * :fa-file-text-o: H9_PRO-seq_2_R2_trimmed_fastqc.zip
            * [:fa-file-code-o: H9_PRO-seq_2_R2_trimmed_fastqc.html](files/examples/paper/results_pipeline/H9_PRO-seq_2/fastqc/H9_PRO-seq_2_R2_trimmed_fastqc.html)
        * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-archive-o: H9_PRO-seq_2_human_rDNA_unmap_R1.fq.gz
            * :fa-file-archive-o: H9_PRO-seq_2_human_rDNA_unmap_R2.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: H9_PRO-seq_2_bamQC.tsv
            * :fa-file-o: H9_PRO-seq_2_exon_intron_ratios.bed.gz
            * [:fa-file-pdf-o: H9_PRO-seq_2_cFRiF.pdf](files/examples/paper/results_pipeline/H9_PRO-seq_2/QC_hg38/H9_PRO-seq_2_cFRiF.pdf)
            * [:fa-file-image-o: H9_PRO-seq_2_cFRiF.png](files/examples/paper/results_pipeline/H9_PRO-seq_2/QC_hg38/H9_PRO-seq_2_cFRiF.png)
            * [:fa-file-pdf-o: H9_PRO-seq_2_mRNA_contamination.pdf](files/examples/paper/results_pipeline/H9_PRO-seq_2/QC_hg38/H9_PRO-seq_2_mRNA_contamination.pdf)
            * [:fa-file-image-o: H9_PRO-seq_2_mRNA_contamination.png](files/examples/paper/results_pipeline/H9_PRO-seq_2/QC_hg38/H9_PRO-seq_2_mRNA_contamination.png)
            * :fa-file-o: H9_PRO-seq_2_pause_index.bed.gz
            * [:fa-file-pdf-o: H9_PRO-seq_2_pause_index.pdf](files/examples/paper/results_pipeline/H9_PRO-seq_2/QC_hg38/H9_PRO-seq_2_pause_index.pdf)
            * [:fa-file-image-o: H9_PRO-seq_2_pause_index.png](files/examples/paper/results_pipeline/H9_PRO-seq_2/QC_hg38/H9_PRO-seq_2_pause_index.png)          
            * :fa-file-text-o: H9_PRO-seq_2_preseq_counts.txt
            * :fa-file-text-o: H9_PRO-seq_2_preseq_out.txt
            * [:fa-file-pdf-o: H9_PRO-seq_2_preseq_plot.pdf](files/examples/paper/results_pipeline/H9_PRO-seq_2/QC_hg38/H9_PRO-seq_2_preseq_plot.pdf)
            * [:fa-file-image-o: H9_PRO-seq_2_preseq_plot.png](files/examples/paper/results_pipeline/H9_PRO-seq_2/QC_hg38/H9_PRO-seq_2_preseq_plot.png)
            * :fa-file-text-o: H9_PRO-seq_2_preseq_yield.txt
            * [:fa-file-pdf-o: H9_PRO-seq_2_FRiF.pdf](files/examples/paper/results_pipeline/H9_PRO-seq_2/QC_hg38/H9_PRO-seq_2_FRiF.pdf)
            * [:fa-file-image-o: H9_PRO-seq_2_FRiF.png](files/examples/paper/results_pipeline/H9_PRO-seq_2/QC_hg38/H9_PRO-seq_2_FRiF.png)
            * [:fa-file-pdf-o: H9_PRO-seq_2_TSSenrichment.pdf](files/examples/paper/results_pipeline/H9_PRO-seq_2/QC_hg38/H9_PRO-seq_2_TSSenrichment.pdf)
            * [:fa-file-image-o: H9_PRO-seq_2_TSSenrichment.png](files/examples/paper/results_pipeline/H9_PRO-seq_2/QC_hg38/H9_PRO-seq_2_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: H9_PRO-seq_2_r1.fastq.gz
            * :fa-file-archive-o: H9_PRO-seq_2_r2.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: H9_PRO-seq_2_gene_coverage.bed
            * :fa-file-o: H9_PRO-seq_2_minus_exact_body_0-mer.bw
            * :fa-file-o: H9_PRO-seq_2_plus_exact_body_0.bw
            * :fa-file-o: H9_PRO-seq_2_minus_smooth_body_0-mer.bw
            * :fa-file-o: H9_PRO-seq_2_plus_smooth_body_0.bw 
        * [:fa-file-text-o: objects.tsv](files/examples/paper/results_pipeline/H9_PRO-seq_2/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](files/examples/paper/results_pipeline/H9_PRO-seq_2/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](files/examples/paper/results_pipeline/H9_PRO-seq_2/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](files/examples/paper/results_pipeline/H9_PRO-seq_2/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](files/examples/paper/results_pipeline/H9_PRO-seq_2/stats.tsv)
    * ??? danger ":fa-folder-open-o: H9_PRO-seq_3/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: H9_PRO-seq_3_fail_qc.bam
            * :fa-file-o: H9_PRO-seq_3_sort_dups.bam
            * :fa-file-o: H9_PRO-seq_3_sort.bam
            * :fa-file-o: H9_PRO-seq_3_PE1.bam
            * :fa-file-o: H9_PRO-seq_3_PE1.bam.bai
            * :fa-file-o: H9_PRO-seq_3_PE2.bam
            * :fa-file-o: H9_PRO-seq_3_dups_PE2.bam
            * :fa-file-o: H9_PRO-seq_3_unmap.bam
            * :fa-file-o: H9_PRO-seq_3_plus.bam 
            * :fa-file-o: H9_PRO-seq_3_plus.bam.bai
            * :fa-file-o: H9_PRO-seq_3_minus.bam 
            * :fa-file-o: H9_PRO-seq_3_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: H9_PRO-seq_3_adapter_insertion_distribution.pdf](files/examples/paper/results_pipeline/H9_PRO-seq_3/cutadapt/H9_PRO-seq_3_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: H9_PRO-seq_3_adapter_insertion_distribution.png](files/examples/paper/results_pipeline/H9_PRO-seq_3/cutadapt/H9_PRO-seq_3_adapter_insertion_distribution.png)
            * :fa-file-text-o: H9_PRO-seq_3_R1_cutadapt.txt
            * :fa-file-text-o: H9_PRO-seq_3_R2_cutadapt.txt
            * :fa-file-text-o: H9_PRO-seq_3.hist
            * :fa-file-text-o: H9_PRO-seq_3.histogram
            * :fa-file-text-o: H9_PRO-seq_3.extendedFrags.fastq.gz
            * :fa-file-text-o: H9_PRO-seq_3.notCombined_1.fastq.gz
            * :fa-file-text-o: H9_PRO-seq_3.notCombined_2.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * :fa-file-code-o: H9_PRO-seq_3_R1_processed_fastqc.zip
            * [:fa-file-code-o: H9_PRO-seq_3_R1_processed_fastqc.html](files/examples/paper/results_pipeline/H9_PRO-seq_3/fastqc/H9_PRO-seq_3_R1_processed_fastqc.html)
            * :fa-file-text-o: H9_PRO-seq_3_R2_trimmed_fastqc.zip
            * [:fa-file-code-o: H9_PRO-seq_3_R2_trimmed_fastqc.html](files/examples/paper/results_pipeline/H9_PRO-seq_3/fastqc/H9_PRO-seq_3_R2_trimmed_fastqc.html)
        * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-archive-o: H9_PRO-seq_3_human_rDNA_unmap_R1.fq.gz
            * :fa-file-archive-o: H9_PRO-seq_3_human_rDNA_unmap_R2.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: H9_PRO-seq_3_bamQC.tsv
            * :fa-file-o: H9_PRO-seq_3_exon_intron_ratios.bed.gz
            * [:fa-file-pdf-o: H9_PRO-seq_3_cFRiF.pdf](files/examples/paper/results_pipeline/H9_PRO-seq_3/QC_hg38/H9_PRO-seq_3_cFRiF.pdf)
            * [:fa-file-pdf-o: H9_PRO-seq_3_cFRiF.png](files/examples/paper/results_pipeline/H9_PRO-seq_3/QC_hg38/H9_PRO-seq_3_cFRiF.png)
            * [:fa-file-pdf-o: H9_PRO-seq_3_mRNA_contamination.pdf](files/examples/paper/results_pipeline/H9_PRO-seq_3/QC_hg38/H9_PRO-seq_3_mRNA_contamination.pdf)
            * [:fa-file-pdf-o: H9_PRO-seq_3_mRNA_contamination.png](files/examples/paper/results_pipeline/H9_PRO-seq_3/QC_hg38/H9_PRO-seq_3_mRNA_contamination.png)
            * :fa-file-o: H9_PRO-seq_3_pause_index.bed.gz
            * [:fa-file-pdf-o: H9_PRO-seq_3_pause_index.pdf](files/examples/paper/results_pipeline/H9_PRO-seq_3/QC_hg38/H9_PRO-seq_3_pause_index.pdf)
            * [:fa-file-pdf-o: H9_PRO-seq_3_pause_index.png](files/examples/paper/results_pipeline/H9_PRO-seq_3/QC_hg38/H9_PRO-seq_3_pause_index.png)          
            * :fa-file-text-o: H9_PRO-seq_3_preseq_counts.txt
            * :fa-file-text-o: H9_PRO-seq_3_preseq_out.txt
            * [:fa-file-pdf-o: H9_PRO-seq_3_preseq_plot.pdf](files/examples/paper/results_pipeline/H9_PRO-seq_3/QC_hg38/H9_PRO-seq_3_preseq_plot.pdf)
            * [:fa-file-pdf-o: H9_PRO-seq_3_preseq_plot.png](files/examples/paper/results_pipeline/H9_PRO-seq_3/QC_hg38/H9_PRO-seq_3_preseq_plot.png)
            * :fa-file-text-o: H9_PRO-seq_3_preseq_yield.txt
            * [:fa-file-pdf-o: H9_PRO-seq_3_FRiF.pdf](files/examples/paper/results_pipeline/H9_PRO-seq_3/QC_hg38/H9_PRO-seq_3_FRiF.pdf)
            * [:fa-file-pdf-o: H9_PRO-seq_3_FRiF.png](files/examples/paper/results_pipeline/H9_PRO-seq_3/QC_hg38/H9_PRO-seq_3_FRiF.png)
            * [:fa-file-pdf-o: H9_PRO-seq_3_TSSenrichment.pdf](files/examples/paper/results_pipeline/H9_PRO-seq_3/QC_hg38/H9_PRO-seq_3_TSSenrichment.pdf)
            * [:fa-file-pdf-o: H9_PRO-seq_3_TSSenrichment.png](files/examples/paper/results_pipeline/H9_PRO-seq_3/QC_hg38/H9_PRO-seq_3_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: H9_PRO-seq_3_r1.fastq.gz
            * :fa-file-archive-o: H9_PRO-seq_3_r2.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: H9_PRO-seq_3_gene_coverage.bed
            * :fa-file-o: H9_PRO-seq_3_minus_exact_body_0-mer.bw
            * :fa-file-o: H9_PRO-seq_3_plus_exact_body_0.bw
            * :fa-file-o: H9_PRO-seq_3_minus_smooth_body_0-mer.bw
            * :fa-file-o: H9_PRO-seq_3_plus_smooth_body_0.bw 
        * [:fa-file-text-o: objects.tsv](files/examples/paper/results_pipeline/H9_PRO-seq_3/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](files/examples/paper/results_pipeline/H9_PRO-seq_3/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](files/examples/paper/results_pipeline/H9_PRO-seq_3/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](files/examples/paper/results_pipeline/H9_PRO-seq_3/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](files/examples/paper/results_pipeline/H9_PRO-seq_3/stats.tsv)
    * ??? danger ":fa-folder-open-o: H9_treated_PRO-seq_1/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: H9_treated_PRO-seq_1_fail_qc.bam
            * :fa-file-o: H9_treated_PRO-seq_1_sort_dups.bam
            * :fa-file-o: H9_treated_PRO-seq_1_sort.bam
            * :fa-file-o: H9_treated_PRO-seq_1_PE1.bam
            * :fa-file-o: H9_treated_PRO-seq_1_PE1.bam.bai
            * :fa-file-o: H9_treated_PRO-seq_1_PE2.bam
            * :fa-file-o: H9_treated_PRO-seq_1_dups_PE2.bam
            * :fa-file-o: H9_treated_PRO-seq_1_unmap.bam
            * :fa-file-o: H9_treated_PRO-seq_1_plus.bam 
            * :fa-file-o: H9_treated_PRO-seq_1_plus.bam.bai
            * :fa-file-o: H9_treated_PRO-seq_1_minus.bam 
            * :fa-file-o: H9_treated_PRO-seq_1_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: H9_treated_PRO-seq_1_adapter_insertion_distribution.pdf](files/examples/paper/results_pipeline/H9_treated_PRO-seq_1/cutadapt/H9_treated_PRO-seq_1_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: H9_treated_PRO-seq_1_adapter_insertion_distribution.png](files/examples/paper/results_pipeline/H9_treated_PRO-seq_1/cutadapt/H9_treated_PRO-seq_1_adapter_insertion_distribution.png)
            * :fa-file-text-o: H9_treated_PRO-seq_1_R1_cutadapt.txt
            * :fa-file-text-o: H9_treated_PRO-seq_1_R2_cutadapt.txt
            * :fa-file-text-o: H9_treated_PRO-seq_1.hist
            * :fa-file-text-o: H9_treated_PRO-seq_1.histogram
            * :fa-file-text-o: H9_treated_PRO-seq_1.extendedFrags.fastq.gz
            * :fa-file-text-o: H9_treated_PRO-seq_1.notCombined_1.fastq.gz
            * :fa-file-text-o: H9_treated_PRO-seq_1.notCombined_2.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * :fa-file-code-o: H9_treated_PRO-seq_1_R1_processed_fastqc.zip
            * [:fa-file-code-o: H9_treated_PRO-seq_1_R1_processed_fastqc.html](files/examples/paper/results_pipeline/H9_treated_PRO-seq_1/fastqc/H9_treated_PRO-seq_1_R1_processed_fastqc.html)
            * :fa-file-text-o: H9_treated_PRO-seq_1_R2_trimmed_fastqc.zip
            * [:fa-file-code-o: H9_treated_PRO-seq_1_R2_trimmed_fastqc.html](files/examples/paper/results_pipeline/H9_treated_PRO-seq_1/fastqc/H9_treated_PRO-seq_1_R2_trimmed_fastqc.html)
        * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-archive-o: H9_treated_PRO-seq_1_human_rDNA_unmap_R1.fq.gz
            * :fa-file-archive-o: H9_treated_PRO-seq_1_human_rDNA_unmap_R2.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: H9_treated_PRO-seq_1_bamQC.tsv
            * :fa-file-o: H9_treated_PRO-seq_1_exon_intron_ratios.bed.gz
            * [:fa-file-pdf-o: H9_treated_PRO-seq_1_cFRiF.pdf](files/examples/paper/results_pipeline/H9_treated_PRO-seq_1/QC_hg38/H9_treated_PRO-seq_1_cFRiF.pdf)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_1_cFRiF.png](files/examples/paper/results_pipeline/H9_treated_PRO-seq_1/QC_hg38/H9_treated_PRO-seq_1_cFRiF.png)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_1_mRNA_contamination.pdf](files/examples/paper/results_pipeline/H9_treated_PRO-seq_1/QC_hg38/H9_treated_PRO-seq_1_mRNA_contamination.pdf)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_1_mRNA_contamination.png](files/examples/paper/results_pipeline/H9_treated_PRO-seq_1/QC_hg38/H9_treated_PRO-seq_1_mRNA_contamination.png)
            * :fa-file-o: H9_treated_PRO-seq_1_pause_index.bed.gz
            * [:fa-file-pdf-o: H9_treated_PRO-seq_1_pause_index.pdf](files/examples/paper/results_pipeline/H9_treated_PRO-seq_1/QC_hg38/H9_treated_PRO-seq_1_pause_index.pdf)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_1_pause_index.png](files/examples/paper/results_pipeline/H9_treated_PRO-seq_1/QC_hg38/H9_treated_PRO-seq_1_pause_index.png)          
            * :fa-file-text-o: H9_treated_PRO-seq_1_preseq_counts.txt
            * :fa-file-text-o: H9_treated_PRO-seq_1_preseq_out.txt
            * [:fa-file-pdf-o: H9_treated_PRO-seq_1_preseq_plot.pdf](files/examples/paper/results_pipeline/H9_treated_PRO-seq_1/QC_hg38/H9_treated_PRO-seq_1_preseq_plot.pdf)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_1_preseq_plot.png](files/examples/paper/results_pipeline/H9_treated_PRO-seq_1/QC_hg38/H9_treated_PRO-seq_1_preseq_plot.png)
            * :fa-file-text-o: H9_treated_PRO-seq_1_preseq_yield.txt
            * [:fa-file-pdf-o: H9_treated_PRO-seq_1_FRiF.pdf](files/examples/paper/results_pipeline/H9_treated_PRO-seq_1/QC_hg38/H9_treated_PRO-seq_1_FRiF.pdf)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_1_FRiF.png](files/examples/paper/results_pipeline/H9_treated_PRO-seq_1/QC_hg38/H9_treated_PRO-seq_1_FRiF.png)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_1_TSSenrichment.pdf](files/examples/paper/results_pipeline/H9_treated_PRO-seq_1/QC_hg38/H9_treated_PRO-seq_1_TSSenrichment.pdf)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_1_TSSenrichment.png](files/examples/paper/results_pipeline/H9_treated_PRO-seq_1/QC_hg38/H9_treated_PRO-seq_1_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: H9_treated_PRO-seq_1_r1.fastq.gz
            * :fa-file-archive-o: H9_treated_PRO-seq_1_r2.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: H9_treated_PRO-seq_1_gene_coverage.bed
            * :fa-file-o: H9_treated_PRO-seq_1_minus_exact_body_0-mer.bw
            * :fa-file-o: H9_treated_PRO-seq_1_plus_exact_body_0.bw
            * :fa-file-o: H9_treated_PRO-seq_1_minus_smooth_body_0-mer.bw
            * :fa-file-o: H9_treated_PRO-seq_1_plus_smooth_body_0.bw 
        * [:fa-file-text-o: objects.tsv](files/examples/paper/results_pipeline/H9_treated_PRO-seq_1/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](files/examples/paper/results_pipeline/H9_treated_PRO-seq_1/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](files/examples/paper/results_pipeline/H9_treated_PRO-seq_1/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](files/examples/paper/results_pipeline/H9_treated_PRO-seq_1/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](files/examples/paper/results_pipeline/H9_treated_PRO-seq_1/stats.tsv)
    * ??? danger ":fa-folder-open-o: H9_treated_PRO-seq_2/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: H9_treated_PRO-seq_2_fail_qc.bam
            * :fa-file-o: H9_treated_PRO-seq_2_sort_dups.bam
            * :fa-file-o: H9_treated_PRO-seq_2_sort.bam
            * :fa-file-o: H9_treated_PRO-seq_2_PE1.bam
            * :fa-file-o: H9_treated_PRO-seq_2_PE1.bam.bai
            * :fa-file-o: H9_treated_PRO-seq_2_PE2.bam
            * :fa-file-o: H9_treated_PRO-seq_2_dups_PE2.bam
            * :fa-file-o: H9_treated_PRO-seq_2_unmap.bam
            * :fa-file-o: H9_treated_PRO-seq_2_plus.bam 
            * :fa-file-o: H9_treated_PRO-seq_2_plus.bam.bai
            * :fa-file-o: H9_treated_PRO-seq_2_minus.bam 
            * :fa-file-o: H9_treated_PRO-seq_2_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: H9_treated_PRO-seq_2_adapter_insertion_distribution.pdf](files/examples/paper/results_pipeline/H9_treated_PRO-seq_2/cutadapt/H9_treated_PRO-seq_2_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: H9_treated_PRO-seq_2_adapter_insertion_distribution.png](files/examples/paper/results_pipeline/H9_treated_PRO-seq_2/cutadapt/H9_treated_PRO-seq_2_adapter_insertion_distribution.png)
            * :fa-file-text-o: H9_treated_PRO-seq_2_R1_cutadapt.txt
            * :fa-file-text-o: H9_treated_PRO-seq_2_R2_cutadapt.txt
            * :fa-file-text-o: H9_treated_PRO-seq_2.hist
            * :fa-file-text-o: H9_treated_PRO-seq_2.histogram
            * :fa-file-text-o: H9_treated_PRO-seq_2.extendedFrags.fastq.gz
            * :fa-file-text-o: H9_treated_PRO-seq_2.notCombined_1.fastq.gz
            * :fa-file-text-o: H9_treated_PRO-seq_2.notCombined_2.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * :fa-file-code-o: H9_treated_PRO-seq_2_R1_processed_fastqc.zip
            * [:fa-file-code-o: H9_treated_PRO-seq_2_R1_processed_fastqc.html](files/examples/paper/results_pipeline/H9_treated_PRO-seq_2/fastqc/H9_treated_PRO-seq_2_R1_processed_fastqc.html)
            * :fa-file-text-o: H9_treated_PRO-seq_2_R2_trimmed_fastqc.zip
            * [:fa-file-code-o: H9_treated_PRO-seq_2_R2_trimmed_fastqc.html](files/examples/paper/results_pipeline/H9_treated_PRO-seq_2/fastqc/H9_treated_PRO-seq_2_R2_trimmed_fastqc.html)
        * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-archive-o: H9_treated_PRO-seq_2_human_rDNA_unmap_R1.fq.gz
            * :fa-file-archive-o: H9_treated_PRO-seq_2_human_rDNA_unmap_R2.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: H9_treated_PRO-seq_2_bamQC.tsv
            * :fa-file-o: H9_treated_PRO-seq_2_exon_intron_ratios.bed.gz
            * [:fa-file-pdf-o: H9_treated_PRO-seq_2_cFRiF.pdf](files/examples/paper/results_pipeline/H9_treated_PRO-seq_2/QC_hg38/H9_treated_PRO-seq_2_cFRiF.pdf)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_2_cFRiF.png](files/examples/paper/results_pipeline/H9_treated_PRO-seq_2/QC_hg38/H9_treated_PRO-seq_2_cFRiF.png)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_2_mRNA_contamination.pdf](files/examples/paper/results_pipeline/H9_treated_PRO-seq_2/QC_hg38/H9_treated_PRO-seq_2_mRNA_contamination.pdf)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_2_mRNA_contamination.png](files/examples/paper/results_pipeline/H9_treated_PRO-seq_2/QC_hg38/H9_treated_PRO-seq_2_mRNA_contamination.png)
            * :fa-file-o: H9_treated_PRO-seq_2_pause_index.bed.gz
            * [:fa-file-pdf-o: H9_treated_PRO-seq_2_pause_index.pdf](files/examples/paper/results_pipeline/H9_treated_PRO-seq_2/QC_hg38/H9_treated_PRO-seq_2_pause_index.pdf)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_2_pause_index.png](files/examples/paper/results_pipeline/H9_treated_PRO-seq_2/QC_hg38/H9_treated_PRO-seq_2_pause_index.png)          
            * :fa-file-text-o: H9_treated_PRO-seq_2_preseq_counts.txt
            * :fa-file-text-o: H9_treated_PRO-seq_2_preseq_out.txt
            * [:fa-file-pdf-o: H9_treated_PRO-seq_2_preseq_plot.pdf](files/examples/paper/results_pipeline/H9_treated_PRO-seq_2/QC_hg38/H9_treated_PRO-seq_2_preseq_plot.pdf)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_2_preseq_plot.png](files/examples/paper/results_pipeline/H9_treated_PRO-seq_2/QC_hg38/H9_treated_PRO-seq_2_preseq_plot.png)
            * :fa-file-text-o: H9_treated_PRO-seq_2_preseq_yield.txt
            * [:fa-file-pdf-o: H9_treated_PRO-seq_2_FRiF.pdf](files/examples/paper/results_pipeline/H9_treated_PRO-seq_2/QC_hg38/H9_treated_PRO-seq_2_FRiF.pdf)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_2_FRiF.png](files/examples/paper/results_pipeline/H9_treated_PRO-seq_2/QC_hg38/H9_treated_PRO-seq_2_FRiF.png)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_2_TSSenrichment.pdf](files/examples/paper/results_pipeline/H9_treated_PRO-seq_2/QC_hg38/H9_treated_PRO-seq_2_TSSenrichment.pdf)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_2_TSSenrichment.png](files/examples/paper/results_pipeline/H9_treated_PRO-seq_2/QC_hg38/H9_treated_PRO-seq_2_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: H9_treated_PRO-seq_2_r1.fastq.gz
            * :fa-file-archive-o: H9_treated_PRO-seq_2_r2.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: H9_treated_PRO-seq_2_gene_coverage.bed
            * :fa-file-o: H9_treated_PRO-seq_2_minus_exact_body_0-mer.bw
            * :fa-file-o: H9_treated_PRO-seq_2_plus_exact_body_0.bw
            * :fa-file-o: H9_treated_PRO-seq_2_minus_smooth_body_0-mer.bw
            * :fa-file-o: H9_treated_PRO-seq_2_plus_smooth_body_0.bw 
        * [:fa-file-text-o: objects.tsv](files/examples/paper/results_pipeline/H9_treated_PRO-seq_2/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](files/examples/paper/results_pipeline/H9_treated_PRO-seq_2/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](files/examples/paper/results_pipeline/H9_treated_PRO-seq_2/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](files/examples/paper/results_pipeline/H9_treated_PRO-seq_2/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](files/examples/paper/results_pipeline/H9_treated_PRO-seq_2/stats.tsv)
    * ??? danger ":fa-folder-open-o: H9_treated_PRO-seq_3/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: H9_treated_PRO-seq_3_fail_qc.bam
            * :fa-file-o: H9_treated_PRO-seq_3_sort_dups.bam
            * :fa-file-o: H9_treated_PRO-seq_3_sort.bam
            * :fa-file-o: H9_treated_PRO-seq_3_PE1.bam
            * :fa-file-o: H9_treated_PRO-seq_3_PE1.bam.bai
            * :fa-file-o: H9_treated_PRO-seq_3_PE2.bam
            * :fa-file-o: H9_treated_PRO-seq_3_dups_PE2.bam
            * :fa-file-o: H9_treated_PRO-seq_3_unmap.bam
            * :fa-file-o: H9_treated_PRO-seq_3_plus.bam 
            * :fa-file-o: H9_treated_PRO-seq_3_plus.bam.bai
            * :fa-file-o: H9_treated_PRO-seq_3_minus.bam 
            * :fa-file-o: H9_treated_PRO-seq_3_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: H9_treated_PRO-seq_3_adapter_insertion_distribution.pdf](files/examples/paper/results_pipeline/H9_treated_PRO-seq_3/cutadapt/H9_treated_PRO-seq_3_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: H9_treated_PRO-seq_3_adapter_insertion_distribution.png](files/examples/paper/results_pipeline/H9_treated_PRO-seq_3/cutadapt/H9_treated_PRO-seq_3_adapter_insertion_distribution.png)
            * :fa-file-text-o: H9_treated_PRO-seq_3_R1_cutadapt.txt
            * :fa-file-text-o: H9_treated_PRO-seq_3_R2_cutadapt.txt
            * :fa-file-text-o: H9_treated_PRO-seq_3.hist
            * :fa-file-text-o: H9_treated_PRO-seq_3.histogram
            * :fa-file-text-o: H9_treated_PRO-seq_3.extendedFrags.fastq.gz
            * :fa-file-text-o: H9_treated_PRO-seq_3.notCombined_1.fastq.gz
            * :fa-file-text-o: H9_treated_PRO-seq_3.notCombined_2.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * :fa-file-code-o: H9_treated_PRO-seq_3_R1_processed_fastqc.zip
            * [:fa-file-code-o: H9_treated_PRO-seq_3_R1_processed_fastqc.html](files/examples/paper/results_pipeline/H9_treated_PRO-seq_3/fastqc/H9_treated_PRO-seq_3_R1_processed_fastqc.html)
            * :fa-file-text-o: H9_treated_PRO-seq_3_R2_trimmed_fastqc.zip
            * [:fa-file-code-o: H9_treated_PRO-seq_3_R2_trimmed_fastqc.html](files/examples/paper/results_pipeline/H9_treated_PRO-seq_3/fastqc/H9_treated_PRO-seq_3_R2_trimmed_fastqc.html)
        * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-archive-o: H9_treated_PRO-seq_3_human_rDNA_unmap_R1.fq.gz
            * :fa-file-archive-o: H9_treated_PRO-seq_3_human_rDNA_unmap_R2.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: H9_treated_PRO-seq_3_bamQC.tsv
            * :fa-file-o: H9_treated_PRO-seq_3_exon_intron_ratios.bed.gz
            * [:fa-file-pdf-o: H9_treated_PRO-seq_3_cFRiF.pdf](files/examples/paper/results_pipeline/H9_treated_PRO-seq_3/QC_hg38/H9_treated_PRO-seq_3_cFRiF.pdf)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_3_cFRiF.png](files/examples/paper/results_pipeline/H9_treated_PRO-seq_3/QC_hg38/H9_treated_PRO-seq_3_cFRiF.png)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_3_mRNA_contamination.pdf](files/examples/paper/results_pipeline/H9_treated_PRO-seq_3/QC_hg38/H9_treated_PRO-seq_3_mRNA_contamination.pdf)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_3_mRNA_contamination.png](files/examples/paper/results_pipeline/H9_treated_PRO-seq_3/QC_hg38/H9_treated_PRO-seq_3_mRNA_contamination.png)
            * :fa-file-o: H9_treated_PRO-seq_3_pause_index.bed.gz
            * [:fa-file-pdf-o: H9_treated_PRO-seq_3_pause_index.pdf](files/examples/paper/results_pipeline/H9_treated_PRO-seq_3/QC_hg38/H9_treated_PRO-seq_3_pause_index.pdf)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_3_pause_index.png](files/examples/paper/results_pipeline/H9_treated_PRO-seq_3/QC_hg38/H9_treated_PRO-seq_3_pause_index.png)          
            * :fa-file-text-o: H9_treated_PRO-seq_3_preseq_counts.txt
            * :fa-file-text-o: H9_treated_PRO-seq_3_preseq_out.txt
            * [:fa-file-pdf-o: H9_treated_PRO-seq_3_preseq_plot.pdf](files/examples/paper/results_pipeline/H9_treated_PRO-seq_3/QC_hg38/H9_treated_PRO-seq_3_preseq_plot.pdf)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_3_preseq_plot.png](files/examples/paper/results_pipeline/H9_treated_PRO-seq_3/QC_hg38/H9_treated_PRO-seq_3_preseq_plot.png)
            * :fa-file-text-o: H9_treated_PRO-seq_3_preseq_yield.txt
            * [:fa-file-pdf-o: H9_treated_PRO-seq_3_FRiF.pdf](files/examples/paper/results_pipeline/H9_treated_PRO-seq_3/QC_hg38/H9_treated_PRO-seq_3_FRiF.pdf)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_3_FRiF.png](files/examples/paper/results_pipeline/H9_treated_PRO-seq_3/QC_hg38/H9_treated_PRO-seq_3_FRiF.png)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_3_TSSenrichment.pdf](files/examples/paper/results_pipeline/H9_treated_PRO-seq_3/QC_hg38/H9_treated_PRO-seq_3_TSSenrichment.pdf)
            * [:fa-file-pdf-o: H9_treated_PRO-seq_3_TSSenrichment.png](files/examples/paper/results_pipeline/H9_treated_PRO-seq_3/QC_hg38/H9_treated_PRO-seq_3_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: H9_treated_PRO-seq_3_r1.fastq.gz
            * :fa-file-archive-o: H9_treated_PRO-seq_3_r2.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: H9_treated_PRO-seq_3_gene_coverage.bed
            * :fa-file-o: H9_treated_PRO-seq_3_minus_exact_body_0-mer.bw
            * :fa-file-o: H9_treated_PRO-seq_3_plus_exact_body_0.bw
            * :fa-file-o: H9_treated_PRO-seq_3_minus_smooth_body_0-mer.bw
            * :fa-file-o: H9_treated_PRO-seq_3_plus_smooth_body_0.bw 
        * [:fa-file-text-o: objects.tsv](files/examples/paper/results_pipeline/H9_treated_PRO-seq_3/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](files/examples/paper/results_pipeline/H9_treated_PRO-seq_3/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](files/examples/paper/results_pipeline/H9_treated_PRO-seq_3/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](files/examples/paper/results_pipeline/H9_treated_PRO-seq_3/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](files/examples/paper/results_pipeline/H9_treated_PRO-seq_3/stats.tsv)
    * ??? danger ":fa-folder-open-o: HEK_ARF_PRO-seq/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: HEK_ARF_PRO-seq_fail_qc.bam
            * :fa-file-o: HEK_ARF_PRO-seq_sort_dups.bam
            * :fa-file-o: HEK_ARF_PRO-seq_sort.bam
            * :fa-file-o: HEK_ARF_PRO-seq_PE1.bam
            * :fa-file-o: HEK_ARF_PRO-seq_PE1.bam.bai
            * :fa-file-o: HEK_ARF_PRO-seq_PE2.bam
            * :fa-file-o: HEK_ARF_PRO-seq_dups_PE2.bam
            * :fa-file-o: HEK_ARF_PRO-seq_unmap.bam
            * :fa-file-o: HEK_ARF_PRO-seq_plus.bam 
            * :fa-file-o: HEK_ARF_PRO-seq_plus.bam.bai
            * :fa-file-o: HEK_ARF_PRO-seq_minus.bam 
            * :fa-file-o: HEK_ARF_PRO-seq_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: HEK_ARF_PRO-seq_adapter_insertion_distribution.pdf](files/examples/paper/results_pipeline/HEK_ARF_PRO-seq/cutadapt/HEK_ARF_PRO-seq_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: HEK_ARF_PRO-seq_adapter_insertion_distribution.png](files/examples/paper/results_pipeline/HEK_ARF_PRO-seq/cutadapt/HEK_ARF_PRO-seq_adapter_insertion_distribution.png)
            * :fa-file-text-o: HEK_ARF_PRO-seq_R1_cutadapt.txt
            * :fa-file-text-o: HEK_ARF_PRO-seq_R2_cutadapt.txt
            * :fa-file-text-o: HEK_ARF_PRO-seq.hist
            * :fa-file-text-o: HEK_ARF_PRO-seq.histogram
            * :fa-file-text-o: HEK_ARF_PRO-seq.extendedFrags.fastq.gz
            * :fa-file-text-o: HEK_ARF_PRO-seq.notCombined_1.fastq.gz
            * :fa-file-text-o: HEK_ARF_PRO-seq.notCombined_2.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * :fa-file-code-o: HEK_ARF_PRO-seq_R1_processed_fastqc.zip
            * [:fa-file-code-o: HEK_ARF_PRO-seq_R1_processed_fastqc.html](files/examples/paper/results_pipeline/HEK_ARF_PRO-seq/fastqc/HEK_ARF_PRO-seq_R1_processed_fastqc.html)
            * :fa-file-text-o: HEK_ARF_PRO-seq_R2_trimmed_fastqc.zip
            * [:fa-file-code-o: HEK_ARF_PRO-seq_R2_trimmed_fastqc.html](files/examples/paper/results_pipeline/HEK_ARF_PRO-seq/fastqc/HEK_ARF_PRO-seq_R2_trimmed_fastqc.html)
        * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-archive-o: HEK_ARF_PRO-seq_human_rDNA_unmap_R1.fq.gz
            * :fa-file-archive-o: HEK_ARF_PRO-seq_human_rDNA_unmap_R2.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: HEK_ARF_PRO-seq_bamQC.tsv
            * :fa-file-o: HEK_ARF_PRO-seq_exon_intron_ratios.bed.gz
            * [:fa-file-pdf-o: HEK_ARF_PRO-seq_cFRiF.pdf](files/examples/paper/results_pipeline/HEK_ARF_PRO-seq/QC_hg38/HEK_ARF_PRO-seq_cFRiF.pdf)
            * [:fa-file-pdf-o: HEK_ARF_PRO-seq_cFRiF.png](files/examples/paper/results_pipeline/HEK_ARF_PRO-seq/QC_hg38/HEK_ARF_PRO-seq_cFRiF.png)
            * [:fa-file-pdf-o: HEK_ARF_PRO-seq_mRNA_contamination.pdf](files/examples/paper/results_pipeline/HEK_ARF_PRO-seq/QC_hg38/HEK_ARF_PRO-seq_mRNA_contamination.pdf)
            * [:fa-file-pdf-o: HEK_ARF_PRO-seq_mRNA_contamination.png](files/examples/paper/results_pipeline/HEK_ARF_PRO-seq/QC_hg38/HEK_ARF_PRO-seq_mRNA_contamination.png)
            * :fa-file-o: HEK_ARF_PRO-seq_pause_index.bed.gz
            * [:fa-file-pdf-o: HEK_ARF_PRO-seq_pause_index.pdf](files/examples/paper/results_pipeline/HEK_ARF_PRO-seq/QC_hg38/HEK_ARF_PRO-seq_pause_index.pdf)
            * [:fa-file-pdf-o: HEK_ARF_PRO-seq_pause_index.png](files/examples/paper/results_pipeline/HEK_ARF_PRO-seq/QC_hg38/HEK_ARF_PRO-seq_pause_index.png)          
            * :fa-file-text-o: HEK_ARF_PRO-seq_preseq_counts.txt
            * :fa-file-text-o: HEK_ARF_PRO-seq_preseq_out.txt
            * [:fa-file-pdf-o: HEK_ARF_PRO-seq_preseq_plot.pdf](files/examples/paper/results_pipeline/HEK_ARF_PRO-seq/QC_hg38/HEK_ARF_PRO-seq_preseq_plot.pdf)
            * [:fa-file-pdf-o: HEK_ARF_PRO-seq_preseq_plot.png](files/examples/paper/results_pipeline/HEK_ARF_PRO-seq/QC_hg38/HEK_ARF_PRO-seq_preseq_plot.png)
            * :fa-file-text-o: HEK_ARF_PRO-seq_preseq_yield.txt
            * [:fa-file-pdf-o: HEK_ARF_PRO-seq_FRiF.pdf](files/examples/paper/results_pipeline/HEK_ARF_PRO-seq/QC_hg38/HEK_ARF_PRO-seq_FRiF.pdf)
            * [:fa-file-pdf-o: HEK_ARF_PRO-seq_FRiF.png](files/examples/paper/results_pipeline/HEK_ARF_PRO-seq/QC_hg38/HEK_ARF_PRO-seq_FRiF.png)
            * [:fa-file-pdf-o: HEK_ARF_PRO-seq_TSSenrichment.pdf](files/examples/paper/results_pipeline/HEK_ARF_PRO-seq/QC_hg38/HEK_ARF_PRO-seq_TSSenrichment.pdf)
            * [:fa-file-pdf-o: HEK_ARF_PRO-seq_TSSenrichment.png](files/examples/paper/results_pipeline/HEK_ARF_PRO-seq/QC_hg38/HEK_ARF_PRO-seq_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: HEK_ARF_PRO-seq_r1.fastq.gz
            * :fa-file-archive-o: HEK_ARF_PRO-seq_r2.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: HEK_ARF_PRO-seq_gene_coverage.bed
            * :fa-file-o: HEK_ARF_PRO-seq_minus_exact_body_0-mer.bw
            * :fa-file-o: HEK_ARF_PRO-seq_plus_exact_body_0.bw
            * :fa-file-o: HEK_ARF_PRO-seq_minus_smooth_body_0-mer.bw
            * :fa-file-o: HEK_ARF_PRO-seq_plus_smooth_body_0.bw 
        * [:fa-file-text-o: objects.tsv](files/examples/paper/results_pipeline/HEK_ARF_PRO-seq/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](files/examples/paper/results_pipeline/HEK_ARF_PRO-seq/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](files/examples/paper/results_pipeline/HEK_ARF_PRO-seq/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](files/examples/paper/results_pipeline/HEK_ARF_PRO-seq/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](files/examples/paper/results_pipeline/HEK_ARF_PRO-seq/stats.tsv)
    * ??? danger ":fa-folder-open-o: HEK_PRO-seq/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: HEK_PRO-seq_fail_qc.bam
            * :fa-file-o: HEK_PRO-seq_sort_dups.bam
            * :fa-file-o: HEK_PRO-seq_sort.bam
            * :fa-file-o: HEK_PRO-seq_PE1.bam
            * :fa-file-o: HEK_PRO-seq_PE1.bam.bai
            * :fa-file-o: HEK_PRO-seq_PE2.bam
            * :fa-file-o: HEK_PRO-seq_dups_PE2.bam
            * :fa-file-o: HEK_PRO-seq_unmap.bam
            * :fa-file-o: HEK_PRO-seq_plus.bam 
            * :fa-file-o: HEK_PRO-seq_plus.bam.bai
            * :fa-file-o: HEK_PRO-seq_minus.bam 
            * :fa-file-o: HEK_PRO-seq_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: HEK_PRO-seq_adapter_insertion_distribution.pdf](files/examples/paper/results_pipeline/HEK_PRO-seq/cutadapt/HEK_PRO-seq_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: HEK_PRO-seq_adapter_insertion_distribution.png](files/examples/paper/results_pipeline/HEK_PRO-seq/cutadapt/HEK_PRO-seq_adapter_insertion_distribution.png)
            * :fa-file-text-o: HEK_PRO-seq_R1_cutadapt.txt
            * :fa-file-text-o: HEK_PRO-seq_R2_cutadapt.txt
            * :fa-file-text-o: HEK_PRO-seq.hist
            * :fa-file-text-o: HEK_PRO-seq.histogram
            * :fa-file-text-o: HEK_PRO-seq.extendedFrags.fastq.gz
            * :fa-file-text-o: HEK_PRO-seq.notCombined_1.fastq.gz
            * :fa-file-text-o: HEK_PRO-seq.notCombined_2.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * :fa-file-code-o: HEK_PRO-seq_R1_processed_fastqc.zip
            * [:fa-file-code-o: HEK_PRO-seq_R1_processed_fastqc.html](files/examples/paper/results_pipeline/HEK_PRO-seq/fastqc/HEK_PRO-seq_R1_processed_fastqc.html)
            * :fa-file-text-o: HEK_PRO-seq_R2_trimmed_fastqc.zip
            * [:fa-file-code-o: HEK_PRO-seq_R2_trimmed_fastqc.html](files/examples/paper/results_pipeline/HEK_PRO-seq/fastqc/HEK_PRO-seq_R2_trimmed_fastqc.html)
        * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-archive-o: HEK_PRO-seq_human_rDNA_unmap_R1.fq.gz
            * :fa-file-archive-o: HEK_PRO-seq_human_rDNA_unmap_R2.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: HEK_PRO-seq_bamQC.tsv
            * :fa-file-o: HEK_PRO-seq_exon_intron_ratios.bed.gz
            * [:fa-file-pdf-o: HEK_PRO-seq_cFRiF.pdf](files/examples/paper/results_pipeline/HEK_PRO-seq/QC_hg38/HEK_PRO-seq_cFRiF.pdf)
            * [:fa-file-pdf-o: HEK_PRO-seq_cFRiF.png](files/examples/paper/results_pipeline/HEK_PRO-seq/QC_hg38/HEK_PRO-seq_cFRiF.png)
            * [:fa-file-pdf-o: HEK_PRO-seq_mRNA_contamination.pdf](files/examples/paper/results_pipeline/HEK_PRO-seq/QC_hg38/HEK_PRO-seq_mRNA_contamination.pdf)
            * [:fa-file-pdf-o: HEK_PRO-seq_mRNA_contamination.png](files/examples/paper/results_pipeline/HEK_PRO-seq/QC_hg38/HEK_PRO-seq_mRNA_contamination.png)
            * :fa-file-o: HEK_PRO-seq_pause_index.bed.gz
            * [:fa-file-pdf-o: HEK_PRO-seq_pause_index.pdf](files/examples/paper/results_pipeline/HEK_PRO-seq/QC_hg38/HEK_PRO-seq_pause_index.pdf)
            * [:fa-file-pdf-o: HEK_PRO-seq_pause_index.png](files/examples/paper/results_pipeline/HEK_PRO-seq/QC_hg38/HEK_PRO-seq_pause_index.png)          
            * :fa-file-text-o: HEK_PRO-seq_preseq_counts.txt
            * :fa-file-text-o: HEK_PRO-seq_preseq_out.txt
            * [:fa-file-pdf-o: HEK_PRO-seq_preseq_plot.pdf](files/examples/paper/results_pipeline/HEK_PRO-seq/QC_hg38/HEK_PRO-seq_preseq_plot.pdf)
            * [:fa-file-pdf-o: HEK_PRO-seq_preseq_plot.png](files/examples/paper/results_pipeline/HEK_PRO-seq/QC_hg38/HEK_PRO-seq_preseq_plot.png)
            * :fa-file-text-o: HEK_PRO-seq_preseq_yield.txt
            * [:fa-file-pdf-o: HEK_PRO-seq_FRiF.pdf](files/examples/paper/results_pipeline/HEK_PRO-seq/QC_hg38/HEK_PRO-seq_FRiF.pdf)
            * [:fa-file-pdf-o: HEK_PRO-seq_FRiF.png](files/examples/paper/results_pipeline/HEK_PRO-seq/QC_hg38/HEK_PRO-seq_FRiF.png)
            * [:fa-file-pdf-o: HEK_PRO-seq_TSSenrichment.pdf](files/examples/paper/results_pipeline/HEK_PRO-seq/QC_hg38/HEK_PRO-seq_TSSenrichment.pdf)
            * [:fa-file-pdf-o: HEK_PRO-seq_TSSenrichment.png](files/examples/paper/results_pipeline/HEK_PRO-seq/QC_hg38/HEK_PRO-seq_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: HEK_PRO-seq_r1.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: HEK_PRO-seq_gene_coverage.bed
            * :fa-file-o: HEK_PRO-seq_minus_exact_body_0-mer.bw
            * :fa-file-o: HEK_PRO-seq_plus_exact_body_0.bw
            * :fa-file-o: HEK_PRO-seq_minus_smooth_body_0-mer.bw
            * :fa-file-o: HEK_PRO-seq_plus_smooth_body_0.bw 
        * [:fa-file-text-o: objects.tsv](files/examples/paper/results_pipeline/HEK_PRO-seq/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](files/examples/paper/results_pipeline/HEK_PRO-seq/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](files/examples/paper/results_pipeline/HEK_PRO-seq/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](files/examples/paper/results_pipeline/HEK_PRO-seq/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](files/examples/paper/results_pipeline/HEK_PRO-seq/stats.tsv)
    * ??? danger ":fa-folder-open-o: HelaS3_GRO-seq/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: HelaS3_GRO-seq_fail_qc.bam
            * :fa-file-o: HelaS3_GRO-seq_sort_dups.bam
            * :fa-file-o: HelaS3_GRO-seq_sort.bam
            * :fa-file-o: HelaS3_GRO-seq_PE1.bam
            * :fa-file-o: HelaS3_GRO-seq_PE1.bam.bai
            * :fa-file-o: HelaS3_GRO-seq_unmap.bam
            * :fa-file-o: HelaS3_GRO-seq_plus.bam 
            * :fa-file-o: HelaS3_GRO-seq_plus.bam.bai
            * :fa-file-o: HelaS3_GRO-seq_minus.bam 
            * :fa-file-o: HelaS3_GRO-seq_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: HelaS3_GRO-seq_R1_adapter_insertion_distribution.pdf](files/examples/paper/results_pipeline/HelaS3_GRO-seq/cutadapt/HelaS3_GRO-seq_R1_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: HelaS3_GRO-seq_R1_adapter_insertion_distribution.png](files/examples/paper/results_pipeline/HelaS3_GRO-seq/cutadapt/HelaS3_GRO-seq_R1_adapter_insertion_distribution.png)
            * :fa-file-text-o: HelaS3_GRO-seq_R1_cutadapt.txt
            * :fa-file-text-o: HelaS3_GRO-seq.hist
            * :fa-file-text-o: HelaS3_GRO-seq.histogram
            * :fa-file-text-o: HelaS3_GRO-seq.extendedFrags.fastq.gz
            * :fa-file-text-o: HelaS3_GRO-seq.notCombined_1.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * :fa-file-code-o: HelaS3_GRO-seq_R1_processed_fastqc.zip
            * [:fa-file-code-o: HelaS3_GRO-seq_R1_processed_fastqc.html](files/examples/paper/results_pipeline/HelaS3_GRO-seq/fastqc/HelaS3_GRO-seq_R1_processed_fastqc.html)
        * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-archive-o: HelaS3_GRO-seq_human_rDNA_unmap_R1.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: HelaS3_GRO-seq_bamQC.tsv
            * :fa-file-o: HelaS3_GRO-seq_exon_intron_ratios.bed.gz
            * [:fa-file-pdf-o: HelaS3_GRO-seq_cFRiF.pdf](files/examples/paper/results_pipeline/HelaS3_GRO-seq/QC_hg38/HelaS3_GRO-seq_cFRiF.pdf)
            * [:fa-file-pdf-o: HelaS3_GRO-seq_cFRiF.png](files/examples/paper/results_pipeline/HelaS3_GRO-seq/QC_hg38/HelaS3_GRO-seq_cFRiF.png)
            * [:fa-file-pdf-o: HelaS3_GRO-seq_mRNA_contamination.pdf](files/examples/paper/results_pipeline/HelaS3_GRO-seq/QC_hg38/HelaS3_GRO-seq_mRNA_contamination.pdf)
            * [:fa-file-pdf-o: HelaS3_GRO-seq_mRNA_contamination.png](files/examples/paper/results_pipeline/HelaS3_GRO-seq/QC_hg38/HelaS3_GRO-seq_mRNA_contamination.png)
            * :fa-file-o: HelaS3_GRO-seq_pause_index.bed.gz
            * [:fa-file-pdf-o: HelaS3_GRO-seq_pause_index.pdf](files/examples/paper/results_pipeline/HelaS3_GRO-seq/QC_hg38/HelaS3_GRO-seq_pause_index.pdf)
            * [:fa-file-pdf-o: HelaS3_GRO-seq_pause_index.png](files/examples/paper/results_pipeline/HelaS3_GRO-seq/QC_hg38/HelaS3_GRO-seq_pause_index.png)          
            * [:fa-file-pdf-o: HelaS3_GRO-seq_FRiF.pdf](files/examples/paper/results_pipeline/HelaS3_GRO-seq/QC_hg38/HelaS3_GRO-seq_FRiF.pdf)
            * [:fa-file-pdf-o: HelaS3_GRO-seq_FRiF.png](files/examples/paper/results_pipeline/HelaS3_GRO-seq/QC_hg38/HelaS3_GRO-seq_FRiF.png)
            * [:fa-file-pdf-o: HelaS3_GRO-seq_TSSenrichment.pdf](files/examples/paper/results_pipeline/HelaS3_GRO-seq/QC_hg38/HelaS3_GRO-seq_TSSenrichment.pdf)
            * [:fa-file-pdf-o: HelaS3_GRO-seq_TSSenrichment.png](files/examples/paper/results_pipeline/HelaS3_GRO-seq/QC_hg38/HelaS3_GRO-seq_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: HelaS3_GRO-seq_r1.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: HelaS3_GRO-seq_gene_coverage.bed
            * :fa-file-o: HelaS3_GRO-seq_minus_exact_body_0-mer.bw
            * :fa-file-o: HelaS3_GRO-seq_plus_exact_body_0.bw
            * :fa-file-o: HelaS3_GRO-seq_minus_smooth_body_0-mer.bw
            * :fa-file-o: HelaS3_GRO-seq_plus_smooth_body_0.bw 
        * [:fa-file-text-o: objects.tsv](files/examples/paper/results_pipeline/HelaS3_GRO-seq/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](files/examples/paper/results_pipeline/HelaS3_GRO-seq/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](files/examples/paper/results_pipeline/HelaS3_GRO-seq/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](files/examples/paper/results_pipeline/HelaS3_GRO-seq/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](files/examples/paper/results_pipeline/HelaS3_GRO-seq/stats.tsv)
    * ??? danger ":fa-folder-open-o: Jurkat_ChRO-seq_1/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: Jurkat_ChRO-seq_1_fail_qc.bam
            * :fa-file-o: Jurkat_ChRO-seq_1_sort_dups.bam
            * :fa-file-o: Jurkat_ChRO-seq_1_sort.bam
            * :fa-file-o: Jurkat_ChRO-seq_1_PE1.bam
            * :fa-file-o: Jurkat_ChRO-seq_1_PE1.bam.bai
            * :fa-file-o: Jurkat_ChRO-seq_1_unmap.bam
            * :fa-file-o: Jurkat_ChRO-seq_1_plus.bam 
            * :fa-file-o: Jurkat_ChRO-seq_1_plus.bam.bai
            * :fa-file-o: Jurkat_ChRO-seq_1_minus.bam 
            * :fa-file-o: Jurkat_ChRO-seq_1_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_1_R1_adapter_insertion_distribution.pdf](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_1/cutadapt/Jurkat_ChRO-seq_1_R1_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: Jurkat_ChRO-seq_1_R1_adapter_insertion_distribution.png](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_1/cutadapt/Jurkat_ChRO-seq_1_R1_adapter_insertion_distribution.png)
            * :fa-file-text-o: Jurkat_ChRO-seq_1_R1_cutadapt.txt
            * :fa-file-text-o: Jurkat_ChRO-seq_1.hist
            * :fa-file-text-o: Jurkat_ChRO-seq_1.histogram
            * :fa-file-text-o: Jurkat_ChRO-seq_1.extendedFrags.fastq.gz
            * :fa-file-text-o: Jurkat_ChRO-seq_1.notCombined_1.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * :fa-file-code-o: Jurkat_ChRO-seq_1_R1_processed_fastqc.zip
            * [:fa-file-code-o: Jurkat_ChRO-seq_1_R1_processed_fastqc.html](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_1/fastqc/Jurkat_ChRO-seq_1_R1_processed_fastqc.html)
        * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-archive-o: Jurkat_ChRO-seq_1_human_rDNA_unmap_R1.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: Jurkat_ChRO-seq_1_bamQC.tsv
            * :fa-file-o: Jurkat_ChRO-seq_1_exon_intron_ratios.bed.gz
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_1_cFRiF.pdf](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_1/QC_hg38/Jurkat_ChRO-seq_1_cFRiF.pdf)
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_1_cFRiF.png](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_1/QC_hg38/Jurkat_ChRO-seq_1_cFRiF.png)
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_1_mRNA_contamination.pdf](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_1/QC_hg38/Jurkat_ChRO-seq_1_mRNA_contamination.pdf)
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_1_mRNA_contamination.png](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_1/QC_hg38/Jurkat_ChRO-seq_1_mRNA_contamination.png)
            * :fa-file-o: Jurkat_ChRO-seq_1_pause_index.bed.gz
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_1_pause_index.pdf](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_1/QC_hg38/Jurkat_ChRO-seq_1_pause_index.pdf)
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_1_pause_index.png](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_1/QC_hg38/Jurkat_ChRO-seq_1_pause_index.png)          
            * :fa-file-text-o: Jurkat_ChRO-seq_1_preseq_counts.txt
            * :fa-file-text-o: Jurkat_ChRO-seq_1_preseq_out.txt
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_1_preseq_plot.pdf](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_1/QC_hg38/Jurkat_ChRO-seq_1_preseq_plot.pdf)
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_1_preseq_plot.png](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_1/QC_hg38/Jurkat_ChRO-seq_1_preseq_plot.png)
            * :fa-file-text-o: Jurkat_ChRO-seq_1_preseq_yield.txt
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_1_FRiF.pdf](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_1/QC_hg38/Jurkat_ChRO-seq_1_FRiF.pdf)
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_1_FRiF.png](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_1/QC_hg38/Jurkat_ChRO-seq_1_FRiF.png)
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_1_TSSenrichment.pdf](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_1/QC_hg38/Jurkat_ChRO-seq_1_TSSenrichment.pdf)
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_1_TSSenrichment.png](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_1/QC_hg38/Jurkat_ChRO-seq_1_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: Jurkat_ChRO-seq_1_r1.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: Jurkat_ChRO-seq_1_gene_coverage.bed
            * :fa-file-o: Jurkat_ChRO-seq_1_minus_exact_body_0-mer.bw
            * :fa-file-o: Jurkat_ChRO-seq_1_plus_exact_body_0.bw
            * :fa-file-o: Jurkat_ChRO-seq_1_minus_smooth_body_0-mer.bw
            * :fa-file-o: Jurkat_ChRO-seq_1_plus_smooth_body_0.bw 
        * [:fa-file-text-o: objects.tsv](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_1/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_1/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_1/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_1/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_1/stats.tsv)
    * ??? danger ":fa-folder-open-o: Jurkat_ChRO-seq_2/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: Jurkat_ChRO-seq_2_fail_qc.bam
            * :fa-file-o: Jurkat_ChRO-seq_2_sort_dups.bam
            * :fa-file-o: Jurkat_ChRO-seq_2_sort.bam
            * :fa-file-o: Jurkat_ChRO-seq_2_PE1.bam
            * :fa-file-o: Jurkat_ChRO-seq_2_PE1.bam.bai
            * :fa-file-o: Jurkat_ChRO-seq_2_unmap.bam
            * :fa-file-o: Jurkat_ChRO-seq_2_plus.bam 
            * :fa-file-o: Jurkat_ChRO-seq_2_plus.bam.bai
            * :fa-file-o: Jurkat_ChRO-seq_2_minus.bam 
            * :fa-file-o: Jurkat_ChRO-seq_2_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_2_R1_adapter_insertion_distribution.pdf](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_2/cutadapt/Jurkat_ChRO-seq_2_R1_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: Jurkat_ChRO-seq_2_R1_adapter_insertion_distribution.png](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_2/cutadapt/Jurkat_ChRO-seq_2_R1_adapter_insertion_distribution.png)
            * :fa-file-text-o: Jurkat_ChRO-seq_2_R1_cutadapt.txt
            * :fa-file-text-o: Jurkat_ChRO-seq_2.hist
            * :fa-file-text-o: Jurkat_ChRO-seq_2.histogram
            * :fa-file-text-o: Jurkat_ChRO-seq_2.extendedFrags.fastq.gz
            * :fa-file-text-o: Jurkat_ChRO-seq_2.notCombined_1.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * :fa-file-code-o: Jurkat_ChRO-seq_2_R1_processed_fastqc.zip
            * [:fa-file-code-o: Jurkat_ChRO-seq_2_R1_processed_fastqc.html](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_2/fastqc/Jurkat_ChRO-seq_2_R1_processed_fastqc.html)
        * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-archive-o: Jurkat_ChRO-seq_2_human_rDNA_unmap_R1.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: Jurkat_ChRO-seq_2_bamQC.tsv
            * :fa-file-o: Jurkat_ChRO-seq_2_exon_intron_ratios.bed.gz
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_2_cFRiF.pdf](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_2/QC_hg38/Jurkat_ChRO-seq_2_cFRiF.pdf)
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_2_cFRiF.png](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_2/QC_hg38/Jurkat_ChRO-seq_2_cFRiF.png)
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_2_mRNA_contamination.pdf](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_2/QC_hg38/Jurkat_ChRO-seq_2_mRNA_contamination.pdf)
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_2_mRNA_contamination.png](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_2/QC_hg38/Jurkat_ChRO-seq_2_mRNA_contamination.png)
            * :fa-file-o: Jurkat_ChRO-seq_2_pause_index.bed.gz
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_2_pause_index.pdf](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_2/QC_hg38/Jurkat_ChRO-seq_2_pause_index.pdf)
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_2_pause_index.png](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_2/QC_hg38/Jurkat_ChRO-seq_2_pause_index.png)          
            * :fa-file-text-o: Jurkat_ChRO-seq_2_preseq_counts.txt
            * :fa-file-text-o: Jurkat_ChRO-seq_2_preseq_out.txt
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_2_preseq_plot.pdf](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_2/QC_hg38/Jurkat_ChRO-seq_2_preseq_plot.pdf)
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_2_preseq_plot.png](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_2/QC_hg38/Jurkat_ChRO-seq_2_preseq_plot.png)
            * :fa-file-text-o: Jurkat_ChRO-seq_2_preseq_yield.txt
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_2_FRiF.pdf](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_2/QC_hg38/Jurkat_ChRO-seq_2_FRiF.pdf)
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_2_FRiF.png](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_2/QC_hg38/Jurkat_ChRO-seq_2_FRiF.png)
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_2_TSSenrichment.pdf](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_2/QC_hg38/Jurkat_ChRO-seq_2_TSSenrichment.pdf)
            * [:fa-file-pdf-o: Jurkat_ChRO-seq_2_TSSenrichment.png](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_2/QC_hg38/Jurkat_ChRO-seq_2_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: Jurkat_ChRO-seq_2_r1.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: Jurkat_ChRO-seq_2_gene_coverage.bed
            * :fa-file-o: Jurkat_ChRO-seq_2_minus_exact_body_0-mer.bw
            * :fa-file-o: Jurkat_ChRO-seq_2_plus_exact_body_0.bw
            * :fa-file-o: Jurkat_ChRO-seq_2_minus_smooth_body_0-mer.bw
            * :fa-file-o: Jurkat_ChRO-seq_2_plus_smooth_body_0.bw 
        * [:fa-file-text-o: objects.tsv](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_2/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_2/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_2/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_2/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](files/examples/paper/results_pipeline/Jurkat_ChRO-seq_2/stats.tsv)
    * ??? danger ":fa-folder-open-o: K562_GRO-seq/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: K562_GRO-seq_fail_qc.bam
            * :fa-file-o: K562_GRO-seq_sort_dups.bam
            * :fa-file-o: K562_GRO-seq_sort.bam
            * :fa-file-o: K562_GRO-seq_PE1.bam
            * :fa-file-o: K562_GRO-seq_PE1.bam.bai
            * :fa-file-o: K562_GRO-seq_unmap.bam
            * :fa-file-o: K562_GRO-seq_plus.bam 
            * :fa-file-o: K562_GRO-seq_plus.bam.bai
            * :fa-file-o: K562_GRO-seq_minus.bam 
            * :fa-file-o: K562_GRO-seq_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: K562_GRO-seq_R1_adapter_insertion_distribution.pdf](files/examples/paper/results_pipeline/K562_GRO-seq/cutadapt/K562_GRO-seq_R1_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: K562_GRO-seq_R1_adapter_insertion_distribution.png](files/examples/paper/results_pipeline/K562_GRO-seq/cutadapt/K562_GRO-seq_R1_adapter_insertion_distribution.png)
            * :fa-file-text-o: K562_GRO-seq_R1_cutadapt.txt
            * :fa-file-text-o: K562_GRO-seq.hist
            * :fa-file-text-o: K562_GRO-seq.histogram
            * :fa-file-text-o: K562_GRO-seq.extendedFrags.fastq.gz
            * :fa-file-text-o: K562_GRO-seq.notCombined_1.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * :fa-file-code-o: K562_GRO-seq_R1_processed_fastqc.zip
            * [:fa-file-code-o: K562_GRO-seq_R1_processed_fastqc.html](files/examples/paper/results_pipeline/K562_GRO-seq/fastqc/K562_GRO-seq_R1_processed_fastqc.html)
        * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-archive-o: K562_GRO-seq_human_rDNA_unmap_R1.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: K562_GRO-seq_bamQC.tsv
            * :fa-file-o: K562_GRO-seq_exon_intron_ratios.bed.gz
            * [:fa-file-pdf-o: K562_GRO-seq_cFRiF.pdf](files/examples/paper/results_pipeline/K562_GRO-seq/QC_hg38/K562_GRO-seq_cFRiF.pdf)
            * [:fa-file-pdf-o: K562_GRO-seq_cFRiF.png](files/examples/paper/results_pipeline/K562_GRO-seq/QC_hg38/K562_GRO-seq_cFRiF.png)
            * [:fa-file-pdf-o: K562_GRO-seq_mRNA_contamination.pdf](files/examples/paper/results_pipeline/K562_GRO-seq/QC_hg38/K562_GRO-seq_mRNA_contamination.pdf)
            * [:fa-file-pdf-o: K562_GRO-seq_mRNA_contamination.png](files/examples/paper/results_pipeline/K562_GRO-seq/QC_hg38/K562_GRO-seq_mRNA_contamination.png)
            * :fa-file-o: K562_GRO-seq_pause_index.bed.gz
            * [:fa-file-pdf-o: K562_GRO-seq_pause_index.pdf](files/examples/paper/results_pipeline/K562_GRO-seq/QC_hg38/K562_GRO-seq_pause_index.pdf)
            * [:fa-file-pdf-o: K562_GRO-seq_pause_index.png](files/examples/paper/results_pipeline/K562_GRO-seq/QC_hg38/K562_GRO-seq_pause_index.png)          
            * [:fa-file-pdf-o: K562_GRO-seq_FRiF.pdf](files/examples/paper/results_pipeline/K562_GRO-seq/QC_hg38/K562_GRO-seq_FRiF.pdf)
            * [:fa-file-pdf-o: K562_GRO-seq_FRiF.png](files/examples/paper/results_pipeline/K562_GRO-seq/QC_hg38/K562_GRO-seq_FRiF.png)
            * [:fa-file-pdf-o: K562_GRO-seq_TSSenrichment.pdf](files/examples/paper/results_pipeline/K562_GRO-seq/QC_hg38/K562_GRO-seq_TSSenrichment.pdf)
            * [:fa-file-pdf-o: K562_GRO-seq_TSSenrichment.png](files/examples/paper/results_pipeline/K562_GRO-seq/QC_hg38/K562_GRO-seq_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: K562_GRO-seq_r1.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: K562_GRO-seq_gene_coverage.bed
            * :fa-file-o: K562_GRO-seq_minus_exact_body_0-mer.bw
            * :fa-file-o: K562_GRO-seq_plus_exact_body_0.bw
            * :fa-file-o: K562_GRO-seq_minus_smooth_body_0-mer.bw
            * :fa-file-o: K562_GRO-seq_plus_smooth_body_0.bw 
        * [:fa-file-text-o: objects.tsv](files/examples/paper/results_pipeline/K562_GRO-seq/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](files/examples/paper/results_pipeline/K562_GRO-seq/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](files/examples/paper/results_pipeline/K562_GRO-seq/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](files/examples/paper/results_pipeline/K562_GRO-seq/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](files/examples/paper/results_pipeline/K562_GRO-seq/stats.tsv)
    * ??? danger ":fa-folder-open-o: K562_PRO-seq_100/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: K562_PRO-seq_100_fail_qc.bam
            * :fa-file-o: K562_PRO-seq_100_sort_dups.bam
            * :fa-file-o: K562_PRO-seq_100_sort.bam
            * :fa-file-o: K562_PRO-seq_100_PE1.bam
            * :fa-file-o: K562_PRO-seq_100_PE1.bam.bai
            * :fa-file-o: K562_PRO-seq_100_unmap.bam
            * :fa-file-o: K562_PRO-seq_100_plus.bam 
            * :fa-file-o: K562_PRO-seq_100_plus.bam.bai
            * :fa-file-o: K562_PRO-seq_100_minus.bam 
            * :fa-file-o: K562_PRO-seq_100_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: K562_PRO-seq_R1_adapter_insertion_distribution.pdf](files/examples/paper/results_pipeline/K562_PRO-seq_100/cutadapt/K562_PRO-seq_100_R1_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: K562_PRO-seq_R1_adapter_insertion_distribution.png](files/examples/paper/results_pipeline/K562_PRO-seq_100/cutadapt/K562_PRO-seq_100_R1_adapter_insertion_distribution.png)
            * :fa-file-text-o: K562_PRO-seq_100_R1_cutadapt.txt
            * :fa-file-text-o: K562_PRO-seq_100.hist
            * :fa-file-text-o: K562_PRO-seq_100.histogram
            * :fa-file-text-o: K562_PRO-seq_100.extendedFrags.fastq.gz
            * :fa-file-text-o: K562_PRO-seq_100.notCombined_1.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * :fa-file-code-o: K562_PRO-seq_100_R1_processed_fastqc.zip
            * [:fa-file-code-o: K562_PRO-seq_100_R1_processed_fastqc.html](files/examples/paper/results_pipeline/K562_PRO-seq_100/fastqc/K562_PRO-seq_100_R1_processed_fastqc.html)
        * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-archive-o: K562_PRO-seq_100_human_rDNA_unmap_R1.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: K562_PRO-seq_100_bamQC.tsv
            * :fa-file-o: K562_PRO-seq_100_exon_intron_ratios.bed.gz
            * [:fa-file-pdf-o: K562_PRO-seq_100_cFRiF.pdf](files/examples/paper/results_pipeline/K562_PRO-seq_100/QC_hg38/K562_PRO-seq_100_cFRiF.pdf)
            * [:fa-file-pdf-o: K562_PRO-seq_100_cFRiF.png](files/examples/paper/results_pipeline/K562_PRO-seq_100/QC_hg38/K562_PRO-seq_100_cFRiF.png)
            * [:fa-file-pdf-o: K562_PRO-seq_100_mRNA_contamination.pdf](files/examples/paper/results_pipeline/K562_PRO-seq_100/QC_hg38/K562_PRO-seq_100_mRNA_contamination.pdf)
            * [:fa-file-pdf-o: K562_PRO-seq_100_mRNA_contamination.png](files/examples/paper/results_pipeline/K562_PRO-seq_100/QC_hg38/K562_PRO-seq_100_mRNA_contamination.png)
            * :fa-file-o: K562_PRO-seq_100_pause_index.bed.gz
            * [:fa-file-pdf-o: K562_PRO-seq_100_pause_index.pdf](files/examples/paper/results_pipeline/K562_PRO-seq_100/QC_hg38/K562_PRO-seq_100_pause_index.pdf)
            * [:fa-file-pdf-o: K562_PRO-seq_100_pause_index.png](files/examples/paper/results_pipeline/K562_PRO-seq_100/QC_hg38/K562_PRO-seq_100_pause_index.png)          
            * [:fa-file-pdf-o: K562_PRO-seq_100_FRiF.pdf](files/examples/paper/results_pipeline/K562_PRO-seq_100/QC_hg38/K562_PRO-seq_FRiF.pdf)
            * [:fa-file-pdf-o: K562_PRO-seq_100_FRiF.png](files/examples/paper/results_pipeline/K562_PRO-seq_100/QC_hg38/K562_PRO-seq_100_FRiF.png)
            * [:fa-file-pdf-o: K562_PRO-seq_100_TSSenrichment.pdf](files/examples/paper/results_pipeline/K562_PRO-seq_100/QC_hg38/K562_PRO-seq_100_TSSenrichment.pdf)
            * [:fa-file-pdf-o: K562_PRO-seq_100_TSSenrichment.png](files/examples/paper/results_pipeline/K562_PRO-seq_100/QC_hg38/K562_PRO-seq_100_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: K562_PRO-seq_100_r1.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: K562_PRO-seq_100_gene_coverage.bed
            * :fa-file-o: K562_PRO-seq_100_minus_exact_body_0-mer.bw
            * :fa-file-o: K562_PRO-seq_100_plus_exact_body_0.bw
            * :fa-file-o: K562_PRO-seq_100_minus_smooth_body_0-mer.bw
            * :fa-file-o: K562_PRO-seq_100_plus_smooth_body_0.bw 
        * [:fa-file-text-o: objects.tsv](files/examples/paper/results_pipeline/K562_PRO-seq_100/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](files/examples/paper/results_pipeline/K562_PRO-seq_100/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](files/examples/paper/results_pipeline/K562_PRO-seq_100/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](files/examples/paper/results_pipeline/K562_PRO-seq_100/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](files/examples/paper/results_pipeline/K562_PRO-seq_100/stats.tsv)
    * ??? danger ":fa-folder-open-o: K562_RNA-seq_10/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: K562_RNA-seq_10_fail_qc.bam
            * :fa-file-o: K562_RNA-seq_10_sort_dups.bam
            * :fa-file-o: K562_RNA-seq_10_sort.bam
            * :fa-file-o: K562_RNA-seq_10_PE1.bam
            * :fa-file-o: K562_RNA-seq_10_PE1.bam.bai
            * :fa-file-o: K562_RNA-seq_10_unmap.bam
            * :fa-file-o: K562_RNA-seq_10_plus.bam 
            * :fa-file-o: K562_RNA-seq_10_plus.bam.bai
            * :fa-file-o: K562_RNA-seq_10_minus.bam 
            * :fa-file-o: K562_RNA-seq_10_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: K562_RNA-seq_10_R1_adapter_insertion_distribution.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_10/cutadapt/K562_RNA-seq_10_R1_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: K562_RNA-seq_10_R1_adapter_insertion_distribution.png](files/examples/paper/results_pipeline/K562_RNA-seq_10/cutadapt/K562_RNA-seq_10_R1_adapter_insertion_distribution.png)
            * :fa-file-text-o: K562_RNA-seq_10_R1_cutadapt.txt
            * :fa-file-text-o: K562_RNA-seq_10.hist
            * :fa-file-text-o: K562_RNA-seq_10.histogram
            * :fa-file-text-o: K562_RNA-seq_10.extendedFrags.fastq.gz
            * :fa-file-text-o: K562_RNA-seq_10.notCombined_1.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * :fa-file-code-o: K562_RNA-seq_10_R1_processed_fastqc.zip
            * [:fa-file-code-o: K562_RNA-seq_10_R1_processed_fastqc.html](files/examples/paper/results_pipeline/K562_RNA-seq_10/fastqc/K562_RNA-seq_10_R1_processed_fastqc.html)
        * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-archive-o: K562_RNA-seq_10_human_rDNA_unmap_R1.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: K562_RNA-seq_10_bamQC.tsv
            * :fa-file-o: K562_RNA-seq_10_exon_intron_ratios.bed.gz
            * [:fa-file-pdf-o: K562_RNA-seq_10_cFRiF.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_10/QC_hg38/K562_RNA-seq_10_cFRiF.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_10_cFRiF.png](files/examples/paper/results_pipeline/K562_RNA-seq_10/QC_hg38/K562_RNA-seq_10_cFRiF.png)
            * [:fa-file-pdf-o: K562_RNA-seq_10_mRNA_contamination.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_10/QC_hg38/K562_RNA-seq_10_mRNA_contamination.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_10_mRNA_contamination.png](files/examples/paper/results_pipeline/K562_RNA-seq_10/QC_hg38/K562_RNA-seq_10_mRNA_contamination.png)
            * :fa-file-o: K562_RNA-seq_10_pause_index.bed.gz
            * [:fa-file-pdf-o: K562_RNA-seq_10_pause_index.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_10/QC_hg38/K562_RNA-seq_10_pause_index.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_10_pause_index.png](files/examples/paper/results_pipeline/K562_RNA-seq_10/QC_hg38/K562_RNA-seq_10_pause_index.png)          
            * [:fa-file-pdf-o: K562_RNA-seq_10_FRiF.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_10/QC_hg38/K562_RNA-seq_10_FRiF.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_10_FRiF.png](files/examples/paper/results_pipeline/K562_RNA-seq_10/QC_hg38/K562_RNA-seq_10_FRiF.png)
            * [:fa-file-pdf-o: K562_RNA-seq_10_TSSenrichment.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_10/QC_hg38/K562_RNA-seq_10_TSSenrichment.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_10_TSSenrichment.png](files/examples/paper/results_pipeline/K562_RNA-seq_10/QC_hg38/K562_RNA-seq_10_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: K562_RNA-seq_10_r1.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: K562_RNA-seq_10_gene_coverage.bed
            * :fa-file-o: K562_RNA-seq_10_minus_exact_body_0-mer.bw
            * :fa-file-o: K562_RNA-seq_10_plus_exact_body_0.bw
            * :fa-file-o: K562_RNA-seq_10_minus_smooth_body_0-mer.bw
            * :fa-file-o: K562_RNA-seq_10_plus_smooth_body_0.bw 
        * [:fa-file-text-o: objects.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_10/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](files/examples/paper/results_pipeline/K562_RNA-seq_10/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](files/examples/paper/results_pipeline/K562_RNA-seq_10/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_10/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_10/stats.tsv)
    * ??? danger ":fa-folder-open-o: K562_RNA-seq_20/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: K562_RNA-seq_20_fail_qc.bam
            * :fa-file-o: K562_RNA-seq_20_sort_dups.bam
            * :fa-file-o: K562_RNA-seq_20_sort.bam
            * :fa-file-o: K562_RNA-seq_20_PE1.bam
            * :fa-file-o: K562_RNA-seq_20_PE1.bam.bai
            * :fa-file-o: K562_RNA-seq_20_unmap.bam
            * :fa-file-o: K562_RNA-seq_20_plus.bam 
            * :fa-file-o: K562_RNA-seq_20_plus.bam.bai
            * :fa-file-o: K562_RNA-seq_20_minus.bam 
            * :fa-file-o: K562_RNA-seq_20_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: K562_RNA-seq_20_R1_adapter_insertion_distribution.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_20/cutadapt/K562_RNA-seq_20_R1_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: K562_RNA-seq_20_R1_adapter_insertion_distribution.png](files/examples/paper/results_pipeline/K562_RNA-seq_20/cutadapt/K562_RNA-seq_20_R1_adapter_insertion_distribution.png)
            * :fa-file-text-o: K562_RNA-seq_20_R1_cutadapt.txt
            * :fa-file-text-o: K562_RNA-seq_20.hist
            * :fa-file-text-o: K562_RNA-seq_20.histogram
            * :fa-file-text-o: K562_RNA-seq_20.extendedFrags.fastq.gz
            * :fa-file-text-o: K562_RNA-seq_20.notCombined_1.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * :fa-file-code-o: K562_RNA-seq_20_R1_processed_fastqc.zip
            * [:fa-file-code-o: K562_RNA-seq_20_R1_processed_fastqc.html](files/examples/paper/results_pipeline/K562_RNA-seq_20/fastqc/K562_RNA-seq_20_R1_processed_fastqc.html)
        * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-archive-o: K562_RNA-seq_20_human_rDNA_unmap_R1.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: K562_RNA-seq_20_bamQC.tsv
            * :fa-file-o: K562_RNA-seq_20_exon_intron_ratios.bed.gz
            * [:fa-file-pdf-o: K562_RNA-seq_20_cFRiF.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_20/QC_hg38/K562_RNA-seq_20_cFRiF.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_20_cFRiF.png](files/examples/paper/results_pipeline/K562_RNA-seq_20/QC_hg38/K562_RNA-seq_20_cFRiF.png)
            * [:fa-file-pdf-o: K562_RNA-seq_20_mRNA_contamination.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_20/QC_hg38/K562_RNA-seq_20_mRNA_contamination.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_20_mRNA_contamination.png](files/examples/paper/results_pipeline/K562_RNA-seq_20/QC_hg38/K562_RNA-seq_20_mRNA_contamination.png)
            * :fa-file-o: K562_RNA-seq_20_pause_index.bed.gz
            * [:fa-file-pdf-o: K562_RNA-seq_20_pause_index.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_20/QC_hg38/K562_RNA-seq_20_pause_index.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_20_pause_index.png](files/examples/paper/results_pipeline/K562_RNA-seq_20/QC_hg38/K562_RNA-seq_20_pause_index.png)          
            * [:fa-file-pdf-o: K562_RNA-seq_20_FRiF.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_20/QC_hg38/K562_RNA-seq_20_FRiF.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_20_FRiF.png](files/examples/paper/results_pipeline/K562_RNA-seq_20/QC_hg38/K562_RNA-seq_20_FRiF.png)
            * [:fa-file-pdf-o: K562_RNA-seq_20_TSSenrichment.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_20/QC_hg38/K562_RNA-seq_20_TSSenrichment.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_20_TSSenrichment.png](files/examples/paper/results_pipeline/K562_RNA-seq_20/QC_hg38/K562_RNA-seq_20_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: K562_RNA-seq_20_r1.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: K562_RNA-seq_20_gene_coverage.bed
            * :fa-file-o: K562_RNA-seq_20_minus_exact_body_0-mer.bw
            * :fa-file-o: K562_RNA-seq_20_plus_exact_body_0.bw
            * :fa-file-o: K562_RNA-seq_20_minus_smooth_body_0-mer.bw
            * :fa-file-o: K562_RNA-seq_20_plus_smooth_body_0.bw 
        * [:fa-file-text-o: objects.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_20/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](files/examples/paper/results_pipeline/K562_RNA-seq_20/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](files/examples/paper/results_pipeline/K562_RNA-seq_20/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_20/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_20/stats.tsv)
    * ??? danger ":fa-folder-open-o: K562_RNA-seq_30/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: K562_RNA-seq_30_fail_qc.bam
            * :fa-file-o: K562_RNA-seq_30_sort_dups.bam
            * :fa-file-o: K562_RNA-seq_30_sort.bam
            * :fa-file-o: K562_RNA-seq_30_PE1.bam
            * :fa-file-o: K562_RNA-seq_30_PE1.bam.bai
            * :fa-file-o: K562_RNA-seq_30_unmap.bam
            * :fa-file-o: K562_RNA-seq_30_plus.bam 
            * :fa-file-o: K562_RNA-seq_30_plus.bam.bai
            * :fa-file-o: K562_RNA-seq_30_minus.bam 
            * :fa-file-o: K562_RNA-seq_30_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: K562_RNA-seq_30_R1_adapter_insertion_distribution.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_30/cutadapt/K562_RNA-seq_30_R1_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: K562_RNA-seq_30_R1_adapter_insertion_distribution.png](files/examples/paper/results_pipeline/K562_RNA-seq_30/cutadapt/K562_RNA-seq_30_R1_adapter_insertion_distribution.png)
            * :fa-file-text-o: K562_RNA-seq_30_R1_cutadapt.txt
            * :fa-file-text-o: K562_RNA-seq_30.hist
            * :fa-file-text-o: K562_RNA-seq_30.histogram
            * :fa-file-text-o: K562_RNA-seq_30.extendedFrags.fastq.gz
            * :fa-file-text-o: K562_RNA-seq_30.notCombined_1.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * :fa-file-code-o: K562_RNA-seq_30_R1_processed_fastqc.zip
            * [:fa-file-code-o: K562_RNA-seq_30_R1_processed_fastqc.html](files/examples/paper/results_pipeline/K562_RNA-seq_30/fastqc/K562_RNA-seq_30_R1_processed_fastqc.html)
        * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-archive-o: K562_RNA-seq_30_human_rDNA_unmap_R1.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: K562_RNA-seq_30_bamQC.tsv
            * :fa-file-o: K562_RNA-seq_30_exon_intron_ratios.bed.gz
            * [:fa-file-pdf-o: K562_RNA-seq_30_cFRiF.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_30/QC_hg38/K562_RNA-seq_30_cFRiF.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_30_cFRiF.png](files/examples/paper/results_pipeline/K562_RNA-seq_30/QC_hg38/K562_RNA-seq_30_cFRiF.png)
            * [:fa-file-pdf-o: K562_RNA-seq_30_mRNA_contamination.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_30/QC_hg38/K562_RNA-seq_30_mRNA_contamination.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_30_mRNA_contamination.png](files/examples/paper/results_pipeline/K562_RNA-seq_30/QC_hg38/K562_RNA-seq_30_mRNA_contamination.png)
            * :fa-file-o: K562_RNA-seq_30_pause_index.bed.gz
            * [:fa-file-pdf-o: K562_RNA-seq_30_pause_index.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_30/QC_hg38/K562_RNA-seq_30_pause_index.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_30_pause_index.png](files/examples/paper/results_pipeline/K562_RNA-seq_30/QC_hg38/K562_RNA-seq_30_pause_index.png)          
            * [:fa-file-pdf-o: K562_RNA-seq_30_FRiF.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_30/QC_hg38/K562_RNA-seq_30_FRiF.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_30_FRiF.png](files/examples/paper/results_pipeline/K562_RNA-seq_30/QC_hg38/K562_RNA-seq_30_FRiF.png)
            * [:fa-file-pdf-o: K562_RNA-seq_30_TSSenrichment.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_30/QC_hg38/K562_RNA-seq_30_TSSenrichment.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_30_TSSenrichment.png](files/examples/paper/results_pipeline/K562_RNA-seq_30/QC_hg38/K562_RNA-seq_30_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: K562_RNA-seq_30_r1.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: K562_RNA-seq_30_gene_coverage.bed
            * :fa-file-o: K562_RNA-seq_30_minus_exact_body_0-mer.bw
            * :fa-file-o: K562_RNA-seq_30_plus_exact_body_0.bw
            * :fa-file-o: K562_RNA-seq_30_minus_smooth_body_0-mer.bw
            * :fa-file-o: K562_RNA-seq_30_plus_smooth_body_0.bw 
        * [:fa-file-text-o: objects.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_30/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](files/examples/paper/results_pipeline/K562_RNA-seq_30/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](files/examples/paper/results_pipeline/K562_RNA-seq_30/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_30/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_30/stats.tsv)
    * ??? danger ":fa-folder-open-o: K562_RNA-seq_40/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: K562_RNA-seq_40_fail_qc.bam
            * :fa-file-o: K562_RNA-seq_40_sort_dups.bam
            * :fa-file-o: K562_RNA-seq_40_sort.bam
            * :fa-file-o: K562_RNA-seq_40_PE1.bam
            * :fa-file-o: K562_RNA-seq_40_PE1.bam.bai
            * :fa-file-o: K562_RNA-seq_40_unmap.bam
            * :fa-file-o: K562_RNA-seq_40_plus.bam 
            * :fa-file-o: K562_RNA-seq_40_plus.bam.bai
            * :fa-file-o: K562_RNA-seq_40_minus.bam 
            * :fa-file-o: K562_RNA-seq_40_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: K562_RNA-seq_40_R1_adapter_insertion_distribution.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_40/cutadapt/K562_RNA-seq_40_R1_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: K562_RNA-seq_40_R1_adapter_insertion_distribution.png](files/examples/paper/results_pipeline/K562_RNA-seq_40/cutadapt/K562_RNA-seq_40_R1_adapter_insertion_distribution.png)
            * :fa-file-text-o: K562_RNA-seq_40_R1_cutadapt.txt
            * :fa-file-text-o: K562_RNA-seq_40.hist
            * :fa-file-text-o: K562_RNA-seq_40.histogram
            * :fa-file-text-o: K562_RNA-seq_40.extendedFrags.fastq.gz
            * :fa-file-text-o: K562_RNA-seq_40.notCombined_1.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * :fa-file-code-o: K562_RNA-seq_40_R1_processed_fastqc.zip
            * [:fa-file-code-o: K562_RNA-seq_40_R1_processed_fastqc.html](files/examples/paper/results_pipeline/K562_RNA-seq_40/fastqc/K562_RNA-seq_40_R1_processed_fastqc.html)
        * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-archive-o: K562_RNA-seq_40_human_rDNA_unmap_R1.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: K562_RNA-seq_40_bamQC.tsv
            * :fa-file-o: K562_RNA-seq_40_exon_intron_ratios.bed.gz
            * [:fa-file-pdf-o: K562_RNA-seq_40_cFRiF.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_40/QC_hg38/K562_RNA-seq_40_cFRiF.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_40_cFRiF.png](files/examples/paper/results_pipeline/K562_RNA-seq_40/QC_hg38/K562_RNA-seq_40_cFRiF.png)
            * [:fa-file-pdf-o: K562_RNA-seq_40_mRNA_contamination.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_40/QC_hg38/K562_RNA-seq_40_mRNA_contamination.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_40_mRNA_contamination.png](files/examples/paper/results_pipeline/K562_RNA-seq_40/QC_hg38/K562_RNA-seq_40_mRNA_contamination.png)
            * :fa-file-o: K562_RNA-seq_40_pause_index.bed.gz
            * [:fa-file-pdf-o: K562_RNA-seq_40_pause_index.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_40/QC_hg38/K562_RNA-seq_40_pause_index.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_40_pause_index.png](files/examples/paper/results_pipeline/K562_RNA-seq_40/QC_hg38/K562_RNA-seq_40_pause_index.png)          
            * [:fa-file-pdf-o: K562_RNA-seq_40_FRiF.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_40/QC_hg38/K562_RNA-seq_40_FRiF.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_40_FRiF.png](files/examples/paper/results_pipeline/K562_RNA-seq_40/QC_hg38/K562_RNA-seq_40_FRiF.png)
            * [:fa-file-pdf-o: K562_RNA-seq_40_TSSenrichment.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_40/QC_hg38/K562_RNA-seq_40_TSSenrichment.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_40_TSSenrichment.png](files/examples/paper/results_pipeline/K562_RNA-seq_40/QC_hg38/K562_RNA-seq_40_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: K562_RNA-seq_40_r1.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: K562_RNA-seq_40_gene_coverage.bed
            * :fa-file-o: K562_RNA-seq_40_minus_exact_body_0-mer.bw
            * :fa-file-o: K562_RNA-seq_40_plus_exact_body_0.bw
            * :fa-file-o: K562_RNA-seq_40_minus_smooth_body_0-mer.bw
            * :fa-file-o: K562_RNA-seq_40_plus_smooth_body_0.bw 
        * [:fa-file-text-o: objects.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_40/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](files/examples/paper/results_pipeline/K562_RNA-seq_40/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](files/examples/paper/results_pipeline/K562_RNA-seq_40/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_40/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_40/stats.tsv)
    * ??? danger ":fa-folder-open-o: K562_RNA-seq_50/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: K562_RNA-seq_50_fail_qc.bam
            * :fa-file-o: K562_RNA-seq_50_sort_dups.bam
            * :fa-file-o: K562_RNA-seq_50_sort.bam
            * :fa-file-o: K562_RNA-seq_50_PE1.bam
            * :fa-file-o: K562_RNA-seq_50_PE1.bam.bai
            * :fa-file-o: K562_RNA-seq_50_unmap.bam
            * :fa-file-o: K562_RNA-seq_50_plus.bam 
            * :fa-file-o: K562_RNA-seq_50_plus.bam.bai
            * :fa-file-o: K562_RNA-seq_50_minus.bam 
            * :fa-file-o: K562_RNA-seq_50_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: K562_RNA-seq_50_R1_adapter_insertion_distribution.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_50/cutadapt/K562_RNA-seq_50_R1_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: K562_RNA-seq_50_R1_adapter_insertion_distribution.png](files/examples/paper/results_pipeline/K562_RNA-seq_50/cutadapt/K562_RNA-seq_50_R1_adapter_insertion_distribution.png)
            * :fa-file-text-o: K562_RNA-seq_50_R1_cutadapt.txt
            * :fa-file-text-o: K562_RNA-seq_50.hist
            * :fa-file-text-o: K562_RNA-seq_50.histogram
            * :fa-file-text-o: K562_RNA-seq_50.extendedFrags.fastq.gz
            * :fa-file-text-o: K562_RNA-seq_50.notCombined_1.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * :fa-file-code-o: K562_RNA-seq_50_R1_processed_fastqc.zip
            * [:fa-file-code-o: K562_RNA-seq_50_R1_processed_fastqc.html](files/examples/paper/results_pipeline/K562_RNA-seq_50/fastqc/K562_RNA-seq_50_R1_processed_fastqc.html)
        * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-archive-o: K562_RNA-seq_50_human_rDNA_unmap_R1.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: K562_RNA-seq_50_bamQC.tsv
            * :fa-file-o: K562_RNA-seq_50_exon_intron_ratios.bed.gz
            * [:fa-file-pdf-o: K562_RNA-seq_50_cFRiF.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_50/QC_hg38/K562_RNA-seq_50_cFRiF.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_50_cFRiF.png](files/examples/paper/results_pipeline/K562_RNA-seq_50/QC_hg38/K562_RNA-seq_50_cFRiF.png)
            * [:fa-file-pdf-o: K562_RNA-seq_50_mRNA_contamination.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_50/QC_hg38/K562_RNA-seq_50_mRNA_contamination.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_50_mRNA_contamination.png](files/examples/paper/results_pipeline/K562_RNA-seq_50/QC_hg38/K562_RNA-seq_50_mRNA_contamination.png)
            * :fa-file-o: K562_RNA-seq_50_pause_index.bed.gz
            * [:fa-file-pdf-o: K562_RNA-seq_50_pause_index.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_50/QC_hg38/K562_RNA-seq_50_pause_index.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_50_pause_index.png](files/examples/paper/results_pipeline/K562_RNA-seq_50/QC_hg38/K562_RNA-seq_50_pause_index.png)          
            * [:fa-file-pdf-o: K562_RNA-seq_50_FRiF.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_50/QC_hg38/K562_RNA-seq_50_FRiF.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_50_FRiF.png](files/examples/paper/results_pipeline/K562_RNA-seq_50/QC_hg38/K562_RNA-seq_50_FRiF.png)
            * [:fa-file-pdf-o: K562_RNA-seq_50_TSSenrichment.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_50/QC_hg38/K562_RNA-seq_50_TSSenrichment.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_50_TSSenrichment.png](files/examples/paper/results_pipeline/K562_RNA-seq_50/QC_hg38/K562_RNA-seq_50_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: K562_RNA-seq_50_r1.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: K562_RNA-seq_50_gene_coverage.bed
            * :fa-file-o: K562_RNA-seq_50_minus_exact_body_0-mer.bw
            * :fa-file-o: K562_RNA-seq_50_plus_exact_body_0.bw
            * :fa-file-o: K562_RNA-seq_50_minus_smooth_body_0-mer.bw
            * :fa-file-o: K562_RNA-seq_50_plus_smooth_body_0.bw 
        * [:fa-file-text-o: objects.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_50/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](files/examples/paper/results_pipeline/K562_RNA-seq_50/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](files/examples/paper/results_pipeline/K562_RNA-seq_50/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_50/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_50/stats.tsv)
    * ??? danger ":fa-folder-open-o: K562_RNA-seq_60/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: K562_RNA-seq_60_fail_qc.bam
            * :fa-file-o: K562_RNA-seq_60_sort_dups.bam
            * :fa-file-o: K562_RNA-seq_60_sort.bam
            * :fa-file-o: K562_RNA-seq_60_PE1.bam
            * :fa-file-o: K562_RNA-seq_60_PE1.bam.bai
            * :fa-file-o: K562_RNA-seq_60_unmap.bam
            * :fa-file-o: K562_RNA-seq_60_plus.bam 
            * :fa-file-o: K562_RNA-seq_60_plus.bam.bai
            * :fa-file-o: K562_RNA-seq_60_minus.bam 
            * :fa-file-o: K562_RNA-seq_60_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: K562_RNA-seq_60_R1_adapter_insertion_distribution.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_60/cutadapt/K562_RNA-seq_60_R1_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: K562_RNA-seq_60_R1_adapter_insertion_distribution.png](files/examples/paper/results_pipeline/K562_RNA-seq_60/cutadapt/K562_RNA-seq_60_R1_adapter_insertion_distribution.png)
            * :fa-file-text-o: K562_RNA-seq_60_R1_cutadapt.txt
            * :fa-file-text-o: K562_RNA-seq_60.hist
            * :fa-file-text-o: K562_RNA-seq_60.histogram
            * :fa-file-text-o: K562_RNA-seq_60.extendedFrags.fastq.gz
            * :fa-file-text-o: K562_RNA-seq_60.notCombined_1.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * :fa-file-code-o: K562_RNA-seq_60_R1_processed_fastqc.zip
            * [:fa-file-code-o: K562_RNA-seq_60_R1_processed_fastqc.html](files/examples/paper/results_pipeline/K562_RNA-seq_60/fastqc/K562_RNA-seq_60_R1_processed_fastqc.html)
        * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-archive-o: K562_RNA-seq_60_human_rDNA_unmap_R1.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: K562_RNA-seq_60_bamQC.tsv
            * :fa-file-o: K562_RNA-seq_60_exon_intron_ratios.bed.gz
            * [:fa-file-pdf-o: K562_RNA-seq_60_cFRiF.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_60/QC_hg38/K562_RNA-seq_60_cFRiF.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_60_cFRiF.png](files/examples/paper/results_pipeline/K562_RNA-seq_60/QC_hg38/K562_RNA-seq_60_cFRiF.png)
            * [:fa-file-pdf-o: K562_RNA-seq_60_mRNA_contamination.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_60/QC_hg38/K562_RNA-seq_60_mRNA_contamination.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_60_mRNA_contamination.png](files/examples/paper/results_pipeline/K562_RNA-seq_60/QC_hg38/K562_RNA-seq_60_mRNA_contamination.png)
            * :fa-file-o: K562_RNA-seq_60_pause_index.bed.gz
            * [:fa-file-pdf-o: K562_RNA-seq_60_pause_index.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_60/QC_hg38/K562_RNA-seq_60_pause_index.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_60_pause_index.png](files/examples/paper/results_pipeline/K562_RNA-seq_60/QC_hg38/K562_RNA-seq_60_pause_index.png)          
            * [:fa-file-pdf-o: K562_RNA-seq_60_FRiF.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_60/QC_hg38/K562_RNA-seq_60_FRiF.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_60_FRiF.png](files/examples/paper/results_pipeline/K562_RNA-seq_60/QC_hg38/K562_RNA-seq_60_FRiF.png)
            * [:fa-file-pdf-o: K562_RNA-seq_60_TSSenrichment.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_60/QC_hg38/K562_RNA-seq_60_TSSenrichment.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_60_TSSenrichment.png](files/examples/paper/results_pipeline/K562_RNA-seq_60/QC_hg38/K562_RNA-seq_60_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: K562_RNA-seq_60_r1.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: K562_RNA-seq_60_gene_coverage.bed
            * :fa-file-o: K562_RNA-seq_60_minus_exact_body_0-mer.bw
            * :fa-file-o: K562_RNA-seq_60_plus_exact_body_0.bw
            * :fa-file-o: K562_RNA-seq_60_minus_smooth_body_0-mer.bw
            * :fa-file-o: K562_RNA-seq_60_plus_smooth_body_0.bw 
        * [:fa-file-text-o: objects.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_60/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](files/examples/paper/results_pipeline/K562_RNA-seq_60/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](files/examples/paper/results_pipeline/K562_RNA-seq_60/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_60/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_60/stats.tsv)
    * ??? danger ":fa-folder-open-o: K562_RNA-seq_70/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: K562_RNA-seq_70_fail_qc.bam
            * :fa-file-o: K562_RNA-seq_70_sort_dups.bam
            * :fa-file-o: K562_RNA-seq_70_sort.bam
            * :fa-file-o: K562_RNA-seq_70_PE1.bam
            * :fa-file-o: K562_RNA-seq_70_PE1.bam.bai
            * :fa-file-o: K562_RNA-seq_70_unmap.bam
            * :fa-file-o: K562_RNA-seq_70_plus.bam 
            * :fa-file-o: K562_RNA-seq_70_plus.bam.bai
            * :fa-file-o: K562_RNA-seq_70_minus.bam 
            * :fa-file-o: K562_RNA-seq_70_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: K562_RNA-seq_70_R1_adapter_insertion_distribution.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_70/cutadapt/K562_RNA-seq_70_R1_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: K562_RNA-seq_70_R1_adapter_insertion_distribution.png](files/examples/paper/results_pipeline/K562_RNA-seq_70/cutadapt/K562_RNA-seq_70_R1_adapter_insertion_distribution.png)
            * :fa-file-text-o: K562_RNA-seq_70_R1_cutadapt.txt
            * :fa-file-text-o: K562_RNA-seq_70.hist
            * :fa-file-text-o: K562_RNA-seq_70.histogram
            * :fa-file-text-o: K562_RNA-seq_70.extendedFrags.fastq.gz
            * :fa-file-text-o: K562_RNA-seq_70.notCombined_1.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * :fa-file-code-o: K562_RNA-seq_70_R1_processed_fastqc.zip
            * [:fa-file-code-o: K562_RNA-seq_70_R1_processed_fastqc.html](files/examples/paper/results_pipeline/K562_RNA-seq_70/fastqc/K562_RNA-seq_70_R1_processed_fastqc.html)
         * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-archive-o: K562_RNA-seq_70_human_rDNA_unmap_R1.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: K562_RNA-seq_70_bamQC.tsv
            * :fa-file-o: K562_RNA-seq_70_exon_intron_ratios.bed.gz
            * [:fa-file-pdf-o: K562_RNA-seq_70_cFRiF.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_70/QC_hg38/K562_RNA-seq_70_cFRiF.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_70_cFRiF.png](files/examples/paper/results_pipeline/K562_RNA-seq_70/QC_hg38/K562_RNA-seq_70_cFRiF.png)
            * [:fa-file-pdf-o: K562_RNA-seq_70_mRNA_contamination.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_70/QC_hg38/K562_RNA-seq_70_mRNA_contamination.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_70_mRNA_contamination.png](files/examples/paper/results_pipeline/K562_RNA-seq_70/QC_hg38/K562_RNA-seq_70_mRNA_contamination.png)
            * :fa-file-o: K562_RNA-seq_70_pause_index.bed.gz
            * [:fa-file-pdf-o: K562_RNA-seq_70_pause_index.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_70/QC_hg38/K562_RNA-seq_70_pause_index.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_70_pause_index.png](files/examples/paper/results_pipeline/K562_RNA-seq_70/QC_hg38/K562_RNA-seq_70_pause_index.png)          
            * [:fa-file-pdf-o: K562_RNA-seq_70_FRiF.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_70/QC_hg38/K562_RNA-seq_70_FRiF.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_70_FRiF.png](files/examples/paper/results_pipeline/K562_RNA-seq_70/QC_hg38/K562_RNA-seq_70_FRiF.png)
            * [:fa-file-pdf-o: K562_RNA-seq_70_TSSenrichment.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_70/QC_hg38/K562_RNA-seq_70_TSSenrichment.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_70_TSSenrichment.png](files/examples/paper/results_pipeline/K562_RNA-seq_70/QC_hg38/K562_RNA-seq_70_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: K562_RNA-seq_70_r1.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: K562_RNA-seq_70_gene_coverage.bed
            * :fa-file-o: K562_RNA-seq_70_minus_exact_body_0-mer.bw
            * :fa-file-o: K562_RNA-seq_70_plus_exact_body_0.bw
            * :fa-file-o: K562_RNA-seq_70_minus_smooth_body_0-mer.bw
            * :fa-file-o: K562_RNA-seq_70_plus_smooth_body_0.bw 
        * [:fa-file-text-o: objects.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_70/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](files/examples/paper/results_pipeline/K562_RNA-seq_70/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](files/examples/paper/results_pipeline/K562_RNA-seq_70/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_70/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_70/stats.tsv)
    * ??? danger ":fa-folder-open-o: K562_RNA-seq_80/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: K562_RNA-seq_80_fail_qc.bam
            * :fa-file-o: K562_RNA-seq_80_sort_dups.bam
            * :fa-file-o: K562_RNA-seq_80_sort.bam
            * :fa-file-o: K562_RNA-seq_80_PE1.bam
            * :fa-file-o: K562_RNA-seq_80_PE1.bam.bai
            * :fa-file-o: K562_RNA-seq_80_unmap.bam
            * :fa-file-o: K562_RNA-seq_80_plus.bam 
            * :fa-file-o: K562_RNA-seq_80_plus.bam.bai
            * :fa-file-o: K562_RNA-seq_80_minus.bam 
            * :fa-file-o: K562_RNA-seq_80_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: K562_RNA-seq_80_R1_adapter_insertion_distribution.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_80/cutadapt/K562_RNA-seq_80_R1_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: K562_RNA-seq_80_R1_adapter_insertion_distribution.png](files/examples/paper/results_pipeline/K562_RNA-seq_80/cutadapt/K562_RNA-seq_80_R1_adapter_insertion_distribution.png)
            * :fa-file-text-o: K562_RNA-seq_80_R1_cutadapt.txt
            * :fa-file-text-o: K562_RNA-seq_80.hist
            * :fa-file-text-o: K562_RNA-seq_80.histogram
            * :fa-file-text-o: K562_RNA-seq_80.extendedFrags.fastq.gz
            * :fa-file-text-o: K562_RNA-seq_80.notCombined_1.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * :fa-file-code-o: K562_RNA-seq_80_R1_processed_fastqc.zip
            * [:fa-file-code-o: K562_RNA-seq_80_R1_processed_fastqc.html](files/examples/paper/results_pipeline/K562_RNA-seq_80/fastqc/K562_RNA-seq_80_R1_processed_fastqc.html)
         * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-archive-o: K562_RNA-seq_80_human_rDNA_unmap_R1.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: K562_RNA-seq_80_bamQC.tsv
            * :fa-file-o: K562_RNA-seq_80_exon_intron_ratios.bed.gz
            * [:fa-file-pdf-o: K562_RNA-seq_80_cFRiF.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_80/QC_hg38/K562_RNA-seq_80_cFRiF.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_80_cFRiF.png](files/examples/paper/results_pipeline/K562_RNA-seq_80/QC_hg38/K562_RNA-seq_80_cFRiF.png)
            * [:fa-file-pdf-o: K562_RNA-seq_80_mRNA_contamination.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_80/QC_hg38/K562_RNA-seq_80_mRNA_contamination.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_80_mRNA_contamination.png](files/examples/paper/results_pipeline/K562_RNA-seq_80/QC_hg38/K562_RNA-seq_80_mRNA_contamination.png)
            * :fa-file-o: K562_RNA-seq_80_pause_index.bed.gz
            * [:fa-file-pdf-o: K562_RNA-seq_80_pause_index.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_80/QC_hg38/K562_RNA-seq_80_pause_index.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_80_pause_index.png](files/examples/paper/results_pipeline/K562_RNA-seq_80/QC_hg38/K562_RNA-seq_80_pause_index.png)          
            * [:fa-file-pdf-o: K562_RNA-seq_80_FRiF.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_80/QC_hg38/K562_RNA-seq_80_FRiF.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_80_FRiF.png](files/examples/paper/results_pipeline/K562_RNA-seq_80/QC_hg38/K562_RNA-seq_80_FRiF.png)
            * [:fa-file-pdf-o: K562_RNA-seq_80_TSSenrichment.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_80/QC_hg38/K562_RNA-seq_80_TSSenrichment.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_80_TSSenrichment.png](files/examples/paper/results_pipeline/K562_RNA-seq_80/QC_hg38/K562_RNA-seq_80_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: K562_RNA-seq_80_r1.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: K562_RNA-seq_80_gene_coverage.bed
            * :fa-file-o: K562_RNA-seq_80_minus_exact_body_0-mer.bw
            * :fa-file-o: K562_RNA-seq_80_plus_exact_body_0.bw
            * :fa-file-o: K562_RNA-seq_80_minus_smooth_body_0-mer.bw
            * :fa-file-o: K562_RNA-seq_80_plus_smooth_body_0.bw 
        * [:fa-file-text-o: objects.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_80/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](files/examples/paper/results_pipeline/K562_RNA-seq_80/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](files/examples/paper/results_pipeline/K562_RNA-seq_80/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_80/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_80/stats.tsv)
    * ??? danger ":fa-folder-open-o: K562_RNA-seq_90/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: K562_RNA-seq_90_fail_qc.bam
            * :fa-file-o: K562_RNA-seq_90_sort_dups.bam
            * :fa-file-o: K562_RNA-seq_90_sort.bam
            * :fa-file-o: K562_RNA-seq_90_PE1.bam
            * :fa-file-o: K562_RNA-seq_90_PE1.bam.bai
            * :fa-file-o: K562_RNA-seq_90_unmap.bam
            * :fa-file-o: K562_RNA-seq_90_plus.bam 
            * :fa-file-o: K562_RNA-seq_90_plus.bam.bai
            * :fa-file-o: K562_RNA-seq_90_minus.bam 
            * :fa-file-o: K562_RNA-seq_90_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: K562_RNA-seq_90_R1_adapter_insertion_distribution.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_90/cutadapt/K562_RNA-seq_90_R1_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: K562_RNA-seq_90_R1_adapter_insertion_distribution.png](files/examples/paper/results_pipeline/K562_RNA-seq_90/cutadapt/K562_RNA-seq_90_R1_adapter_insertion_distribution.png)
            * :fa-file-text-o: K562_RNA-seq_90_R1_cutadapt.txt
            * :fa-file-text-o: K562_RNA-seq_90.hist
            * :fa-file-text-o: K562_RNA-seq_90.histogram
            * :fa-file-text-o: K562_RNA-seq_90.extendedFrags.fastq.gz
            * :fa-file-text-o: K562_RNA-seq_90.notCombined_1.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * :fa-file-code-o: K562_RNA-seq_90_R1_processed_fastqc.zip
            * [:fa-file-code-o: K562_RNA-seq_90_R1_processed_fastqc.html](files/examples/paper/results_pipeline/K562_RNA-seq_90/fastqc/K562_RNA-seq_90_R1_processed_fastqc.html)
        * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-archive-o: K562_RNA-seq_90_human_rDNA_unmap_R1.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: K562_RNA-seq_90_bamQC.tsv
            * :fa-file-o: K562_RNA-seq_90_exon_intron_ratios.bed.gz
            * [:fa-file-pdf-o: K562_RNA-seq_90_cFRiF.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_90/QC_hg38/K562_RNA-seq_90_cFRiF.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_90_cFRiF.png](files/examples/paper/results_pipeline/K562_RNA-seq_90/QC_hg38/K562_RNA-seq_90_cFRiF.png)
            * [:fa-file-pdf-o: K562_RNA-seq_90_mRNA_contamination.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_90/QC_hg38/K562_RNA-seq_90_mRNA_contamination.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_90_mRNA_contamination.png](files/examples/paper/results_pipeline/K562_RNA-seq_90/QC_hg38/K562_RNA-seq_90_mRNA_contamination.png)
            * :fa-file-o: K562_RNA-seq_90_pause_index.bed.gz
            * [:fa-file-pdf-o: K562_RNA-seq_90_pause_index.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_90/QC_hg38/K562_RNA-seq_90_pause_index.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_90_pause_index.png](files/examples/paper/results_pipeline/K562_RNA-seq_90/QC_hg38/K562_RNA-seq_90_pause_index.png)          
            * [:fa-file-pdf-o: K562_RNA-seq_90_FRiF.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_90/QC_hg38/K562_RNA-seq_90_FRiF.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_90_FRiF.png](files/examples/paper/results_pipeline/K562_RNA-seq_90/QC_hg38/K562_RNA-seq_90_FRiF.png)
            * [:fa-file-pdf-o: K562_RNA-seq_90_TSSenrichment.pdf](files/examples/paper/results_pipeline/K562_RNA-seq_90/QC_hg38/K562_RNA-seq_90_TSSenrichment.pdf)
            * [:fa-file-pdf-o: K562_RNA-seq_90_TSSenrichment.png](files/examples/paper/results_pipeline/K562_RNA-seq_90/QC_hg38/K562_RNA-seq_90_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: K562_RNA-seq_90_r1.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: K562_RNA-seq_90_gene_coverage.bed
            * :fa-file-o: K562_RNA-seq_90_minus_exact_body_0-mer.bw
            * :fa-file-o: K562_RNA-seq_90_plus_exact_body_0.bw
            * :fa-file-o: K562_RNA-seq_90_minus_smooth_body_0-mer.bw
            * :fa-file-o: K562_RNA-seq_90_plus_smooth_body_0.bw 
        * [:fa-file-text-o: objects.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_90/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](files/examples/paper/results_pipeline/K562_RNA-seq_90/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](files/examples/paper/results_pipeline/K562_RNA-seq_90/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_90/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](files/examples/paper/results_pipeline/K562_RNA-seq_90/stats.tsv)
* ??? danger ":fa-folder-open-o: submission/"
    * :fa-file-code-o: H9_PRO-seq_1.yaml
    * :fa-file-code-o: H9_PRO-seq_2.yaml
    * :fa-file-code-o: H9_PRO-seq_3.yaml
    * :fa-file-code-o: H9_treated_PRO-seq_1.yaml
    * :fa-file-code-o: H9_treated_PRO-seq_2.yaml
    * :fa-file-code-o: H9_treated_PRO-seq_3.yaml
    * :fa-file-code-o: H9_PRO-seq_10.yaml
    * :fa-file-code-o: H9_PRO-seq_20.yaml
    * :fa-file-code-o: H9_PRO-seq_30.yaml
    * :fa-file-code-o: H9_PRO-seq_40.yaml
    * :fa-file-code-o: H9_PRO-seq_50.yaml
    * :fa-file-code-o: H9_PRO-seq_60.yaml
    * :fa-file-code-o: H9_PRO-seq_70.yaml
    * :fa-file-code-o: H9_PRO-seq_80.yaml
    * :fa-file-code-o: H9_PRO-seq_90.yaml
    * :fa-file-code-o: H9_PRO-seq_100.yaml
    * :fa-file-code-o: HEK_ARF_PRO-seq.yaml
    * :fa-file-code-o: HEK_PRO-seq.yaml
    * :fa-file-code-o: HelaS3_GRO-seq.yaml
    * :fa-file-code-o: Jurkat_ChRO-seq_1.yaml
    * :fa-file-code-o: Jurkat_ChRO-seq_2.yaml
    * :fa-file-code-o: K562_GRO-seq.yaml
    * :fa-file-code-o: K562_PRO-seq_02.yaml
    * :fa-file-code-o: K562_PRO-seq_04.yaml
    * :fa-file-code-o: K562_PRO-seq_06.yaml
    * :fa-file-code-o: K562_PRO-seq_08.yaml
    * :fa-file-code-o: K562_PRO-seq_10.yaml
    * :fa-file-code-o: K562_PRO-seq_20.yaml
    * :fa-file-code-o: K562_PRO-seq_30.yaml
    * :fa-file-code-o: K562_PRO-seq_40.yaml
    * :fa-file-code-o: K562_PRO-seq_50.yaml
    * :fa-file-code-o: K562_PRO-seq_60.yaml
    * :fa-file-code-o: K562_PRO-seq_70.yaml
    * :fa-file-code-o: K562_PRO-seq_80.yaml
    * :fa-file-code-o: K562_PRO-seq_90.yaml
    * :fa-file-code-o: K562_PRO-seq_100.yaml
    * :fa-file-code-o: K562_RNA-seq_10.yaml
    * :fa-file-code-o: K562_RNA-seq_20.yaml
    * :fa-file-code-o: K562_RNA-seq_30.yaml
    * :fa-file-code-o: K562_RNA-seq_40.yaml
    * :fa-file-code-o: K562_RNA-seq_50.yaml
    * :fa-file-code-o: K562_RNA-seq_60.yaml
    * :fa-file-code-o: K562_RNA-seq_70.yaml
    * :fa-file-code-o: K562_RNA-seq_80.yaml
    * :fa-file-code-o: K562_RNA-seq_90.yaml
    * :fa-file-code-o: K562_RNA-seq_100.yaml
    * :fa-file-text-o: peppro_H9_PRO-seq_1.log
    * :fa-file-code-o: peppro_H9_PRO-seq_1.sub
    * :fa-file-text-o: peppro_H9_PRO-seq_2.log
    * :fa-file-code-o: peppro_H9_PRO-seq_2.sub
    * :fa-file-text-o: peppro_H9_PRO-seq_3.log
    * :fa-file-code-o: peppro_H9_PRO-seq_3.sub
    * :fa-file-text-o: peppro_H9_treated_PRO-seq_1.log
    * :fa-file-code-o: peppro_H9_treated_PRO-seq_1.sub
    * :fa-file-text-o: peppro_H9_treated_PRO-seq_2.log
    * :fa-file-code-o: peppro_H9_treated_PRO-seq_2.sub
    * :fa-file-text-o: peppro_H9_treated_PRO-seq_3.log
    * :fa-file-code-o: peppro_H9_treated_PRO-seq_3.sub
    * :fa-file-text-o: peppro_H9_PRO-seq_10.log
    * :fa-file-code-o: peppro_H9_PRO-seq_10.sub
    * :fa-file-text-o: peppro_H9_PRO-seq_20.log
    * :fa-file-code-o: peppro_H9_PRO-seq_20.sub
    * :fa-file-text-o: peppro_H9_PRO-seq_30.log
    * :fa-file-code-o: peppro_H9_PRO-seq_30.sub
    * :fa-file-text-o: peppro_H9_PRO-seq_40.log
    * :fa-file-code-o: peppro_H9_PRO-seq_40.sub
    * :fa-file-text-o: peppro_H9_PRO-seq_50.log
    * :fa-file-code-o: peppro_H9_PRO-seq_50.sub
    * :fa-file-text-o: peppro_H9_PRO-seq_60.log
    * :fa-file-code-o: peppro_H9_PRO-seq_60.sub
    * :fa-file-text-o: peppro_H9_PRO-seq_70.log
    * :fa-file-code-o: peppro_H9_PRO-seq_70.sub
    * :fa-file-text-o: peppro_H9_PRO-seq_80.log
    * :fa-file-code-o: peppro_H9_PRO-seq_80.sub
    * :fa-file-text-o: peppro_H9_PRO-seq_90.log
    * :fa-file-code-o: peppro_H9_PRO-seq_90.sub
    * :fa-file-text-o: peppro_H9_PRO-seq_100.log
    * :fa-file-code-o: peppro_H9_PRO-seq_100.sub
    * :fa-file-text-o: peppro_HEK_ARF_PRO-seq.log
    * :fa-file-code-o: peppro_HEK_ARF_PRO-seq.sub
    * :fa-file-text-o: peppro_HEK_PRO-seq.log
    * :fa-file-code-o: peppro_HEK_PRO-seq.sub
    * :fa-file-text-o: peppro_HelaS3_GRO-seq.log
    * :fa-file-code-o: peppro_HelaS3_GRO-seq.sub
    * :fa-file-text-o: peppro_Jurkat_ChRO-seq_1.log
    * :fa-file-code-o: peppro_Jurkat_ChRO-seq_1.sub
    * :fa-file-text-o: peppro_Jurkat_ChRO-seq_2.log
    * :fa-file-code-o: peppro_Jurkat_ChRO-seq_2.sub
    * :fa-file-text-o: peppro_K562_GRO-seq.log
    * :fa-file-code-o: peppro_K562_GRO-seq.sub
    * :fa-file-text-o: peppro_K562_PRO-seq_02.log
    * :fa-file-code-o: peppro_K562_PRO-seq_02.sub
    * :fa-file-text-o: peppro_K562_PRO-seq_04.log
    * :fa-file-code-o: peppro_K562_PRO-seq_04.sub
    * :fa-file-text-o: peppro_K562_PRO-seq_06.log
    * :fa-file-code-o: peppro_K562_PRO-seq_06.sub
    * :fa-file-text-o: peppro_K562_PRO-seq_08.log
    * :fa-file-code-o: peppro_K562_PRO-seq_08.sub
    * :fa-file-text-o: peppro_K562_PRO-seq_10.log
    * :fa-file-code-o: peppro_K562_PRO-seq_10.sub
    * :fa-file-text-o: peppro_K562_PRO-seq_20.log
    * :fa-file-code-o: peppro_K562_PRO-seq_20.sub
    * :fa-file-text-o: peppro_K562_PRO-seq_30.log
    * :fa-file-code-o: peppro_K562_PRO-seq_30.sub
    * :fa-file-text-o: peppro_K562_PRO-seq_40.log
    * :fa-file-code-o: peppro_K562_PRO-seq_40.sub
    * :fa-file-text-o: peppro_K562_PRO-seq_50.log
    * :fa-file-code-o: peppro_K562_PRO-seq_50.sub
    * :fa-file-text-o: peppro_K562_PRO-seq_60.log
    * :fa-file-code-o: peppro_K562_PRO-seq_60.sub
    * :fa-file-text-o: peppro_K562_PRO-seq_70.log
    * :fa-file-code-o: peppro_K562_PRO-seq_70.sub
    * :fa-file-text-o: peppro_K562_PRO-seq_80.log
    * :fa-file-code-o: peppro_K562_PRO-seq_80.sub
    * :fa-file-text-o: peppro_K562_PRO-seq_90.log
    * :fa-file-code-o: peppro_K562_PRO-seq_90.sub
    * :fa-file-text-o: peppro_K562_PRO-seq_100.log
    * :fa-file-code-o: peppro_K562_PRO-seq_100.sub
    * :fa-file-text-o: peppro_K562_RNA-seq_10.log
    * :fa-file-code-o: peppro_K562_RNA-seq_10.sub
    * :fa-file-text-o: peppro_K562_RNA-seq_20.log
    * :fa-file-code-o: peppro_K562_RNA-seq_20.sub
    * :fa-file-text-o: peppro_K562_RNA-seq_30.log
    * :fa-file-code-o: peppro_K562_RNA-seq_30.sub
    * :fa-file-text-o: peppro_K562_RNA-seq_40.log
    * :fa-file-code-o: peppro_K562_RNA-seq_40.sub
    * :fa-file-text-o: peppro_K562_RNA-seq_50.log
    * :fa-file-code-o: peppro_K562_RNA-seq_50.sub
    * :fa-file-text-o: peppro_K562_RNA-seq_60.log
    * :fa-file-code-o: peppro_K562_RNA-seq_60.sub
    * :fa-file-text-o: peppro_K562_RNA-seq_70.log
    * :fa-file-code-o: peppro_K562_RNA-seq_70.sub
    * :fa-file-text-o: peppro_K562_RNA-seq_80.log
    * :fa-file-code-o: peppro_K562_RNA-seq_80.sub
    * :fa-file-text-o: peppro_K562_RNA-seq_90.log
    * :fa-file-code-o: peppro_K562_RNA-seq_90.sub
    * :fa-file-text-o: peppro_K562_RNA-seq_100.log
    * :fa-file-code-o: peppro_K562_RNA-seq_100.sub
* ??? danger ":fa-folder-open-o: summary/"
    * :fa-file-code-o: [PEPPRO collator_commands.sh](files/examples/paper/summary/PEPPRO collator_commands.sh)
    * :fa-file-text-o: PEPPRO collator_completed.flag
    * :fa-file-text-o: [PEPPRO collator_log.md](files/examples/paper/summary/PEPPRO collator_log.md)
    * :fa-file-text-o: [PEPPRO collator_profile.tsv](files/examples/paper/summary/PEPPRO collator_profile.tsv)
    * :fa-file-text-o: [PEPPRO_countData.csv](files/examples/paper/summary/PEPPRO_countData.csv)
    * :fa-file-pdf-o: [PEPPRO_libComplexity.pdf](files/examples/paper/summary/PEPPRO_libComplexity.pdf)
    * :fa-file-image-o: [PEPPRO_libComplexity.png](files/examples/paper/summary/PEPPRO_libComplexity.png)
* :fa-file-text-o: [PEPPRO_assets_summary.tsv](files/examples/paper/PEPPRO_assets_summary.tsv)
* :fa-file-text-o: [PEPPRO_stats_summary.tsv](files/examples/paper/PEPPRO_stats_summary.tsv)
* :fa-file-text-o: [PEPPRO_objs_summary.tsv](files/examples/paper/PEPPRO_objs_summary.tsv)
* :fa-file-code-o: [PEPPRO_summary.html](files/examples/paper/PEPPRO_summary.html)
