# Browse the output directory

This is an interactive display of exactly what results you'll get as output from PEPPRO:

* ??? danger ":fa-folder-open-o: reports/"
    * :fa-file-code-o: [fastqc_report.html](../files/examples/paper/reports/fastqc_report_r1.html)
    * :fa-file-code-o: [adapter_insertion_distribution.html](../files/examples/paper/reports/adapter_insertion_distribution.html)
    * :fa-file-code-o: [library_complexity.html](../files/examples/paper/reports/library_complexity.html)
    * :fa-file-code-o: [frif.html](../files/examples/paper/reports/frif.html)
    * :fa-file-code-o: [prif.html](../files/examples/paper/reports/prif.html)
    * :fa-file-code-o: [mrna_contamination.html](../files/examples/paper/reports/mrna_contamination.html)
    * :fa-file-code-o: [objects.html](../files/examples/paper/reports/objects.html)
    * :fa-file-code-o: [pause_index.html](../files/examples/paper/reports/pause_index.html)
    * :fa-file-code-o: [samples.html](../files/examples/paper/reports/samples.html)
    * :fa-file-code-o: [status.html](../files/examples/paper/reports/status.html)
    * :fa-file-code-o: [tss_enrichment.html](../files/examples/paper/reports/tss_enrichment.html)
    * :fa-file-code-o: [h9_pro-seq_1.html](../files/examples/paper/reports/h9_pro-seq_1.html)
    * :fa-file-code-o: [h9_pro-seq_2.html](../files/examples/paper/reports/h9_pro-seq_2.html)
    * :fa-file-code-o: [h9_pro-seq_3.html](../files/examples/paper/reports/h9_pro-seq_3.html)
    * :fa-file-code-o: [h9_treated_pro-seq_1.html](../files/examples/paper/reports/h9_treated_pro-seq_1.html)
    * :fa-file-code-o: [h9_treated_pro-seq_2.html](../files/examples/paper/reports/h9_treated_pro-seq_2.html)
    * :fa-file-code-o: [h9_treated_pro-seq_3.html](../files/examples/paper/reports/h9_treated_pro-seq_3.html)
    * :fa-file-code-o: [hek_arf_pro-seq.html](../files/examples/paper/reports/hek_arf_pro-seq.html)
    * :fa-file-code-o: [hek_pro-seq.html](../files/examples/paper/reports/hek_pro-seq.html)
    * :fa-file-code-o: [helas3_gro-seq.html](../files/examples/paper/reports/helas3_gro-seq.html)
    * :fa-file-code-o: [jurkat_chro-seq_1.html](../files/examples/paper/reports/jurkat_chro-seq_1.html)
    * :fa-file-code-o: [jurkat_chro-seq_2.html](../files/examples/paper/reports/jurkat_chro-seq_2.html)
    * :fa-file-code-o: [k562_gro-seq.html](../files/examples/paper/reports/k562_gro-seq.html)
    * :fa-file-code-o: [k562_pro-seq.html](../files/examples/paper/reports/k562_pro-seq.html)
    * :fa-file-code-o: [ k562_rna-seq_10.html](../files/examples/paper/reports/ k562_rna-seq_10.html)
    * :fa-file-code-o: [ k562_rna-seq_20.html](../files/examples/paper/reports/ k562_rna-seq_20.html)
    * :fa-file-code-o: [ k562_rna-seq_30.html](../files/examples/paper/reports/ k562_rna-seq_30.html)
    * :fa-file-code-o: [ k562_rna-seq_40.html](../files/examples/paper/reports/ k562_rna-seq_40.html)
    * :fa-file-code-o: [ k562_rna-seq_50.html](../files/examples/paper/reports/ k562_rna-seq_50.html)
    * :fa-file-code-o: [ k562_rna-seq_60.html](../files/examples/paper/reports/ k562_rna-seq_60.html)
    * :fa-file-code-o: [ k562_rna-seq_70.html](../files/examples/paper/reports/ k562_rna-seq_70.html)
    * :fa-file-code-o: [ k562_rna-seq_80.html](../files/examples/paper/reports/ k562_rna-seq_80.html)
    * :fa-file-code-o: [ k562_rna-seq_90.html](../files/examples/paper/reports/ k562_rna-seq_90.html)
* ??? danger ":fa-folder-open-o: results_pipeline/"
    * ??? danger ":fa-folder-open-o: H9_PRO-seq_1/"
        * ??? danger ":fa-folder-open-o: aligned_hg38/"
            * :fa-file-o: H9_PRO-seq_1_fail_qc.bam
            * :fa-file-o: H9_PRO-seq_1_sort_dups.bam
            * :fa-file-o: H9_PRO-seq_1_sort.bam
            * :fa-file-o: H9_PRO-seq_1_PE1.bam
            * :fa-file-o: H9_PRO-seq_1_PE1.bam.bai
            * :fa-file-o: H9_PRO-seq_1_PE2.bam
            * :fa-file-o: H9_PRO-seq_1_dups_PE2.bam
            * :fa-file-o: H9_PRO-seq_1_unmap.bam
            * :fa-file-o: H9_PRO-seq_1_plus.bam 
            * :fa-file-o: H9_PRO-seq_1_plus.bam.bai
            * :fa-file-o: H9_PRO-seq_1_minus.bam 
            * :fa-file-o: H9_PRO-seq_1_minus.bam.bai
        * ??? danger ":fa-folder-open-o: cutadapt/"   
            * [:fa-file-pdf-o: H9_PRO-seq_1_adapter_insertion_distribution.pdf](../files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_adapter_insertion_distribution.pdf)
            * [:fa-file-image-o: H9_PRO-seq_1_adapter_insertion_distribution.png](../files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_adapter_insertion_distribution.png)
            * :fa-file-text-o: H9_PRO-seq_1_R1_cutadapt.txt
            * :fa-file-text-o: H9_PRO-seq_1_R2_cutadapt.txt
            * :fa-file-text-o: H9_PRO-seq_1.hist
            * :fa-file-text-o: H9_PRO-seq_1.histogram
            * :fa-file-text-o: H9_PRO-seq_1.extendedFrags.fastq.gz
            * :fa-file-text-o: H9_PRO-seq_1.notCombined_1.fastq.gz
            * :fa-file-text-o: H9_PRO-seq_1.notCombined_2.fastq.gz
        * ???+ danger ":fa-folder-open-o: fastq/"
        * ??? danger ":fa-folder-open-o: fastqc/"
            * [:fa-file-code-o: H9_PRO-seq_1_R1_processed_fastqc.zip](../files/examples/paper/results_pipeline/H9_PRO-seq_1/fastqc/H9_PRO-seq_1_R1_processed_fastqc.zip)
            * [:fa-file-code-o: H9_PRO-seq_1_R1_processed_fastqc.html](../files/examples/paper/results_pipeline/H9_PRO-seq_1/fastqc/H9_PRO-seq_1_R1_processed_fastqc.html)
            * [:fa-file-text-o: H9_PRO-seq_1_R2_trimmed_fastqc.zip](../files/examples/paper/results_pipeline/H9_PRO-seq_1/fastqc/H9_PRO-seq_1_R2_trimmed_fastqc.zip)
            * [:fa-file-code-o: H9_PRO-seq_1_R2_trimmed_fastqc.html](../files/examples/paper/results_pipeline/H9_PRO-seq_1/fastqc/H9_PRO-seq_1_R2_trimmed_fastqc.html)
        * ??? danger ":fa-folder-open-o: prealignments/"
            * :fa-file-text-o: H9_PRO-seq_1_rCRSd_3k_bt_aln_summary.log
            * :fa-file-archive-o: H9_PRO-seq_1_rCRSd_3k_unmap_R1.fq.gz
            * :fa-file-archive-o: H9_PRO-seq_1_rCRSd_3k_unmap_R2.fq.gz
            * :fa-file-text-o: H9_PRO-seq_1_human_rDNA_bt_aln_summary.log
            * :fa-file-archive-o: H9_PRO-seq_1_human_rDNA_unmap_R1.fq.gz
            * :fa-file-archive-o: H9_PRO-seq_1_human_rDNA_unmap_R2.fq.gz
         * ??? danger ":fa-folder-open-o: QC_hg38/"
            * :fa-file-o: H9_PRO-seq_1_bamQC.tsv
            * [:fa-file-pdf-o: H9_PRO-seq_1_frif.pdf](../files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_frif.pdf)
            * [:fa-file-pdf-o: H9_PRO-seq_1_frif.png](../files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_frif.png)
            * :fa-file-text-o: H9_PRO-seq_1_minus_TssEnrichment.txt
            * [:fa-file-pdf-o: H9_PRO-seq_1_mRNA_contamination.pdf](../files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_mRNA_contamination.pdf)
            * [:fa-file-pdf-o: H9_PRO-seq_1_mRNA_contamination.png](../files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_mRNA_contamination.png)
            * [:fa-file-pdf-o: H9_PRO-seq_1_prif.pdf](../files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_prif.pdf)
            * [:fa-file-pdf-o: H9_PRO-seq_1_prif.png](../files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_prif.png)
            * :fa-file-text-o: H9_PRO-seq_1_plus_TssEnrichment.txt           
            * :fa-file-text-o: H9_PRO-seq_1_preseq_counts.txt
            * :fa-file-text-o: H9_PRO-seq_1_preseq_coverage.txt
            * :fa-file-text-o: H9_PRO-seq_1_preseq_out.txt
            * [:fa-file-pdf-o: H9_PRO-seq_1_preseq_plot.pdf](../files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_preseq_plot.pdf)
            * [:fa-file-pdf-o: H9_PRO-seq_1_preseq_plot.png](../files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_preseq_plot.png)
            * :fa-file-text-o: H9_PRO-seq_1_preseq_yield.txt
            * [:fa-file-pdf-o: H9_PRO-seq_1_TSSenrichment.pdf](../files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_TSSenrichment.pdf)
            * [:fa-file-pdf-o: H9_PRO-seq_1_TSSenrichment.png](../files/examples/paper/results_pipeline/H9_PRO-seq_1/QC_hg38/H9_PRO-seq_1_TSSenrichment.png)
        * ??? danger ":fa-folder-open-o: raw/"
            * :fa-file-archive-o: H9_PRO-seq_1_r1.fastq.gz
            * :fa-file-archive-o: H9_PRO-seq_1_r2.fastq.gz
            * :fa-file-archive-o: hg38_annotations.bed.gz 
        * ??? danger ":fa-folder-open-o: signal_hg38/"
            * :fa-file-o: H9_PRO-seq_1_minus_body_0-mer.bw
            * :fa-file-o: H9_PRO-seq_1_plus_body_0.bw      
        * [:fa-file-text-o: objects.tsv](../files/examples/paper/results_pipeline/H9_PRO-seq_1/objects.tsv) 
        * :fa-file-code-o: PEPPRO_cleanup.sh
        * [:fa-file-code-o: PEPPRO_commands.sh](../files/examples/paper/results_pipeline/H9_PRO-seq_1/PEPPRO_commands.sh)
        * :fa-file-o: PEPPRO_completed.flag
        * [:fa-file-o: PEPPRO_log.md](../files/examples/paper/results_pipeline/H9_PRO-seq_1/PEPPRO_log.md)
        * [:fa-file-text-o: PEPPRO_profile.tsv](../files/examples/paper/results_pipeline/H9_PRO-seq_1/PEPPRO_profile.tsv)
        * [:fa-file-text-o: stats.tsv](../files/examples/paper/results_pipeline/H9_PRO-seq_1/stats.tsv)
* ??? danger ":fa-folder-open-o: submission/"
    * :fa-file-code-o: peppro.py_tutorial.sub
    * :fa-file-code-o: tutorial.yaml
    * :fa-file-text-o: peppro.py_tutorial.log
* ??? danger ":fa-folder-open-o: summary/"
    * :fa-file-pdf-o: tutorial_libComplexity.pdf
    * :fa-file-image-o: tutorial_liComplexity.png
* :fa-file-text-o: [tutorial_stats_summary.tsv](../files/examples/paper/tutorial_stats_summary.tsv)
* :fa-file-code-o: [tutorial_summary.html](../files/examples/paper/tutorial_summary.html)
